<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
    "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en">
<head>
  <!-- 2021-05-20 Thu 13:36 -->
  <meta http-equiv="Content-Type" content=
  "text/html; charset=utf-8" />
  <meta name="viewport" content=
  "width=device-width, initial-scale=1" />
  <title>RNA-seq Leg Vs Body workflow</title>
  <meta name="author" content="David Lewis" />
  <meta name="generator" content="Org Mode" />
  <style type="text/css">
  <!--/*--><![CDATA[/*><!--*/
  .title  { text-align: center;
             margin-bottom: .2em; }
  .subtitle { text-align: center;
              font-size: medium;
              font-weight: bold;
              margin-top:0; }
  .todo   { font-family: monospace; color: red; }
  .done   { font-family: monospace; color: green; }
  .priority { font-family: monospace; color: orange; }
  .tag    { background-color: #eee; font-family: monospace;
            padding: 2px; font-size: 80%; font-weight: normal; }
  .timestamp { color: #bebebe; }
  .timestamp-kwd { color: #5f9ea0; }
  .org-right  { margin-left: auto; margin-right: 0px;  text-align: right; }
  .org-left   { margin-left: 0px;  margin-right: auto; text-align: left; }
  .org-center { margin-left: auto; margin-right: auto; text-align: center; }
  .underline { text-decoration: underline; }
  #postamble p, #preamble p { font-size: 90%; margin: .2em; }
  p.verse { margin-left: 3%; }
  pre {
    border: 1px solid #ccc;
    box-shadow: 3px 3px 3px #eee;
    padding: 8pt;
    font-family: monospace;
    overflow: auto;
    margin: 1.2em;
  }
  pre.src {
    position: relative;
    overflow: auto;
    padding-top: 1.2em;
  }
  pre.src:before {
    display: none;
    position: absolute;
    background-color: white;
    top: -10px;
    right: 10px;
    padding: 3px;
    border: 1px solid black;
  }
  pre.src:hover:before { display: inline; margin-top: 14px;}
  /* Languages per Org manual */
  pre.src-asymptote:before { content: 'Asymptote'; }
  pre.src-awk:before { content: 'Awk'; }
  pre.src-C:before { content: 'C'; }
  /* pre.src-C++ doesn't work in CSS */
  pre.src-clojure:before { content: 'Clojure'; }
  pre.src-css:before { content: 'CSS'; }
  pre.src-D:before { content: 'D'; }
  pre.src-ditaa:before { content: 'ditaa'; }
  pre.src-dot:before { content: 'Graphviz'; }
  pre.src-calc:before { content: 'Emacs Calc'; }
  pre.src-emacs-lisp:before { content: 'Emacs Lisp'; }
  pre.src-fortran:before { content: 'Fortran'; }
  pre.src-gnuplot:before { content: 'gnuplot'; }
  pre.src-haskell:before { content: 'Haskell'; }
  pre.src-hledger:before { content: 'hledger'; }
  pre.src-java:before { content: 'Java'; }
  pre.src-js:before { content: 'Javascript'; }
  pre.src-latex:before { content: 'LaTeX'; }
  pre.src-ledger:before { content: 'Ledger'; }
  pre.src-lisp:before { content: 'Lisp'; }
  pre.src-lilypond:before { content: 'Lilypond'; }
  pre.src-lua:before { content: 'Lua'; }
  pre.src-matlab:before { content: 'MATLAB'; }
  pre.src-mscgen:before { content: 'Mscgen'; }
  pre.src-ocaml:before { content: 'Objective Caml'; }
  pre.src-octave:before { content: 'Octave'; }
  pre.src-org:before { content: 'Org mode'; }
  pre.src-oz:before { content: 'OZ'; }
  pre.src-plantuml:before { content: 'Plantuml'; }
  pre.src-processing:before { content: 'Processing.js'; }
  pre.src-python:before { content: 'Python'; }
  pre.src-R:before { content: 'R'; }
  pre.src-ruby:before { content: 'Ruby'; }
  pre.src-sass:before { content: 'Sass'; }
  pre.src-scheme:before { content: 'Scheme'; }
  pre.src-screen:before { content: 'Gnu Screen'; }
  pre.src-sed:before { content: 'Sed'; }
  pre.src-sh:before { content: 'shell'; }
  pre.src-sql:before { content: 'SQL'; }
  pre.src-sqlite:before { content: 'SQLite'; }
  /* additional languages in org.el's org-babel-load-languages alist */
  pre.src-forth:before { content: 'Forth'; }
  pre.src-io:before { content: 'IO'; }
  pre.src-J:before { content: 'J'; }
  pre.src-makefile:before { content: 'Makefile'; }
  pre.src-maxima:before { content: 'Maxima'; }
  pre.src-perl:before { content: 'Perl'; }
  pre.src-picolisp:before { content: 'Pico Lisp'; }
  pre.src-scala:before { content: 'Scala'; }
  pre.src-shell:before { content: 'Shell Script'; }
  pre.src-ebnf2ps:before { content: 'ebfn2ps'; }
  /* additional language identifiers per "defun org-babel-execute"
       in ob-*.el */
  pre.src-cpp:before  { content: 'C++'; }
  pre.src-abc:before  { content: 'ABC'; }
  pre.src-coq:before  { content: 'Coq'; }
  pre.src-groovy:before  { content: 'Groovy'; }
  /* additional language identifiers from org-babel-shell-names in
     ob-shell.el: ob-shell is the only babel language using a lambda to put
     the execution function name together. */
  pre.src-bash:before  { content: 'bash'; }
  pre.src-csh:before  { content: 'csh'; }
  pre.src-ash:before  { content: 'ash'; }
  pre.src-dash:before  { content: 'dash'; }
  pre.src-ksh:before  { content: 'ksh'; }
  pre.src-mksh:before  { content: 'mksh'; }
  pre.src-posh:before  { content: 'posh'; }
  /* Additional Emacs modes also supported by the LaTeX listings package */
  pre.src-ada:before { content: 'Ada'; }
  pre.src-asm:before { content: 'Assembler'; }
  pre.src-caml:before { content: 'Caml'; }
  pre.src-delphi:before { content: 'Delphi'; }
  pre.src-html:before { content: 'HTML'; }
  pre.src-idl:before { content: 'IDL'; }
  pre.src-mercury:before { content: 'Mercury'; }
  pre.src-metapost:before { content: 'MetaPost'; }
  pre.src-modula-2:before { content: 'Modula-2'; }
  pre.src-pascal:before { content: 'Pascal'; }
  pre.src-ps:before { content: 'PostScript'; }
  pre.src-prolog:before { content: 'Prolog'; }
  pre.src-simula:before { content: 'Simula'; }
  pre.src-tcl:before { content: 'tcl'; }
  pre.src-tex:before { content: 'TeX'; }
  pre.src-plain-tex:before { content: 'Plain TeX'; }
  pre.src-verilog:before { content: 'Verilog'; }
  pre.src-vhdl:before { content: 'VHDL'; }
  pre.src-xml:before { content: 'XML'; }
  pre.src-nxml:before { content: 'XML'; }
  /* add a generic configuration mode; LaTeX export needs an additional
     (add-to-list 'org-latex-listings-langs '(conf " ")) in .emacs */
  pre.src-conf:before { content: 'Configuration File'; }

  table { border-collapse:collapse; }
  caption.t-above { caption-side: top; }
  caption.t-bottom { caption-side: bottom; }
  td, th { vertical-align:top;  }
  th.org-right  { text-align: center;  }
  th.org-left   { text-align: center;   }
  th.org-center { text-align: center; }
  td.org-right  { text-align: right;  }
  td.org-left   { text-align: left;   }
  td.org-center { text-align: center; }
  dt { font-weight: bold; }
  .footpara { display: inline; }
  .footdef  { margin-bottom: 1em; }
  .figure { padding: 1em; }
  .figure p { text-align: center; }
  .equation-container {
    display: table;
    text-align: center;
    width: 100%;
  }
  .equation {
    vertical-align: middle;
  }
  .equation-label {
    display: table-cell;
    text-align: right;
    vertical-align: middle;
  }
  .inlinetask {
    padding: 10px;
    border: 2px solid gray;
    margin: 10px;
    background: #ffffcc;
  }
  #org-div-home-and-up
   { text-align: right; font-size: 70%; white-space: nowrap; }
  textarea { overflow-x: auto; }
  .linenr { font-size: smaller }
  .code-highlighted { background-color: #ffff00; }
  .org-info-js_info-navigation { border-style: none; }
  #org-info-js_console-label
    { font-size: 10px; font-weight: bold; white-space: nowrap; }
  .org-info-js_search-highlight
    { background-color: #ffff00; color: #000000; font-weight: bold; }
  .org-svg { width: 90%; }
  /*]]>*/-->
  </style>
  <link rel="stylesheet" type="text/css" href=
  "https://fniessen.github.io/org-html-themes/src/readtheorg_theme/css/htmlize.css" />
  <link rel="stylesheet" type="text/css" href=
  "https://fniessen.github.io/org-html-themes/src/readtheorg_theme/css/readtheorg.css" />

  <script src=
  "https://ajax.googleapis.com/ajax/libs/jquery/2.1.3/jquery.min.js"></script>
  <script src=
  "https://maxcdn.bootstrapcdn.com/bootstrap/3.3.4/js/bootstrap.min.js"></script>
  <script type="text/javascript" src=
  "https://fniessen.github.io/org-html-themes/src/lib/js/jquery.stickytableheaders.min.js"></script>
  <script type="text/javascript" src=
  "https://fniessen.github.io/org-html-themes/src/readtheorg_theme/js/readtheorg.js"></script>
  <script type="text/javascript">
  // @license magnet:?xt=urn:btih:e95b018ef3580986a04669f1b5879592219e2a7a&dn=public-domain.txt Public Domain
  <!--/*--><![CDATA[/*><!--*/
     function CodeHighlightOn(elem, id)
     {
       var target = document.getElementById(id);
       if(null != target) {
         elem.classList.add("code-highlighted");
         target.classList.add("code-highlighted");
       }
     }
     function CodeHighlightOff(elem, id)
     {
       var target = document.getElementById(id);
       if(null != target) {
         elem.classList.remove("code-highlighted");
         target.classList.remove("code-highlighted");
       }
     }
    /*]]>*///-->
  // @license-end
  </script>
</head>
<body>
  <div id="content">
    <h1 class="title">RNA-seq Leg Vs Body workflow</h1>
    <div id="table-of-contents">
      <h2>Table of Contents</h2>
      <div id="text-table-of-contents">
        <ul>
          <li><a href="#org30cca0e">1. Introduction</a></li>
          <li><a href="#orge6b27e9">2. Galaxy</a></li>
          <li>
            <a href="#org86320d4">3. R functions</a>
            <ul>
              <li><a href="#org03962b2">3.1. txi function</a></li>
              <li><a href="#org0610e93">3.2. Initial Deseq
              Function</a></li>
              <li><a href="#org9b99dde">3.3. ReDeseq</a></li>
              <li><a href="#org34de828">3.4. initialEdgeR</a></li>
              <li><a href="#org3d03b57">3.5. estEdgeR</a></li>
            </ul>
          </li>
          <li>
            <a href="#org41415ac">4. initial R step</a>
            <ul>
              <li>
                <a href="#orgfa0a2ca">4.1. <span class=
                "todo TODO">TODO</span> Deseq</a>
                <ul>
                  <li><a href="#org6ab4d51">4.1.1. Package
                  imports</a></li>
                  <li><a href="#orgfd194bc">4.1.2.
                  Tximport</a></li>
                  <li><a href="#org675fc44">4.1.3. Create the Deseq
                  Object</a></li>
                  <li><a href="#org3f137c4">4.1.4. Do initial data
                  manipulation</a></li>
                  <li><a href="#org4c2e169">4.1.5. Initial
                  selection of the data</a></li>
                  <li><a href="#org94bdf8e">4.1.6. Save lists of
                  the respective up-regulated genes</a></li>
                  <li><a href="#org8c39636">4.1.7. Create a new
                  analysis design</a></li>
                  <li><a href="#org09da6b5">4.1.8. Prepare the data
                  for pheatmap plotting</a></li>
                  <li><a href="#orgc1edbbe">4.1.9. Integrity
                  plots</a></li>
                  <li><a href="#orge9a0f5b">4.1.10. Another Deseq
                  Design</a></li>
                  <li><a href="#org8e695ba">4.1.11. Altered
                  Selection of genes</a></li>
                  <li><a href="#orgdc3f0ca">4.1.12. Save new
                  selection to file</a></li>
                  <li><a href="#org14576d6">4.1.13. Extra integrity
                  plot</a></li>
                  <li><a href="#org2d5aa49">4.1.14. <span class=
                  "todo TODO">TODO</span> Extra EdgeR for venn
                  diagrams</a></li>
                </ul>
              </li>
              <li>
                <a href="#org7f122a2">4.2. EdgeR</a>
                <ul>
                  <li><a href="#org3aa73ed">4.2.1. Initial
                  settings</a></li>
                  <li><a href="#org5a72e06">4.2.2. Do the initial
                  processing of the data</a></li>
                  <li><a href="#org4b6aaee">4.2.3. Partion the data
                  into body and leg tables</a></li>
                  <li><a href="#org0389aee">4.2.4. Save the data
                  tables to a file</a></li>
                  <li><a href="#orgeb5ef68">4.2.5. Find the
                  relative expression values when compared to the
                  control</a></li>
                  <li><a href="#orgcf6e500">4.2.6. Selection of the
                  data</a></li>
                  <li><a href="#org1745de5">4.2.7. Create a csv for
                  bar plotting</a></li>
                  <li><a href="#org2780626">4.2.8. Gets the
                  row-names</a></li>
                  <li><a href="#orga0d7f45">4.2.9. Create a list of
                  dataframes corresponding to the total, leg, and
                  body</a></li>
                  <li><a href="#orgfb8f4b4">4.2.10. Save the
                  genelist to files</a></li>
                </ul>
              </li>
            </ul>
          </li>
          <li>
            <a href="#org0459a36">5. Fasta Manipulation</a>
            <ul>
              <li><a href="#org742e04d">5.1. Get Gene
              names</a></li>
            </ul>
          </li>
          <li><a href="#org599597c">6. Blast</a></li>
          <li>
            <a href="#org123b092">7. Plotting</a>
            <ul>
              <li>
                <a href="#org8d905a3">7.1. Required Packages</a>
                <ul>
                  <li><a href="#org9e11df1">7.1.1. R
                  packages</a></li>
                  <li><a href="#orge14ce24">7.1.2. Python
                  packages</a></li>
                </ul>
              </li>
              <li>
                <a href="#org1d772e8">7.2. Venn Diagrams</a>
                <ul>
                  <li><a href="#orge8eb653">7.2.1. Imports</a></li>
                  <li><a href="#org797d54b">7.2.2. Venn 1</a></li>
                  <li><a href="#org046c289">7.2.3. Venn 2</a></li>
                  <li><a href="#org0a5e26e">7.2.4. Venn 3</a></li>
                  <li><a href="#org8068903">7.2.5. Venn 4</a></li>
                  <li><a href="#org8a20a05">7.2.6. Venn 5</a></li>
                  <li><a href="#org2d731c5">7.2.7. Venn 6</a></li>
                  <li><a href="#orgbbacdd6">7.2.8. Venn 7</a></li>
                  <li><a href="#org14fa8b8">7.2.9. Venn 8</a></li>
                  <li><a href="#orgdb0e21d">7.2.10. Venn 9</a></li>
                </ul>
              </li>
              <li>
                <a href="#org7cc63b1">7.3. Pearson Correlation</a>
                <ul>
                  <li><a href="#org4dc927e">7.3.1.
                  Libraries</a></li>
                  <li><a href="#org35653b5">7.3.2. Import the
                  data</a></li>
                  <li><a href="#org639cbe6">7.3.3. Plot the
                  data</a></li>
                  <li><a href="#orgc4cbc9c">7.3.4. Venn
                  Diagram</a></li>
                </ul>
              </li>
              <li>
                <a href="#org1c5d647">7.4. Dotplot</a>
                <ul>
                  <li><a href="#orgca6a5d7">7.4.1. imports</a></li>
                  <li><a href="#orgf9fc0bb">7.4.2. Theme
                  settings</a></li>
                  <li><a href="#orgd2fbb04">7.4.3. Import</a></li>
                  <li><a href="#orgdc7fda4">7.4.4. Plot the
                  figure</a></li>
                  <li><a href="#orgdb7651d">7.4.5. Save the
                  Figure</a></li>
                </ul>
              </li>
            </ul>
          </li>
        </ul>
      </div>
    </div>
    <div id="outline-container-org30cca0e" class="outline-2">
      <h2 id="org30cca0e"><span class="section-number-2">1</span>
      Introduction</h2>
      <div class="outline-text-2" id="text-1">
        <ul class="org-ul">
          <li>The Ticks are sprayed with Permethrin and DEET</li>
          <li>The Legs are separated from the bodies of the ticks
          and analyzed separately</li>
          <li>The Fasta files + assembly are sent to Galaxy</li>
        </ul>
      </div>
    </div>
    <div id="outline-container-orge6b27e9" class="outline-2">
      <h2 id="orge6b27e9"><span class="section-number-2">2</span>
      Galaxy</h2>
      <div class="outline-text-2" id="text-2">
        <ul class="org-ul">
          <li>Import the fasta files + assembly to Galaxy</li>
          <li>Run FastQC to determine data integrity and quality on
          each fasta</li>
          <li>Run Trimmomatic to trim superfluous data from the
          fasta files</li>
          <li>Run A fish analysis tool on the data (salmon,
          sailfish, kallisto quant) on each fasta file from
          trimmomatic. This requires the assembly as an input</li>
          <li>Export the quant files for local analysis with R</li>
        </ul>
      </div>
    </div>
    <div id="outline-container-org86320d4" class="outline-2">
      <h2 id="org86320d4"><span class="section-number-2">3</span> R
      functions</h2>
      <div class="outline-text-2" id="text-3"></div>
      <div id="outline-container-org03962b2" class="outline-3">
        <h3 id="org03962b2"><span class=
        "section-number-3">3.1</span> txi function</h3>
        <div class="outline-text-3" id="text-3-1">
          <p>uses the tximport package to import the quant data.
          The tximport package is used in both deseq and edgeR for
          data import. It is optional, but saves a lot of time on
          initial data organization</p>
          <div class="org-src-container">
            <pre class="src src-jupyter-R"><span style=
            "color: #b8bb26;">txi</span> <span style=
            "color: #d3869b;">&lt;-</span> <span style=
            "color: #fb4934;">function</span>(directory, quantdatafoldername, metadata, fishtype){
  <span style="color: #d3869b;">library</span>(tximport)
  <span style="color: #d3869b;">setwd</span>(directory)
  metadata <span style=
"color: #d3869b;">&lt;-</span> read.csv(metadata, header=<span style="color: #fabd2f;">TRUE</span>)
  files <span style=
"color: #d3869b;">&lt;-</span> file.path(directory, quantdatafoldername, metadata$name)
  names(files) <span style=
"color: #d3869b;">&lt;-</span> metadata$name
  genes <span style=
"color: #d3869b;">&lt;-</span> read.table(<span style=
"color: #b8bb26;">"genes2.tabular"</span>)
  txi <span style=
"color: #d3869b;">&lt;-</span> tximport(files, type=fishtype, tx2gene=genes)
  <span style="color: #fb4934;">return</span>(list(txi, metadata))
}
</pre>
          </div>
        </div>
      </div>
      <div id="outline-container-org0610e93" class="outline-3">
        <h3 id="org0610e93"><span class=
        "section-number-3">3.2</span> Initial Deseq Function</h3>
        <div class="outline-text-3" id="text-3-2">
          <p>Sets up the deseq data set object and returns it.
          Deseq uses a linear model to fit the data. Read more
          about it here <a href=
          "https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#theory-behind-deseq2">
          Deseq2</a>.</p>
          <div class="org-src-container">
            <pre class="src src-jupyter-R"><span style=
            "color: #b8bb26;">initialDeseq</span> <span style=
            "color: #d3869b;">&lt;-</span> <span style=
            "color: #fb4934;">function</span>(txi, metadata, design=~part + pesticides){
  <span style="color: #d3869b;">library</span>(DESeq2)
  <span style="color: #928374;"># </span><span style=
"color: #928374;">Deseq2 dataset from tximport</span>
  data <span style=
"color: #d3869b;">&lt;-</span> DESeqDataSetFromTximport(txi, colData=metadata, design = design)
  Deseq <span style="color: #d3869b;">&lt;-</span> DESeq(data)
  results <span style="color: #d3869b;">&lt;-</span> results(Deseq)
  print(summary(results))
  <span style="color: #fb4934;">return</span>(list(Deseq, results))
}
</pre>
          </div>
        </div>
      </div>
      <div id="outline-container-org9b99dde" class="outline-3">
        <h3 id="org9b99dde"><span class=
        "section-number-3">3.3</span> ReDeseq</h3>
        <div class="outline-text-3" id="text-3-3">
          <p>sets up the deseq object with a different design
          criteria. The design changes how log2foldchange is
          expressed. It is more difficult to get multifactorial
          analysis in a single deseq object than with edgeR.
          Changing the design is necessary for complicated
          analysis.</p>
          <div class="org-src-container">
            <pre class="src src-jupyter-R"><span style=
            "color: #b8bb26;">reDeseq</span> <span style=
            "color: #d3869b;">&lt;-</span> <span style=
            "color: #fb4934;">function</span>(dds){
  ddsMF <span style="color: #d3869b;">&lt;-</span> dds
  design(ddsMF) <span style=
"color: #d3869b;">&lt;-</span> formula(~pesticides)
  ddsMF <span style="color: #d3869b;">&lt;-</span> DESeq(ddsMF)
  Mres <span style="color: #d3869b;">&lt;-</span> results(ddsMF)
  <span style="color: #fb4934;">return</span>(list(ddsMf, Mres))
  }
</pre>
          </div>
        </div>
      </div>
      <div id="outline-container-org34de828" class="outline-3">
        <h3 id="org34de828"><span class=
        "section-number-3">3.4</span> initialEdgeR</h3>
        <div class="outline-text-3" id="text-3-4">
          <p>Sets up the EdgeR object and returns it. EdgeR uses a
          different linear model. Read more about it here <a href=
          "https://www.bioconductor.org/packages/release/bioc/vignettes/edgeR/inst/doc/edgeRUsersGuide.pdf">
          edgeR</a>.</p>
          <div class="org-src-container">
            <pre class="src src-jupyter-R"><span style=
            "color: #b8bb26;">initialEdgeR</span> <span style=
            "color: #d3869b;">&lt;-</span> <span style=
            "color: #fb4934;">function</span>(txi, meta){
  <span style="color: #d3869b;">library</span>(edgeR)
  <span style="color: #928374;"># </span><span style=
"color: #928374;">EdgR dataset from tximport</span>
  counts <span style="color: #d3869b;">&lt;-</span> txi$counts
  length <span style="color: #d3869b;">&lt;-</span> txi$length
  normMat <span style=
"color: #d3869b;">&lt;-</span> length / exp(rowMeans(log(length)))
  normCts <span style=
"color: #d3869b;">&lt;-</span> counts / normMat
  eff.lib <span style=
"color: #d3869b;">&lt;-</span> calcNormFactors(normCts) * colSums(normCts)
  normMat <span style=
"color: #d3869b;">&lt;-</span> sweep(normMat, <span style=
"color: #d3869b; font-weight: bold;">2</span>, eff.lib, <span style="color: #b8bb26;">"*"</span>)
  normMat <span style="color: #d3869b;">&lt;-</span> log(normMat)
  y <span style="color: #d3869b;">&lt;-</span> DGEList(counts)
  y <span style=
"color: #d3869b;">&lt;-</span> scaleOffset(y, normMat)
  
  pest <span style=
"color: #d3869b;">&lt;-</span> factor(meta$pesticides, levels=c(<span style="color: #b8bb26;">"perm"</span>, <span style="color: #b8bb26;">"deet"</span>, <span style="color: #b8bb26;">"control"</span>))
  part <span style=
"color: #d3869b;">&lt;-</span> factor(meta$part, levels=c(<span style="color: #b8bb26;">"leg"</span>, <span style="color: #b8bb26;">"body"</span>))
  design <span style=
"color: #d3869b;">&lt;-</span> model.matrix(~<span style=
"color: #d3869b; font-weight: bold;">0</span> + pest + part)
  keep <span style=
"color: #d3869b;">&lt;-</span> filterByExpr(y, design)
  EdgeR <span style="color: #d3869b;">&lt;-</span> y[keep, ]
  <span style="color: #fb4934;">return</span>(list(EdgeR, design))
}
</pre>
          </div>
        </div>
      </div>
      <div id="outline-container-org3d03b57" class="outline-3">
        <h3 id="org3d03b57"><span class=
        "section-number-3">3.5</span> estEdgeR</h3>
        <div class="outline-text-3" id="text-3-5">
          <p>finds the fit of the data and returns it for further
          analysis. Uses the edgeR functions to estimate dispersion
          of the genes. It is much faster than deseq2.</p>
          <div class="org-src-container">
            <pre class="src src-jupyter-R"><span style=
            "color: #b8bb26;">estEdgeR</span> <span style=
            "color: #d3869b;">&lt;-</span> <span style=
            "color: #fb4934;">function</span>(EdgeR, design){
  <span style="color: #d3869b;">library</span>(edgeR)
  y <span style=
"color: #d3869b;">&lt;-</span> estimateGLMCommonDisp(EdgeR, design)
  y <span style=
"color: #d3869b;">&lt;-</span> estimateGLMTrendedDisp(y, design)
  y <span style=
"color: #d3869b;">&lt;-</span> estimateGLMTagwiseDisp(y, design)
  fit <span style=
"color: #d3869b;">&lt;-</span> glmQLFit(y, design) 
  
  <span style="color: #fb4934;">return</span>(fit)
}
</pre>
          </div>
        </div>
      </div>
    </div>
    <div id="outline-container-org41415ac" class="outline-2">
      <h2 id="org41415ac"><span class="section-number-2">4</span>
      initial R step</h2>
      <div class="outline-text-2" id="text-4">
        <p>EdgeR and Deseq2 are used to find lists of genes that
        are up-regulated and down-regulated when compared to design
        criteria. The design is used to curated the value of
        logfoldchange in Deseq2. They are compared using a pearson
        correlation in the plotting section.</p>
      </div>
      <div id="outline-container-orgfa0a2ca" class="outline-3">
        <h3 id="orgfa0a2ca"><span class=
        "section-number-3">4.1</span> <span class=
        "todo TODO">TODO</span> Deseq</h3>
        <div class="outline-text-3" id="text-4-1">
          <p>This needs to incorporate the function definitions.
          Much of it is being restated in an unintuitive way</p>
        </div>
        <div id="outline-container-org6ab4d51" class="outline-4">
          <h4 id="org6ab4d51"><span class=
          "section-number-4">4.1.1</span> Package imports</h4>
          <div class="outline-text-4" id="text-4-1-1">
            <div class="org-src-container">
              <pre class=
              "src src-jupyter-R">vignette(package=<span style=
              "color: #b8bb26;">"dplyr"</span>)

rdata <span style=
"color: #d3869b;">&lt;-</span> list.files(pattern=<span style=
"color: #b8bb26;">"\\.Rdata"</span>)
args <span style=
"color: #d3869b;">&lt;-</span> commandArgs(trailingOnly = <span style="color: #fabd2f;">TRUE</span>)
<span style=
"color: #fb4934;">if</span>(length(args) &gt; <span style=
"color: #d3869b; font-weight: bold;">0</span>){
    <span style="color: #fb4934;">if</span>(args == <span style=
"color: #b8bb26;">"rerun"</span>){
        print(<span style=
"color: #b8bb26;">"rerunning data"</span>)
        r
    }
}
<span style="color: #928374;"># </span><span style=
"color: #928374;">general library imports ----</span>
<span style="color: #d3869b;">library</span>(tximport)
<span style="color: #d3869b;">library</span>(readr)
<span style="color: #d3869b;">library</span>(<span style=
"color: #b8bb26;">"pheatmap"</span>)
</pre>
            </div>
          </div>
        </div>
        <div id="outline-container-orgfd194bc" class="outline-4">
          <h4 id="orgfd194bc"><span class=
          "section-number-4">4.1.2</span> Tximport</h4>
          <div class="outline-text-4" id="text-4-1-2">
            <div class="org-src-container">
              <pre class="src src-jupyter-R"><span style=
              "color: #928374;">#</span><span style=
              "color: #928374;">change these values to your own situations</span>
directory <span style="color: #d3869b;">&lt;-</span> <span style=
"color: #b8bb26;">"/home/david/Documents/BenoitLab/R/largeRNASEQ"</span>
quantdatafoldername <span style=
"color: #d3869b;">&lt;-</span> <span style=
"color: #b8bb26;">"CorrectedQuantData"</span>
fishtype <span style="color: #d3869b;">&lt;-</span> <span style=
"color: #b8bb26;">"sailfish"</span>
</pre>
            </div>
          </div>
          <ol class="org-ol">
            <li>
              <a id="org658ec91" name="org658ec91"></a>Convert
              quant files to txi object<br />
              <div class="outline-text-5" id="text-4-1-2-1">
                <div class="org-src-container">
                  <pre class="src src-jupyter-R"><span style=
                  "color: #928374;"># </span><span style=
                  "color: #928374;">imports the genes as a tximport file</span>
<span style="color: #d3869b;">setwd</span>(directory)
metadata <span style=
"color: #d3869b;">&lt;-</span> read.csv(<span style=
"color: #b8bb26;">"metadata.csv"</span>, header=<span style=
"color: #fabd2f;">TRUE</span>)
files <span style=
"color: #d3869b;">&lt;-</span> file.path(directory, quantdatafoldername, metadata$name)
names(files) <span style=
"color: #d3869b;">&lt;-</span> metadata$name
genes <span style=
"color: #d3869b;">&lt;-</span> read.table(<span style=
"color: #b8bb26;">"genes2.tabular"</span>)
txi <span style=
"color: #d3869b;">&lt;-</span> tximport(files, type=fishtype, tx2gene=genes)
</pre>
                </div>
                <pre class="example" id="orgdc6b2c0">
reading in files with read_tsv

1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17


summarizing abundance

summarizing counts

summarizing length
</pre>
              </div>
            </li>
          </ol>
        </div>
        <div id="outline-container-org675fc44" class="outline-4">
          <h4 id="org675fc44"><span class=
          "section-number-4">4.1.3</span> Create the Deseq
          Object</h4>
          <div class="outline-text-4" id="text-4-1-3">
            <div class="org-src-container">
              <pre class="src src-jupyter-R"><span style=
              "color: #d3869b;">library</span>(DESeq2)
<span style="color: #928374;"># </span><span style=
"color: #928374;">Deseq2 dataset from tximport</span>
data <span style=
"color: #d3869b;">&lt;-</span> DESeqDataSetFromTximport(txi, colData=metadata, design = ~part + pesticides)
Deseq <span style="color: #d3869b;">&lt;-</span> DESeq(data)
results <span style="color: #d3869b;">&lt;-</span> results(Deseq)
</pre>
            </div>
            <pre class="example" id="orga4837b5">
Loading required package: S4Vectors

Loading required package: stats4

Loading required package: BiocGenerics

Loading required package: parallel


Attaching package: ‘BiocGenerics’


The following objects are masked from ‘package:parallel’:

    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
    clusterExport, clusterMap, parApply, parCapply, parLapply,
    parLapplyLB, parRapply, parSapply, parSapplyLB


The following object is masked from ‘package:limma’:

    plotMA


The following objects are masked from ‘package:stats’:

    IQR, mad, sd, var, xtabs


The following objects are masked from ‘package:base’:

    anyDuplicated, append, as.data.frame, basename, cbind, colnames,
    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,
    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,
    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,
    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,
    union, unique, unsplit, which.max, which.min



Attaching package: ‘S4Vectors’


The following object is masked from ‘package:base’:

    expand.grid


Loading required package: IRanges

Loading required package: GenomicRanges

Loading required package: GenomeInfoDb

Loading required package: SummarizedExperiment

Loading required package: MatrixGenerics

Loading required package: matrixStats


Attaching package: ‘MatrixGenerics’


The following objects are masked from ‘package:matrixStats’:

    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,
    colCounts, colCummaxs, colCummins, colCumprods, colCumsums,
    colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,
    colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,
    colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,
    colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,
    colWeightedMeans, colWeightedMedians, colWeightedSds,
    colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,
    rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,
    rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,
    rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,
    rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,
    rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,
    rowWeightedMads, rowWeightedMeans, rowWeightedMedians,
    rowWeightedSds, rowWeightedVars


Loading required package: Biobase

Welcome to Bioconductor

    Vignettes contain introductory material; view with
    'browseVignettes()'. To cite Bioconductor, see
    'citation("Biobase")', and for packages 'citation("pkgname")'.



Attaching package: ‘Biobase’


The following object is masked from ‘package:MatrixGenerics’:

    rowMedians


The following objects are masked from ‘package:matrixStats’:

    anyMissing, rowMedians


Warning message in DESeqDataSet(se, design = design, ignoreRank):
“some variables in design formula are characters, converting to factors”
using counts and average transcript lengths from tximport

estimating size factors

using 'avgTxLength' from assays(dds), correcting for library size

estimating dispersions

gene-wise dispersion estimates

mean-dispersion relationship

final dispersion estimates

fitting model and testing
</pre>
          </div>
        </div>
        <div id="outline-container-org3f137c4" class="outline-4">
          <h4 id="org3f137c4"><span class=
          "section-number-4">4.1.4</span> Do initial data
          manipulation</h4>
          <div class="outline-text-4" id="text-4-1-4">
            <p>The part design is used to compare the leg to the
            body in total, disregarding pesticide. This is
            explained in detail in the pearson correlation
            section.</p>
            <div class="org-src-container">
              <pre class="src src-jupyter-R"><span style=
              "color: #928374;">## </span><span style=
              "color: #928374;">hello</span>
dds <span style="color: #d3869b;">&lt;-</span> Deseq
<span style="color: #928374;">#</span><span style=
"color: #928374;">* vsd ntd rld</span>
vsd <span style=
"color: #d3869b;">&lt;-</span> vst(dds, blind=<span style=
"color: #fabd2f;">FALSE</span>)
rld <span style=
"color: #d3869b;">&lt;-</span> rlog(dds, blind=<span style=
"color: #fabd2f;">FALSE</span>)
ntd <span style="color: #d3869b;">&lt;-</span> normTransform(dds)

ddsMF <span style="color: #d3869b;">&lt;-</span> dds
design(ddsMF) <span style=
"color: #d3869b;">&lt;-</span> formula(~pesticides)
ddsMF <span style="color: #d3869b;">&lt;-</span> DESeq(ddsMF)
Mres <span style="color: #d3869b;">&lt;-</span> results(ddsMF)
design(dds) <span style=
"color: #d3869b;">&lt;-</span> formula(~part)
part <span style="color: #d3869b;">&lt;-</span> DESeq(dds)
res <span style="color: #d3869b;">&lt;-</span> results(part)
</pre>
            </div>
            <pre class="example" id="org14e81fa">
using pre-existing normalization factors

estimating dispersions

found already estimated dispersions, replacing these

gene-wise dispersion estimates

mean-dispersion relationship

final dispersion estimates

fitting model and testing

using pre-existing normalization factors

estimating dispersions

found already estimated dispersions, replacing these

gene-wise dispersion estimates

mean-dispersion relationship

final dispersion estimates

fitting model and testing

-- replacing outliers and refitting for 4107 genes
-- DESeq argument 'minReplicatesForReplace' = 7
-- original counts are preserved in counts(dds)

estimating dispersions

fitting model and testing
</pre>
          </div>
        </div>
        <div id="outline-container-org4c2e169" class="outline-4">
          <h4 id="org4c2e169"><span class=
          "section-number-4">4.1.5</span> Initial selection of the
          data</h4>
          <div class="outline-text-4" id="text-4-1-5">
            <ul class="org-ul">
              <li>p value of less than 0.01</li>
              <li>Leg is compared to body, therefore down-regulated
              are up-regulated in body</li>
            </ul>
            <div class="org-src-container">
              <pre class="src src-jupyter-R">DR <span style=
              "color: #d3869b;">&lt;-</span> Mres[ which(Mres$padj &lt; <span style="color: #d3869b; font-weight: bold;">0.01</span>), ]
Body <span style=
"color: #d3869b;">&lt;-</span> DR[which(DR$log2FoldChange &lt; -<span style="color: #d3869b; font-weight: bold;">2</span>),]
Leg <span style=
"color: #d3869b;">&lt;-</span> DR[which(DR$log2FoldChange &gt; <span style="color: #d3869b; font-weight: bold;">2</span>), ]
</pre>
            </div>
            <pre class="example" id="org2860f68">
log2 fold change (MLE): pesticides perm vs control
Wald test p-value: pesticides perm vs control
DataFrame with 6 rows and 6 columns
            baseMean log2FoldChange     lfcSE       stat    pvalue      padj
           &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt;  &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
dv.1      47203.3539       0.365233  0.488997  0.7469016  0.455123  0.999996
dv.10       724.7318       0.116932  1.232421  0.0948798  0.924410  0.999996
dv.100     1474.4296       0.116376  0.297011  0.3918226  0.695189  0.999996
dv.1000     246.0981       1.380686  0.909599  1.5179057  0.129038  0.999996
dv.10000     56.9568      -0.744815  1.389576 -0.5360020  0.591957  0.999996
dv.100000    19.8319       0.319016  0.972153  0.3281536  0.742796  0.999996
</pre>
          </div>
        </div>
        <div id="outline-container-org94bdf8e" class="outline-4">
          <h4 id="org94bdf8e"><span class=
          "section-number-4">4.1.6</span> Save lists of the
          respective up-regulated genes</h4>
          <div class="outline-text-4" id="text-4-1-6">
            <div class="org-src-container">
              <pre class=
              "src src-jupyter-R">write.table(Body, file=<span style=
              "color: #b8bb26;">"/home/david/Documents/BenoitLab/RNA-seq/Deseq-Genes/Body.csv"</span>, quote=<span style="color: #fabd2f;">FALSE</span>, col.names=<span style="color: #fabd2f;">TRUE</span>, row.names=<span style="color: #fabd2f;">TRUE</span>, sep=<span style="color: #b8bb26;">","</span>)
write.table(Leg, file=<span style=
"color: #b8bb26;">"/home/david/Documents/BenoitLab/RNA-seq/Deseq-Genes/Leg.csv"</span>, quote=<span style="color: #fabd2f;">FALSE</span>, col.names=<span style="color: #fabd2f;">TRUE</span>, row.names=<span style="color: #fabd2f;">TRUE</span>, sep=<span style="color: #b8bb26;">","</span>)
write.table(res, file=<span style=
"color: #b8bb26;">"/home/david/Documents/BenoitLab/RNA-seq/Deseq-Genes/total.csv"</span>, quote=<span style="color: #fabd2f;">FALSE</span>, col.names=<span style="color: #fabd2f;">TRUE</span>, row.names=<span style="color: #fabd2f;">TRUE</span>, sep=<span style="color: #b8bb26;">","</span>)
</pre>
            </div>
          </div>
        </div>
        <div id="outline-container-org8c39636" class="outline-4">
          <h4 id="org8c39636"><span class=
          "section-number-4">4.1.7</span> Create a new analysis
          design</h4>
          <div class="outline-text-4" id="text-4-1-7">
            <div class="org-src-container">
              <pre class="src src-jupyter-R">newdd <span style=
              "color: #d3869b;">&lt;-</span> dds
design(newdd) <span style=
"color: #d3869b;">&lt;-</span> formula(~part + part:pesticides)
newdd <span style="color: #d3869b;">&lt;-</span> DESeq(newdd)
newres <span style="color: #d3869b;">&lt;-</span> results(newdd)
</pre>
            </div>
          </div>
        </div>
        <div id="outline-container-org09da6b5" class="outline-4">
          <h4 id="org09da6b5"><span class=
          "section-number-4">4.1.8</span> Prepare the data for
          pheatmap plotting</h4>
          <div class="outline-text-4" id="text-4-1-8">
            <div class="org-src-container">
              <pre class="src src-jupyter-R">select <span style=
              "color: #d3869b;">&lt;-</span> order(rowMeans(counts(dds, normalized=<span style="color: #fabd2f;">FALSE</span>)), decreasing = <span style="color: #fabd2f;">TRUE</span>)[<span style="color: #d3869b; font-weight: bold;">1</span>:<span style="color: #d3869b; font-weight: bold;">20</span>]
df <span style=
"color: #d3869b;">&lt;-</span> as.data.frame(colData(dds)[,c(<span style="color: #b8bb26;">"pesticides"</span>,<span style="color: #b8bb26;">"part"</span>)])
</pre>
            </div>
          </div>
        </div>
        <div id="outline-container-orgc1edbbe" class="outline-4">
          <h4 id="orgc1edbbe"><span class=
          "section-number-4">4.1.9</span> Integrity plots</h4>
          <div class="outline-text-4" id="text-4-1-9">
            <div class="org-src-container">
              <pre class="src src-jupyter-R">jpeg(<span style=
              "color: #b8bb26;">"generalfigures/plotMA_deseq.jpg"</span>)
plotMA(results)
dev.off()

jpeg(<span style=
"color: #b8bb26;">"generalfigures/plotcounts_deseq.jpg"</span>)
plotCounts(data, gene=which.min(results$padj), intgroup = <span style="color: #b8bb26;">"part"</span>)
dev.off()

jpeg(<span style=
"color: #b8bb26;">"generalfigures/meanSdvsd_deseq.jpg"</span>)
vsn::meanSdPlot(assay(vsd))
dev.off()

jpeg(<span style=
"color: #b8bb26;">"generalfigures/meadnSdrld_deseq.jpg"</span>)
vsn::meanSdPlot(assay(rld))
dev.off()

jpeg(<span style=
"color: #b8bb26;">"generalfigures/meanSdntd_deseq.jpg"</span>)
vsn::meanSdPlot(assay(ntd))
dev.off()

jpeg(<span style=
"color: #b8bb26;">"generalfigures/pcaplot_deseq.jpg"</span>, width = <span style="color: #d3869b; font-weight: bold;">800</span>, height=<span style="color: #d3869b; font-weight: bold;">800</span>)
plotPCA(vsd, intgroup=c(<span style=
"color: #b8bb26;">"part"</span>, <span style=
"color: #b8bb26;">"pesticides"</span>))
dev.off()

jpeg(<span style=
"color: #b8bb26;">"generalfigures/pheatmapplotntd.jpg"</span>, width=<span style="color: #d3869b; font-weight: bold;">800</span>, height=<span style="color: #d3869b; font-weight: bold;">800</span>)
pheatmap(assay(ntd)[select,], cluster_rows=<span style=
"color: #fabd2f;">FALSE</span>, show_rownames=<span style=
"color: #fabd2f;">TRUE</span>, cluster_cols=<span style=
"color: #fabd2f;">TRUE</span>, annotation_col=df)
dev.off()

jpeg(<span style=
"color: #b8bb26;">"generalfigures/pheatmapplotvsd.jpg"</span>, width=<span style="color: #d3869b; font-weight: bold;">800</span>, height=<span style="color: #d3869b; font-weight: bold;">800</span>)
pheatmap(assay(vsd)[select,], cluster_rows=<span style=
"color: #fabd2f;">FALSE</span>, show_rownames=<span style=
"color: #fabd2f;">TRUE</span>, cluster_cols=<span style=
"color: #fabd2f;">FALSE</span>, annotation_col=df)
dev.off()

jpeg(<span style=
"color: #b8bb26;">"generalfigures/pheatmapplotrld.jpg"</span>, width=<span style="color: #d3869b; font-weight: bold;">800</span>, height=<span style="color: #d3869b; font-weight: bold;">800</span>)
pheatmap(assay(rld)[select,], cluster_rows=<span style=
"color: #fabd2f;">FALSE</span>, show_rownames=<span style=
"color: #fabd2f;">TRUE</span>, cluster_cols=<span style=
"color: #fabd2f;">FALSE</span>, annotation_col=df)
dev.off()
</pre>
            </div>
          </div>
        </div>
        <div id="outline-container-orge9a0f5b" class="outline-4">
          <h4 id="orge9a0f5b"><span class=
          "section-number-4">4.1.10</span> Another Deseq
          Design</h4>
          <div class="outline-text-4" id="text-4-1-10">
            <div class="org-src-container">
              <pre class="src src-jupyter-R">
<span style="color: #d3869b;">library</span>(tidyverse)
vstarray <span style="color: #d3869b;">&lt;-</span> assay(vsd)

range <span style="color: #d3869b;">&lt;-</span> vstarray[]
names <span style="color: #d3869b;">&lt;-</span> rownames(range)
maxs <span style="color: #d3869b;">&lt;-</span> rowMaxs(range)
mins <span style="color: #d3869b;">&lt;-</span> rowMins(range)

maxs <span style="color: #d3869b;">&lt;-</span> maxs-mins
pt <span style="color: #d3869b;">&lt;-</span> tibble(names, maxs)
pt
pt <span style="color: #d3869b;">&lt;-</span> pt <span style=
"color: #d3869b;">%&gt;%</span> filter(maxs &gt; <span style=
"color: #d3869b; font-weight: bold;">10</span>)
pt
ggplot(pt, aes(x=names, y= maxs)) + geom_col()
descending <span style=
"color: #d3869b;">&lt;-</span> arrange(pt, desc(maxs))
descending <span style=
"color: #d3869b;">&lt;-</span> factor(descending[<span style=
"color: #b8bb26;">"maxs"</span>])
descending <span style=
"color: #d3869b;">&lt;-</span> tibble(row = seq(length(descending)), descending)
ggplot(descending, aes(x=row, y= descending)) + geom_col()

sum(results$padj &lt; <span style=
"color: #d3869b; font-weight: bold;">0.05</span>, na.rm=T)
res <span style="color: #d3869b;">&lt;-</span> results
</pre>
            </div>
          </div>
        </div>
        <div id="outline-container-org8e695ba" class="outline-4">
          <h4 id="org8e695ba"><span class=
          "section-number-4">4.1.11</span> Altered Selection of
          genes</h4>
          <div class="outline-text-4" id="text-4-1-11">
            <div class="org-src-container">
              <pre class="src src-jupyter-R">
Upregulated <span style=
"color: #d3869b;">&lt;-</span> res[ which(res$padj &lt; <span style="color: #d3869b; font-weight: bold;">0.03</span> &amp; res$log2FoldChange &gt; <span style="color: #d3869b; font-weight: bold;">0</span>), ]
DownRegulated <span style=
"color: #d3869b;">&lt;-</span> res[ which(res$padj &lt; <span style="color: #d3869b; font-weight: bold;">0.03</span> &amp; res$log2FoldChange &lt; <span style="color: #d3869b; font-weight: bold;">0</span>), ]
DownRegulated[order(DownRegulated$padj), ]
head(DownRegulated$padj)
D <span style=
"color: #d3869b;">&lt;-</span> rownames(DownRegulated)
Up <span style="color: #d3869b;">&lt;-</span> rownames(Upregulated)
DR <span style=
"color: #d3869b;">&lt;-</span> res[ which(res$padj &lt; <span style="color: #d3869b; font-weight: bold;">0.03</span>), ]
DR[order(DR$padj),]
</pre>
            </div>
          </div>
        </div>
        <div id="outline-container-orgdc3f0ca" class="outline-4">
          <h4 id="orgdc3f0ca"><span class=
          "section-number-4">4.1.12</span> Save new selection to
          file</h4>
          <div class="outline-text-4" id="text-4-1-12">
            <div class="org-src-container">
              <pre class=
              "src src-jupyter-R">write.table(D, file=<span style=
              "color: #b8bb26;">"Genes/DownRegulated.txt"</span>, quote=<span style="color: #fabd2f;">FALSE</span>, col.names=<span style="color: #fabd2f;">FALSE</span>, row.names=<span style="color: #fabd2f;">FALSE</span>, sep=<span style="color: #b8bb26;">","</span>)
write.table(Up, file=<span style=
"color: #b8bb26;">"Genes/UpRegulated.txt"</span>, quote=<span style="color: #fabd2f;">FALSE</span>, col.names=<span style="color: #fabd2f;">FALSE</span>, row.names=<span style="color: #fabd2f;">FALSE</span>, sep=<span style="color: #b8bb26;">","</span>)
</pre>
            </div>
          </div>
        </div>
        <div id="outline-container-org14576d6" class="outline-4">
          <h4 id="org14576d6"><span class=
          "section-number-4">4.1.13</span> Extra integrity
          plot</h4>
          <div class="outline-text-4" id="text-4-1-13">
            <div class="org-src-container">
              <pre class="src src-jupyter-R">d <span style=
              "color: #d3869b;">&lt;-</span> mcols(Deseq,use.names=<span style=
              "color: #fabd2f;">TRUE</span>)
a <span style=
"color: #d3869b;">&lt;-</span> data.frame(rownumb = seq(nrow(d)), d)
a <span style=
"color: #d3869b;">&lt;-</span> data.frame(rownumb = seq(nrow(d)), basemean = d[,<span style="color: #d3869b; font-weight: bold;">1</span>])
<span style="color: #d3869b;">library</span>(ggplot2)
ggplot(a, aes(x=rownumb, y=dispGeneEst)) + geom_point()
</pre>
            </div>
          </div>
        </div>
        <div id="outline-container-org2d5aa49" class="outline-4">
          <h4 id="org2d5aa49"><span class=
          "section-number-4">4.1.14</span> <span class=
          "todo TODO">TODO</span> Extra EdgeR for venn
          diagrams</h4>
          <div class="outline-text-4" id="text-4-1-14">
            <p>This is mostly contained in the EdgeR section.</p>
            <div class="org-src-container">
              <pre class="src src-jupyter-R">
<span style="color: #928374;">#</span><span style=
"color: #928374;">! EdgeR section ----</span>
<span style="color: #d3869b;">library</span>(edgeR)
<span style="color: #928374;"># </span><span style=
"color: #928374;">EdgR dataset from tximport</span>
counts <span style="color: #d3869b;">&lt;-</span> txi$counts
length <span style="color: #d3869b;">&lt;-</span> txi$length
normMat <span style=
"color: #d3869b;">&lt;-</span> length / exp(rowMeans(log(length)))
normCts <span style="color: #d3869b;">&lt;-</span> counts / normMat
eff.lib <span style=
"color: #d3869b;">&lt;-</span> calcNormFactors(normCts) * colSums(normCts)
normMat <span style=
"color: #d3869b;">&lt;-</span> sweep(normMat, <span style=
"color: #d3869b; font-weight: bold;">2</span>, eff.lib, <span style="color: #b8bb26;">"*"</span>)
normMat <span style="color: #d3869b;">&lt;-</span> log(normMat)
y <span style="color: #d3869b;">&lt;-</span> DGEList(counts)
y <span style=
"color: #d3869b;">&lt;-</span> scaleOffset(y, normMat)



<span style="color: #928374;"># </span><span style=
"color: #928374;">EdgR heatmap</span>
logcpm <span style="color: #d3869b;">&lt;-</span> cpm(EdgeR)
meta <span style=
"color: #d3869b;">&lt;-</span> read.csv(<span style=
"color: #b8bb26;">"metadata.csv"</span>, sep= <span style=
"color: #b8bb26;">","</span>)
pest <span style=
"color: #d3869b;">&lt;-</span> factor(meta$pesticides, levels=c(<span style="color: #b8bb26;">"perm"</span>, <span style="color: #b8bb26;">"deet"</span>, <span style="color: #b8bb26;">"control"</span>))
part <span style=
"color: #d3869b;">&lt;-</span> factor(meta$part, levels=c(<span style="color: #b8bb26;">"leg"</span>, <span style="color: #b8bb26;">"body"</span>))
design <span style=
"color: #d3869b;">&lt;-</span> model.matrix(~<span style=
"color: #d3869b; font-weight: bold;">0</span> + pest + part)
design
keep <span style=
"color: #d3869b;">&lt;-</span> filterByExpr(y, design)
EdgeR <span style="color: #d3869b;">&lt;-</span> y[keep, ]
estimateDisp(EdgeR)
y <span style=
"color: #d3869b;">&lt;-</span> estimateGLMCommonDisp(EdgeR, design)
y <span style=
"color: #d3869b;">&lt;-</span> estimateGLMTrendedDisp(y, design)
y <span style=
"color: #d3869b;">&lt;-</span> estimateGLMTagwiseDisp(y, design)
est <span style=
"color: #d3869b;">&lt;-</span> estimateDisp(EdgeR, design, robust=<span style="color: #fabd2f;">TRUE</span>)
design
fit <span style="color: #d3869b;">&lt;-</span> glmQLFit(y, design)
lrt <span style=
"color: #d3869b;">&lt;-</span> glmLRT(fit, coef=<span style=
"color: #d3869b; font-weight: bold;">1</span>:<span style=
"color: #d3869b; font-weight: bold;">4</span>)
test <span style=
"color: #d3869b;">&lt;-</span> glmQLFTest(fit, coef=<span style=
"color: #d3869b; font-weight: bold;">1</span>:<span style=
"color: #d3869b; font-weight: bold;">4</span>)
lrc$PValue
lrt

test$logFC.pestperm - test$logFC.pestdeet

ggplot(test$table, aes(y=logFC.partleg, x=seq(nrow(test)))) + geom_hex()
<span style="color: #928374;">#</span><span style=
"color: #928374;">additional analysis----</span>
table <span style="color: #d3869b;">&lt;-</span> test$table
body <span style=
"color: #d3869b;">&lt;-</span> table[table$logFC.partbody &gt; <span style="color: #d3869b; font-weight: bold;">2</span>, ]
leg <span style=
"color: #d3869b;">&lt;-</span> table[table$logFC.partbody &lt; -<span style="color: #d3869b; font-weight: bold;">2</span>, ]
table <span style="color: #d3869b;">&lt;-</span> body
diff <span style=
"color: #d3869b;">&lt;-</span> table$logFC.pestcontrol - table$logFC.pestdeet
<span style="color: #d3869b;">library</span>(tidyverse)
data <span style=
"color: #d3869b;">&lt;-</span> tibble(data = diff, rows = seq(length(diff)))
ggplot(data, aes(x=rows, y=data)) + geom_point()
table$diff <span style="color: #d3869b;">&lt;-</span> diff
cont <span style=
"color: #d3869b;">&lt;-</span> table[table$diff &gt; <span style=
"color: #d3869b; font-weight: bold;">2</span>, ]
deet <span style=
"color: #d3869b;">&lt;-</span> table[table$diff &lt; -<span style=
"color: #d3869b; font-weight: bold;">2</span>, ]
diff2 <span style=
"color: #d3869b;">&lt;-</span> table$logFC.pestcontrol - table$logFC.pestperm
table$diff2 <span style="color: #d3869b;">&lt;-</span> diff2
cont2 <span style=
"color: #d3869b;">&lt;-</span> table[table$diff2 &gt; <span style=
"color: #d3869b; font-weight: bold;">2</span>, ]
perm <span style=
"color: #d3869b;">&lt;-</span> table[table$diff2 &lt; -<span style=
"color: #d3869b; font-weight: bold;">2</span>, ]



<span style="color: #d3869b;">library</span>(VennDiagram)
futile.logger::flog.threshold(futile.logger::ERROR, name = <span style="color: #b8bb26;">"VennDiagramLogger"</span>)
Legcont1 <span style="color: #d3869b;">&lt;-</span> cont
Legcont2 <span style="color: #d3869b;">&lt;-</span> cont2
deet1 <span style="color: #d3869b;">&lt;-</span> deet
perm1 <span style="color: #d3869b;">&lt;-</span> perm
d <span style=
"color: #d3869b;">&lt;-</span> data.frame(Pest = c(rep(c(<span style="color: #b8bb26;">"DEET"</span>, <span style="color: #b8bb26;">"DEET"</span>, <span style="color: #b8bb26;">"PERM"</span>, <span style="color: #b8bb26;">"PERM"</span>))),
                Part = c(rep(<span style=
"color: #b8bb26;">"Body"</span>, <span style=
"color: #d3869b; font-weight: bold;">4</span>), rep(<span style=
"color: #b8bb26;">"Leg"</span>, <span style=
"color: #d3869b; font-weight: bold;">4</span>)),
                Names = c(rep(c(<span style=
"color: #b8bb26;">"Up Regulated"</span>, <span style=
"color: #b8bb26;">"Down Regulated"</span>), <span style=
"color: #d3869b; font-weight: bold;">4</span>)),
                GeneCount = c(nrow(deet), nrow(cont), nrow(perm), nrow(cont2), nrow(deet1), nrow(Legcont1), nrow(perm1), nrow(Legcont2)))

write.table(d, file=<span style=
"color: #b8bb26;">"Genes/regulation.csv"</span>, quote=<span style=
"color: #fabd2f;">FALSE</span>, col.names=<span style=
"color: #fabd2f;">TRUE</span>, row.names=<span style=
"color: #fabd2f;">FALSE</span>, sep=<span style=
"color: #b8bb26;">","</span>)
write.table(rownames(perm), file=<span style=
"color: #b8bb26;">"Genes/perm.csv"</span>, quote=<span style=
"color: #fabd2f;">FALSE</span>, col.names=<span style=
"color: #fabd2f;">FALSE</span>, row.names=<span style=
"color: #fabd2f;">FALSE</span>, sep=<span style=
"color: #b8bb26;">","</span>)

pl ggplot(d, aes(x=Names, y = GeneCount, fill=Names)) + geom_col()
venn.diagram(
  x = list(rownames(deet), rownames(cont)),
  category.names = c(<span style=
"color: #b8bb26;">"DEET"</span> , <span style=
"color: #b8bb26;">"Control"</span>),
  filename = <span style=
"color: #b8bb26;">"Deetvcontorlvleg.png"</span>,
  compression = <span style="color: #b8bb26;">"lzw"</span>,


  lwd = <span style="color: #d3869b; font-weight: bold;">2</span>,
  lty = <span style="color: #b8bb26;">'blank'</span>,
  fill = c(<span style="color: #b8bb26;">"red"</span>, <span style=
"color: #b8bb26;">"blue"</span>),
)
grid.draw(tmp)

write.table(rownames(perm), file=<span style=
"color: #b8bb26;">"Genes/perm.csv"</span>, quote=<span style=
"color: #fabd2f;">FALSE</span>, col.names=<span style=
"color: #fabd2f;">FALSE</span>, row.names=<span style=
"color: #fabd2f;">FALSE</span>, sep=<span style=
"color: #b8bb26;">","</span>)
write.table(rownames(deet), file=<span style=
"color: #b8bb26;">"Genes/deet.csv"</span>, quote=<span style=
"color: #fabd2f;">FALSE</span>, col.names=<span style=
"color: #fabd2f;">FALSE</span>, row.names=<span style=
"color: #fabd2f;">FALSE</span>, sep=<span style=
"color: #b8bb26;">","</span>)
write.table(rownames(cont), file=<span style=
"color: #b8bb26;">"Genes/deetcont.csv"</span>, quote=<span style=
"color: #fabd2f;">FALSE</span>, col.names=<span style=
"color: #fabd2f;">FALSE</span>, row.names=<span style=
"color: #fabd2f;">FALSE</span>, sep=<span style=
"color: #b8bb26;">","</span>)
write.table(rownames(cont2), file=<span style=
"color: #b8bb26;">"Genes/permcont.csv"</span>, quote=<span style=
"color: #fabd2f;">FALSE</span>, col.names=<span style=
"color: #fabd2f;">FALSE</span>, row.names=<span style=
"color: #fabd2f;">FALSE</span>, sep=<span style=
"color: #b8bb26;">","</span>)

ggplot(contdata, aes(x=rows, y=data)) + geom_point()

rownames(cont)
<span style="color: #928374;"># </span><span style=
"color: #928374;">EdgeR plots ----</span>
jpeg(<span style=
"color: #b8bb26;">"generalfigures/BCVplot_edgeR.jpg"</span>)
plotBCV(est)
dev.off()

jpeg(<span style=
"color: #b8bb26;">"generalfigures/QLDISP_edgeR.jpg"</span>)
plotQLDisp(fit)
dev.off()

jpeg(<span style=
"color: #b8bb26;">"generalfigures/MDS_edgeR.jpg"</span>)
plotMDS(logcpm)
dev.off()


<span style="color: #928374;"># </span><span style=
"color: #928374;">jpeg("generalfigur&gt;es/SpliceDGE_edgeR.jpg")</span>
<span style="color: #928374;"># </span><span style=
"color: #928374;">plotSpliceDGE(logcpm)</span>
<span style="color: #928374;"># </span><span style=
"color: #928374;">dev.off()</span>

<span style="color: #928374;"># </span><span style=
"color: #928374;">end ----</span>
<span style="color: #d3869b;">library</span>(matrixStats)
stats <span style="color: #d3869b;">&lt;-</span> txi$abundance
tpm <span style=
"color: #d3869b;">&lt;-</span> transform(stats, sd=rowSds(stats), avg=rowMeans(stats), median = rowMedians(stats))

dge <span style="color: #d3869b;">&lt;-</span> DGEList(txi$counts)
cpm <span style=
"color: #d3869b;">&lt;-</span> cpm(dge, log=<span style=
"color: #fabd2f;">TRUE</span>)
head(cpm)
data.frame(cpm)
cpmtibble <span style=
"color: #d3869b;">&lt;-</span> as_tibble(cpm, rownames=<span style=
"color: #b8bb26;">"geneID"</span>)
cpmpivot <span style=
"color: #d3869b;">&lt;-</span> pivot_longer(cpmtibble, cols=seq(<span style="color: #d3869b; font-weight: bold;">2</span>, ncol(cpmtibble)), names_to = <span style="color: #b8bb26;">"samples"</span>, values_to = <span style="color: #b8bb26;">"expression"</span>)



ggplot(tpm) +
  aes(x = sd, y = median) +
  geom_hex()

ggplot(tpm) +
  aes(x = sd, y = avg) +
  geom_point(shape=<span style=
"color: #d3869b; font-weight: bold;">16</span>, size=<span style=
"color: #d3869b; font-weight: bold;">2</span>) +
  geom_smooth(method=lm) +
  geom_hex() +
  labs(y=<span style=
"color: #b8bb26;">"Median"</span>, x = <span style=
"color: #b8bb26;">"Standard deviation"</span>, title=<span style=
"color: #b8bb26;">"Transcripts per million (TPM)"</span>, subtitle=<span style="color: #b8bb26;">"unfiltered, non-normalized data"</span>) +
  theme_classic() +
  theme_dark() +
  theme_bw()

ggplot(cpmpivot) +
  aes(x=samples, y=expression, fill=samples) +
  geom_violin(trim = <span style=
"color: #fabd2f;">FALSE</span>, show.legend = <span style=
"color: #fabd2f;">FALSE</span>) +
  stat_summary(fun = <span style=
"color: #b8bb26;">"median"</span>,geom = <span style=
"color: #b8bb26;">"point"</span>,shape = <span style=
"color: #d3869b; font-weight: bold;">95</span>,size = <span style=
"color: #d3869b; font-weight: bold;">10</span>,color = <span style=
"color: #b8bb26;">"black"</span>,show.legend = <span style=
"color: #fabd2f;">FALSE</span>) +
  labs(y=<span style=
"color: #b8bb26;">"log2 expression"</span>, x = <span style=
"color: #b8bb26;">"sample"</span>,title=<span style=
"color: #b8bb26;">"Log2 Counts per Million (CPM)"</span>,subtitle=<span style="color: #b8bb26;">"unfiltered, non-normalized"</span>,caption=paste0(<span style="color: #b8bb26;">"produced on "</span>, Sys.time())) +
  theme_bw()

keepers <span style=
"color: #d3869b;">&lt;-</span> rowSums(cpm &gt; <span style=
"color: #d3869b; font-weight: bold;">1</span>)&gt;=<span style=
"color: #d3869b; font-weight: bold;">3</span>
table(rowSums(dge$counts==<span style=
"color: #d3869b; font-weight: bold;">0</span>)==<span style=
"color: #d3869b; font-weight: bold;">10</span>)


<span style="color: #928374;"># </span><span style=
"color: #928374;">venndiagrams ----</span>

<span style="color: #d3869b;">library</span>(VennDiagram)


bodygenes <span style="color: #d3869b;">&lt;-</span> names(Deseq)
bodygenes <span style=
"color: #d3869b;">&lt;-</span> bodygenes[!bodygenes <span style=
"color: #d3869b;">%in%</span> rownames(Leg)]
leggenes <span style="color: #d3869b;">&lt;-</span> names(Deseq)
leggenes <span style=
"color: #d3869b;">&lt;-</span> leggenes[!leggenes <span style=
"color: #d3869b;">%in%</span> rownames(Body)]
names(Deseq) - rownames(Body)

<span style="color: #928374;">#</span>
bodyup <span style=
"color: #d3869b;">&lt;-</span> table[table[<span style=
"color: #b8bb26;">"partleg"</span>] &gt; <span style=
"color: #d3869b; font-weight: bold;">0</span>]


<span style="color: #928374;"># </span><span style=
"color: #928374;">Venn Diagram 1----</span>
futile.logger::flog.threshold(futile.logger::ERROR, name = <span style="color: #b8bb26;">"VennDiagramLogger"</span>)
venn.diagram(
  x = list(bodygenes, leggenes),
  category.names = c(<span style=
"color: #b8bb26;">"Body"</span> , <span style=
"color: #b8bb26;">"Leg"</span>),

  filename = <span style=
"color: #b8bb26;">'14_venn_diagramm.png'</span>,
  output=<span style="color: #fabd2f;">FALSE</span>,
  imagetype=<span style="color: #b8bb26;">"png"</span>,
  compression=<span style="color: #b8bb26;">"lzw"</span>,


  lwd = <span style="color: #d3869b; font-weight: bold;">2</span>,
  lty = <span style="color: #b8bb26;">'blank'</span>,
  fill = c(<span style="color: #b8bb26;">"red"</span>, <span style=
"color: #b8bb26;">"blue"</span>),
)


</pre>
            </div>
          </div>
        </div>
      </div>
      <div id="outline-container-org7f122a2" class="outline-3">
        <h3 id="org7f122a2"><span class=
        "section-number-3">4.2</span> EdgeR</h3>
        <div class="outline-text-3" id="text-4-2"></div>
        <div id="outline-container-org3aa73ed" class="outline-4">
          <h4 id="org3aa73ed"><span class=
          "section-number-4">4.2.1</span> Initial settings</h4>
          <div class="outline-text-4" id="text-4-2-1">
            <p>Directory and function setup, configure this when
            running the script. This is an attempt at simplifying
            the analysis process in R to a point that is hopefully
            nonspecific to deseq2 or edgeR. Enter the directory
            where the RNA-seq data is located, the folder
            containing the quant data, and the type of quant tool
            used (fishtype). The script should take care of the
            rest.</p>
            <div class="org-src-container">
              <pre class="src src-jupyter-R"><span style=
              "color: #d3869b;">setwd</span>(<span style=
              "color: #b8bb26;">"/home/david/Documents/BenoitLab/RNA-seq/"</span>)
<span style="color: #d3869b;">source</span>(<span style=
"color: #b8bb26;">"scripts/functions.r"</span>)
directory <span style="color: #d3869b;">&lt;-</span> <span style=
"color: #b8bb26;">"/home/david/Documents/BenoitLab/R/largeRNASEQ/"</span>
quantdatafoldername <span style=
"color: #d3869b;">&lt;-</span> <span style=
"color: #b8bb26;">"CorrectedQuantData"</span>
fishtype <span style="color: #d3869b;">&lt;-</span> <span style=
"color: #b8bb26;">"sailfish"</span>
</pre>
            </div>
          </div>
        </div>
        <div id="outline-container-org5a72e06" class="outline-4">
          <h4 id="org5a72e06"><span class=
          "section-number-4">4.2.2</span> Do the initial processing
          of the data</h4>
          <div class="outline-text-4" id="text-4-2-2">
            <p>Imports the data and runs it through initial edgeR
            analysis using the functions created above. The
            functions do the initial manipulation of the data as
            detailed in the edgeR paper. This outputs the objects
            we will trim based on expression.</p>
            <div class="org-src-container">
              <pre class="src src-jupyter-R">a <span style=
              "color: #d3869b;">&lt;-</span> txi(directory, quantdatafoldername, <span style="color: #b8bb26;">"metadata.csv"</span>, fishtype)
<span style="color: #928374;">#</span><span style=
"color: #928374;">De &lt;- initialDeseq(a[[1]], a[[2]])</span>
Ed <span style=
"color: #d3869b;">&lt;-</span> initialEdgeR(a[[<span style=
"color: #d3869b; font-weight: bold;">1</span>]], a[[<span style=
"color: #d3869b; font-weight: bold;">2</span>]])
Eest <span style=
"color: #d3869b;">&lt;-</span> estEdgeR(Ed[[<span style=
"color: #d3869b; font-weight: bold;">1</span>]], Ed[[<span style=
"color: #d3869b; font-weight: bold;">2</span>]])
lrt <span style=
"color: #d3869b;">&lt;-</span> glmLRT(Eest, coef=<span style=
"color: #d3869b; font-weight: bold;">1</span>:<span style=
"color: #d3869b; font-weight: bold;">4</span>)
test <span style=
"color: #d3869b;">&lt;-</span> glmQLFTest(Eest, coef=<span style=
"color: #d3869b; font-weight: bold;">1</span>:<span style=
"color: #d3869b; font-weight: bold;">4</span>)
</pre>
            </div>
            <pre class="example" id="orgfd06f95">
reading in files with read_tsv

1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17


summarizing abundance

summarizing counts

summarizing length

Loading required package: limma
</pre>
          </div>
        </div>
        <div id="outline-container-org4b6aaee" class="outline-4">
          <h4 id="org4b6aaee"><span class=
          "section-number-4">4.2.3</span> Partion the data into
          body and leg tables</h4>
          <div class="outline-text-4" id="text-4-2-3">
            <p>The Table object from the test object is separated
            into a body and a leg section, where the genes in the
            leg section are down regulated in the body and vice
            versa.</p>
            <div class="org-src-container">
              <pre class="src src-jupyter-R">table <span style=
              "color: #d3869b;">&lt;-</span> test$table
pSelectedTable <span style=
"color: #d3869b;">&lt;-</span> table[table$PValue &lt; <span style=
"color: #d3869b; font-weight: bold;">0.01</span>, ]
body <span style=
"color: #d3869b;">&lt;-</span> pSelectedTable[pSelectedTable$logFC.partbody &gt; <span style="color: #d3869b; font-weight: bold;">2</span>, ]
leg <span style=
"color: #d3869b;">&lt;-</span> pSelectedTable[pSelectedTable$logFC.partbody &lt; -<span style="color: #d3869b; font-weight: bold;">2</span>, ]
head(table)
</pre>
            </div>
          </div>
        </div>
        <div id="outline-container-org0389aee" class="outline-4">
          <h4 id="org0389aee"><span class=
          "section-number-4">4.2.4</span> Save the data tables to a
          file</h4>
          <div class="outline-text-4" id="text-4-2-4">
            <p>Configure this to fit your computer. This section
            saves csv files with the expression data from the
            tables in the previous section.</p>
            <div class="org-src-container">
              <pre class="src src-jupyter-R"><span style=
              "color: #d3869b;">setwd</span>(<span style=
              "color: #b8bb26;">"/home/david/Documents/BenoitLab/RNA-seq/"</span>)
write.table(table, file=<span style=
"color: #b8bb26;">"edgeR-Genes/TotalExpr.csv"</span>, quote=<span style="color: #fabd2f;">FALSE</span>, col.names=<span style="color: #fabd2f;">TRUE</span>, row.names=<span style="color: #fabd2f;">TRUE</span>, sep=<span style="color: #b8bb26;">","</span>)
write.table(leg, file=<span style=
"color: #b8bb26;">"edgeR-Genes/legExpr.csv"</span>, quote=<span style="color: #fabd2f;">FALSE</span>, col.names=<span style="color: #fabd2f;">TRUE</span>, row.names=<span style="color: #fabd2f;">TRUE</span>, sep=<span style="color: #b8bb26;">","</span>)
write.table(body, file=<span style=
"color: #b8bb26;">"edgeR-Genes/bodyExpr.csv"</span>, quote=<span style="color: #fabd2f;">FALSE</span>, col.names=<span style="color: #fabd2f;">TRUE</span>, row.names=<span style="color: #fabd2f;">TRUE</span>, sep=<span style="color: #b8bb26;">","</span>)
</pre>
            </div>
          </div>
        </div>
        <div id="outline-container-orgeb5ef68" class="outline-4">
          <h4 id="orgeb5ef68"><span class=
          "section-number-4">4.2.5</span> Find the relative
          expression values when compared to the control</h4>
          <div class="outline-text-4" id="text-4-2-5">
            <p>This is the main comparison with the pesticides done
            with the edgeR table. The control is subtracted from
            the pesticide expression to give a relative expression
            value. This is not necessary in Deseq and is instead
            done by manipulating the design.</p>
            <div class="org-src-container">
              <pre class=
              "src src-jupyter-R">deetvcontrol <span style=
              "color: #d3869b;">&lt;-</span> table$logFC.pestdeet - table$logFC.pestcontrol
permvcontrol <span style=
"color: #d3869b;">&lt;-</span> table$logFC.pestperm - table$logFC.pestcontrol
deetvcontrolvbody <span style=
"color: #d3869b;">&lt;-</span> body$logFC.pestdeet - body$logFC.pestcontrol
permvcontrolvbody <span style=
"color: #d3869b;">&lt;-</span> body$logFC.pestperm - body$logFC.pestcontrol
deetvcontrolvleg <span style=
"color: #d3869b;">&lt;-</span> leg$logFC.pestdeet - leg$logFC.pestcontrol
permvcontrolvleg <span style=
"color: #d3869b;">&lt;-</span> leg$logFC.pestperm - leg$logFC.pestcontrol

table$deetvcontrol <span style=
"color: #d3869b;">&lt;-</span> deetvcontrol
table$permvcontrol <span style=
"color: #d3869b;">&lt;-</span> permvcontrol
body$deetvcontrol <span style=
"color: #d3869b;">&lt;-</span> deetvcontrolvbody
body$permvcontrol <span style=
"color: #d3869b;">&lt;-</span> permvcontrolvbody
leg$deetvcontrol <span style=
"color: #d3869b;">&lt;-</span> deetvcontrolvleg
leg$permvcontrol <span style=
"color: #d3869b;">&lt;-</span> permvcontrolvleg
</pre>
            </div>
          </div>
        </div>
        <div id="outline-container-orgcf6e500" class="outline-4">
          <h4 id="orgcf6e500"><span class=
          "section-number-4">4.2.6</span> Selection of the
          data</h4>
          <div class="outline-text-4" id="text-4-2-6">
            <p>Selects only the columns that have an expression
            value greater than 2(up-regulated) or less than
            -2(down-regulated). This is the same selection done
            earlier in the script with the body and leg.</p>
            <div class="org-src-container">
              <pre class="src src-jupyter-R">updeet <span style=
              "color: #d3869b;">&lt;-</span> table[table$deetvcontrol &gt; <span style="color: #d3869b; font-weight: bold;">2</span>,]
downdeet <span style=
"color: #d3869b;">&lt;-</span> table[table$deetvcontrol &lt; -<span style="color: #d3869b; font-weight: bold;">2</span>,]



updeetbody <span style=
"color: #d3869b;">&lt;-</span> body[body$deetvcontrol &gt; <span style="color: #d3869b; font-weight: bold;">2</span>,]
downdeetbody <span style=
"color: #d3869b;">&lt;-</span> body[body$deetvcontrol &lt; -<span style="color: #d3869b; font-weight: bold;">2</span>,]
updeetleg <span style=
"color: #d3869b;">&lt;-</span> leg[leg$deetvcontrol &gt; <span style="color: #d3869b; font-weight: bold;">2</span>,]
downdeetleg <span style=
"color: #d3869b;">&lt;-</span> leg[leg$deetvcontrol &lt; -<span style="color: #d3869b; font-weight: bold;">2</span>,]


upperm <span style=
"color: #d3869b;">&lt;-</span> table[table$permvcontrol &gt; <span style="color: #d3869b; font-weight: bold;">2</span>,]
downperm <span style=
"color: #d3869b;">&lt;-</span> table[table$permvcontrol &lt; -<span style="color: #d3869b; font-weight: bold;">2</span>,]
uppermbody <span style=
"color: #d3869b;">&lt;-</span> body[body$permvcontrol &gt; <span style="color: #d3869b; font-weight: bold;">2</span>,]
downpermbody <span style=
"color: #d3869b;">&lt;-</span> body[body$permvcontrol &lt; -<span style="color: #d3869b; font-weight: bold;">2</span>,]
uppermleg <span style=
"color: #d3869b;">&lt;-</span> leg[leg$permvcontrol &gt; <span style="color: #d3869b; font-weight: bold;">2</span>,]
downpermleg <span style=
"color: #d3869b;">&lt;-</span> leg[leg$permvcontrol &lt; -<span style="color: #d3869b; font-weight: bold;">2</span>,]
</pre>
            </div>
          </div>
        </div>
        <div id="outline-container-org1745de5" class="outline-4">
          <h4 id="org1745de5"><span class=
          "section-number-4">4.2.7</span> Create a csv for bar
          plotting</h4>
          <div class="outline-text-4" id="text-4-2-7">
            <p>This is just a simple organization of the number of
            genes deferentially expressed for each comparison. The
            csv is saved using the write.table line, so change the
            path to fit your directory structure.</p>
            <div class="org-src-container">
              <pre class="src src-jupyter-R">d <span style=
              "color: #d3869b;">&lt;-</span> data.frame(Pest = c(rep(c(<span style="color: #b8bb26;">"DEET"</span>, <span style="color: #b8bb26;">"DEET"</span>, <span style="color: #b8bb26;">"PERM"</span>, <span style="color: #b8bb26;">"PERM"</span>))),
                Part = c(rep(<span style=
"color: #b8bb26;">"Body"</span>, <span style=
"color: #d3869b; font-weight: bold;">4</span>), rep(<span style=
"color: #b8bb26;">"Leg"</span>, <span style=
"color: #d3869b; font-weight: bold;">4</span>)),
                Names = c(rep(c(<span style=
"color: #b8bb26;">"Up Regulated"</span>, <span style=
"color: #b8bb26;">"Down Regulated"</span>), <span style=
"color: #d3869b; font-weight: bold;">4</span>)),
                GeneCount = c(nrow(downdeetbody), nrow(updeetbody), nrow(downpermbody), nrow(uppermbody), nrow(downdeetleg), nrow(updeetleg), nrow(downpermleg), nrow(uppermleg)))
write.table(d, file=<span style=
"color: #b8bb26;">"edgeR-Genes/venn-regulation.csv"</span>, quote=<span style="color: #fabd2f;">FALSE</span>, col.names=<span style="color: #fabd2f;">TRUE</span>, row.names=<span style="color: #fabd2f;">FALSE</span>, sep=<span style="color: #b8bb26;">","</span>)
</pre>
            </div>
          </div>
        </div>
        <div id="outline-container-org2780626" class="outline-4">
          <h4 id="org2780626"><span class=
          "section-number-4">4.2.8</span> Gets the row-names</h4>
          <div class="outline-text-4" id="text-4-2-8">
            <p>This separates the data into just the gene names for
            each comparison. This is helpful for venn diagrams and
            other comparison.</p>
            <div class="org-src-container">
              <pre class="src src-jupyter-R">upperm <span style=
              "color: #d3869b;">&lt;-</span> table[table$permvcontrol &gt; <span style="color: #d3869b; font-weight: bold;">2</span>,]
downperm <span style=
"color: #d3869b;">&lt;-</span> table[table$permvcontrol &lt; -<span style="color: #d3869b; font-weight: bold;">2</span>,]
uppermbody <span style=
"color: #d3869b;">&lt;-</span> body[body$permvcontrol &gt; <span style="color: #d3869b; font-weight: bold;">2</span>,]
downpermbody <span style=
"color: #d3869b;">&lt;-</span> body[body$permvcontrol &lt; -<span style="color: #d3869b; font-weight: bold;">2</span>,]
uppermleg <span style=
"color: #d3869b;">&lt;-</span> leg[leg$permvcontrol &gt; <span style="color: #d3869b; font-weight: bold;">2</span>,]
downpermleg <span style=
"color: #d3869b;">&lt;-</span> leg[leg$permvcontrol &lt; -<span style="color: #d3869b; font-weight: bold;">2</span>,]

uppermgenes <span style=
"color: #d3869b;">&lt;-</span> rownames(upperm)
downpermgenes <span style=
"color: #d3869b;">&lt;-</span> rownames(downperm)
uppermleggenes <span style=
"color: #d3869b;">&lt;-</span> rownames(uppermleg)
downpermleggenes <span style=
"color: #d3869b;">&lt;-</span> rownames(downpermleg)
uppermbodygenes <span style=
"color: #d3869b;">&lt;-</span> rownames(uppermbody)
downpermbodygenes <span style=
"color: #d3869b;">&lt;-</span> rownames(downpermbody)

updeetgenes <span style=
"color: #d3869b;">&lt;-</span> rownames(updeet)
downdeetgenes <span style=
"color: #d3869b;">&lt;-</span> rownames(downdeet)
updeetleggenes <span style=
"color: #d3869b;">&lt;-</span> rownames(updeetleg)
downdeetleggenes <span style=
"color: #d3869b;">&lt;-</span> rownames(downdeetleg)
updeetbodygenes <span style=
"color: #d3869b;">&lt;-</span> rownames(updeetbody)
downdeetbodygenes <span style=
"color: #d3869b;">&lt;-</span> rownames(downdeetbody)
</pre>
            </div>
          </div>
        </div>
        <div id="outline-container-orga0d7f45" class="outline-4">
          <h4 id="orga0d7f45"><span class=
          "section-number-4">4.2.9</span> Create a list of
          dataframes corresponding to the total, leg, and body</h4>
          <div class="outline-text-4" id="text-4-2-9">
            <p>Consolidates the genes into a list of dataframes for
            easy saving</p>
            <div class="org-src-container">
              <pre class="src src-jupyter-R">genelist <span style=
              "color: #d3869b;">&lt;-</span> list(upPerm = uppermgenes, downPerm = downpermgenes,
                upPermBody=uppermbodygenes, downPermBody=downpermbodygenes,
                upPermLeg = uppermleggenes, downPermLeg = downpermleggenes,
                upDeet = updeetgenes, downDeet = downdeetgenes,
                upDeetLeg = updeetleggenes, downDeetLeg = downdeetleggenes,
                upDeetBody = updeetbodygenes, downDeetBody = downdeetbodygenes, table=rownames(table), leg=rownames(leg), body=rownames(body))
</pre>
            </div>
          </div>
        </div>
        <div id="outline-container-orgfb8f4b4" class="outline-4">
          <h4 id="orgfb8f4b4"><span class=
          "section-number-4">4.2.10</span> Save the genelist to
          files</h4>
          <div class="outline-text-4" id="text-4-2-10">
            <p>Saves all of the comparisons to file.</p>
            <div class="org-src-container">
              <pre class="src src-jupyter-R"><span style=
              "color: #fb4934;">for</span> (i <span style=
              "color: #fb4934;">in</span> seq(length(genelist))){
    namelist <span style=
"color: #d3869b;">&lt;-</span> names(genelist)
    write.table(genelist[i], file=paste(<span style=
"color: #b8bb26;">"edgeR-Genes/Genelists"</span>,<span style=
"color: #b8bb26;">"/"</span>, namelist[i], <span style=
"color: #b8bb26;">".csv"</span>, sep=<span style=
"color: #b8bb26;">""</span>), quote=<span style=
"color: #fabd2f;">FALSE</span>, col.names=<span style=
"color: #fabd2f;">FALSE</span>, row.names=<span style=
"color: #fabd2f;">FALSE</span>, sep=<span style=
"color: #b8bb26;">","</span>)
}
</pre>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div id="outline-container-org0459a36" class="outline-2">
      <h2 id="org0459a36"><span class="section-number-2">5</span>
      Fasta Manipulation</h2>
      <div class="outline-text-2" id="text-5"></div>
      <div id="outline-container-org742e04d" class="outline-3">
        <h3 id="org742e04d"><span class=
        "section-number-3">5.1</span> Get Gene names</h3>
        <div class="outline-text-3" id="text-5-1">
          <p>This program just pulls the names of the genes out of
          the fasta file. There are probably less messy ways to do
          this. Online programs exist for this as well, but this is
          a fairly simple c++ program.</p>
          <div class="org-src-container">
            <pre class="src src-cpp"><span style=
            "color: #ebdbb2; font-weight: bold;">#include</span> <span style=
            "color: #fe8019;">&lt;</span><span style=
            "color: #b8bb26;">iostream</span><span style=
            "color: #fe8019;">&gt;</span>
<span style=
"color: #ebdbb2; font-weight: bold;">#include</span> <span style=
"color: #fe8019;">&lt;</span><span style=
"color: #b8bb26;">fstream</span><span style=
"color: #fe8019;">&gt;</span>
<span style=
"color: #ebdbb2; font-weight: bold;">#include</span> <span style=
"color: #fe8019;">&lt;</span><span style=
"color: #b8bb26;">string</span><span style=
"color: #fe8019;">&gt;</span>
<span style=
"color: #ebdbb2; font-weight: bold;">#include</span> <span style=
"color: #fe8019;">&lt;</span><span style=
"color: #b8bb26;">vector</span><span style=
"color: #fe8019;">&gt;</span>

<span style="color: #fabd2f;">int</span> <span style=
"color: #b8bb26;">main</span><span style=
"color: #fe8019;">(){</span>
    <span style="color: #d3869b;">std</span>::<span style=
"color: #fabd2f;">ifstream</span> <span style=
"color: #83a598;">in</span>;
    in.open<span style="color: #fb4934;">(</span><span style=
"color: #b8bb26;">"GeneReference/genelist.fasta"</span><span style=
"color: #fb4934;">)</span>;
    <span style="color: #d3869b;">std</span>::<span style=
"color: #fabd2f;">string</span> <span style=
"color: #83a598;">line</span>;
    <span style="color: #d3869b;">std</span>::<span style=
"color: #fabd2f;">ofstream</span> <span style=
"color: #83a598;">out</span>;
    out.open<span style="color: #fb4934;">(</span><span style=
"color: #b8bb26;">"GeneReference/listofgenes.csv"</span><span style="color: #fb4934;">)</span>;
    <span style="color: #fb4934;">if</span> <span style=
"color: #fb4934;">(</span>in.is_open<span style=
"color: #cc241d;">()</span><span style="color: #fb4934;">){</span>
        <span style="color: #fb4934;">while</span><span style=
"color: #cc241d;">(</span>getline<span style=
"color: #fe8019;">(</span>in, line<span style=
"color: #fe8019;">)</span><span style="color: #cc241d;">){</span>
            <span style="color: #fb4934;">if</span> <span style=
"color: #fe8019;">(</span>line<span style=
"color: #fb4934;">[</span><span style=
"color: #d3869b; font-weight: bold;">0</span><span style=
"color: #fb4934;">]</span> == <span style=
"color: #b8bb26;">'&gt;'</span><span style=
"color: #fe8019;">){</span>
                line.erase<span style=
"color: #fb4934;">(</span><span style=
"color: #d3869b; font-weight: bold;">0</span>, <span style=
"color: #d3869b; font-weight: bold;">1</span><span style=
"color: #fb4934;">)</span>;
                out &lt;&lt; line &lt;&lt; <span style=
"color: #d3869b;">std</span>::endl;
            <span style="color: #fe8019;">}</span>
        <span style="color: #cc241d;">}</span>
    <span style="color: #fb4934;">}</span>
<span style="color: #fe8019;">}</span>
</pre>
          </div>
        </div>
      </div>
    </div>
    <div id="outline-container-org599597c" class="outline-2">
      <h2 id="org599597c"><span class="section-number-2">6</span>
      Blast</h2>
    </div>
    <div id="outline-container-org123b092" class="outline-2">
      <h2 id="org123b092"><span class="section-number-2">7</span>
      Plotting</h2>
      <div class="outline-text-2" id="text-7">
        <p>This uses R and python to plot data.</p>
      </div>
      <div id="outline-container-org8d905a3" class="outline-3">
        <h3 id="org8d905a3"><span class=
        "section-number-3">7.1</span> Required Packages</h3>
        <div class="outline-text-3" id="text-7-1">
          <p>These packages are only required for plotting the
          data</p>
        </div>
        <div id="outline-container-org9e11df1" class="outline-4">
          <h4 id="org9e11df1"><span class=
          "section-number-4">7.1.1</span> R packages</h4>
          <div class="outline-text-4" id="text-7-1-1">
            <ul class="org-ul">
              <li>tidyverse (ggplot)</li>
              <li>ggpubr (correlation methods)</li>
            </ul>
          </div>
        </div>
        <div id="outline-container-orge14ce24" class="outline-4">
          <h4 id="orge14ce24"><span class=
          "section-number-4">7.1.2</span> Python packages</h4>
          <div class="outline-text-4" id="text-7-1-2">
            <ul class="org-ul">
              <li>Matplotlib</li>
              <li>pandas</li>
              <li>numpy</li>
              <li>matplotlib<sub>venn</sub></li>
            </ul>
          </div>
        </div>
      </div>
      <div id="outline-container-org1d772e8" class="outline-3">
        <h3 id="org1d772e8"><span class=
        "section-number-3">7.2</span> Venn Diagrams</h3>
        <div class="outline-text-3" id="text-7-2"></div>
        <div id="outline-container-orge8eb653" class="outline-4">
          <h4 id="orge8eb653"><span class=
          "section-number-4">7.2.1</span> Imports</h4>
          <div class="outline-text-4" id="text-7-2-1">
            <p>Change directory based on project</p>
            <div class="org-src-container">
              <pre class="src src-jupyter-python"><span style=
              "color: #fb4934;">import</span> matplotlib.pyplot <span style=
              "color: #fb4934;">as</span> plt
<span style=
"color: #fb4934;">from</span> matplotlib_venn <span style=
"color: #fb4934;">import</span> venn2, venn2_circles, venn2_unweighted
<span style="color: #fb4934;">import</span> pandas <span style=
"color: #fb4934;">as</span> pd
<span style="color: #fb4934;">import</span> numpy <span style=
"color: #fb4934;">as</span> np
<span style="color: #fb4934;">import</span> os
plt.style.use(<span style="color: #b8bb26;">"ggplot"</span>)
os.chdir(<span style=
"color: #b8bb26;">"/home/david/Documents/BenoitLab/RNA-seq"</span>)
</pre>
            </div>
          </div>
        </div>
        <div id="outline-container-org797d54b" class="outline-4">
          <h4 id="org797d54b"><span class=
          "section-number-4">7.2.2</span> Venn 1</h4>
          <div class="outline-text-4" id="text-7-2-2">
            <div class="org-src-container">
              <pre class="src src-jupyter-python"><span style=
              "color: #83a598;">data</span> = pd.read_csv(<span style=
              "color: #b8bb26;">"edgeR-Genes/venn-regulation.csv"</span>)
plt.style.use(<span style="color: #b8bb26;">"ggplot"</span>)
<span style="color: #83a598;">fig</span>, <span style=
"color: #83a598;">axes</span> = plt.subplots(<span style=
"color: #d3869b; font-weight: bold;">1</span>, <span style=
"color: #d3869b; font-weight: bold;">4</span>)


<span style="color: #83a598;">i</span> = <span style=
"color: #d3869b; font-weight: bold;">0</span>

<span style="color: #83a598;">maximum</span> = data[<span style=
"color: #b8bb26;">"GeneCount"</span>].<span style=
"color: #fe8019;">max</span>()

<span style="color: #fb4934;">for</span> pest <span style=
"color: #fb4934;">in</span> data[<span style=
"color: #b8bb26;">"Part"</span>].unique():
    <span style=
"color: #83a598;">pestdata</span> = data[data[<span style=
"color: #b8bb26;">"Part"</span>] == pest]
    <span style="color: #fb4934;">for</span> part <span style=
"color: #fb4934;">in</span> pestdata[<span style=
"color: #b8bb26;">"Pest"</span>].unique():
        partdata = pestdata[pestdata[<span style=
"color: #b8bb26;">"Pest"</span>] == part]
        x = partdata[<span style="color: #b8bb26;">"Names"</span>]
        y = partdata[<span style=
"color: #b8bb26;">"GeneCount"</span>]
        axes[i].bar(x, y, color=[<span style=
"color: #b8bb26;">"r"</span>, <span style=
"color: #b8bb26;">"b"</span>])
        axes[i].set_xticks(partdata[<span style=
"color: #b8bb26;">"Names"</span>])
        axes[i].set_ylim(<span style=
"color: #d3869b; font-weight: bold;">0</span>, maximum)
        <span style="color: #fb4934;">if</span> i != <span style=
"color: #d3869b; font-weight: bold;">0</span>:
            axes[i].get_yaxis().set_visible(<span style=
"color: #d3869b;">False</span>)
        axes[i].set_xlabel(part + <span style=
"color: #b8bb26;">"\n"</span> + pest)
        axes[<span style=
"color: #d3869b; font-weight: bold;">0</span>].set_ylabel(<span style="color: #b8bb26;">"Gene Count"</span>)

        i += <span style=
"color: #d3869b; font-weight: bold;">1</span>
fig.suptitle(<span style=
"color: #b8bb26;">"Gene Regulation"</span>)
fig.savefig(<span style=
"color: #b8bb26;">"fig.png"</span>, bbox_inches=<span style=
"color: #b8bb26;">"tight"</span>, dpi=<span style=
"color: #d3869b; font-weight: bold;">250</span>)
</pre>
            </div>
          </div>
        </div>
        <div id="outline-container-org046c289" class="outline-4">
          <h4 id="org046c289"><span class=
          "section-number-4">7.2.3</span> Venn 2</h4>
          <div class="outline-text-4" id="text-7-2-3">
            <div class="org-src-container">
              <pre class=
              "src src-jupyter-python">plt.style.use(<span style=
              "color: #b8bb26;">"ggplot"</span>)

<span style=
"color: #fb4934;">from</span> matplotlib_venn <span style=
"color: #fb4934;">import</span> venn2, venn2_circles, venn2_unweighted
<span style=
"color: #fb4934;">from</span> matplotlib_venn <span style=
"color: #fb4934;">import</span> venn3, venn3_circles


<span style=
"color: #83a598;">deet</span> = pd.read_csv(<span style="color: #b8bb26;">"Genes/deet.csv"</span>, header=<span style="color: #d3869b;">None</span>)
deetcont = pd.read_csv(<span style=
"color: #b8bb26;">"Genes/deetcont.csv"</span>, header=<span style=
"color: #d3869b;">None</span>)
perm = pd.read_csv(<span style=
"color: #b8bb26;">"Genes/perm.csv"</span>, header=<span style=
"color: #d3869b;">None</span>)
permcont = pd.read_csv(<span style=
"color: #b8bb26;">"Genes/permcont.csv"</span>, header=<span style=
"color: #d3869b;">None</span>)
genes = pd.read_csv(<span style=
"color: #b8bb26;">"Genes/listofgenes.csv"</span>, header=<span style="color: #d3869b;">None</span>)
<span style="color: #fb4934;">def</span> <span style=
"color: #b8bb26;">compare</span>(s1, s2):
    c = <span style="color: #fe8019;">len</span>(<span style=
"color: #fe8019;">list</span>(<span style=
"color: #fe8019;">set</span>(s1.iloc[:, <span style=
"color: #d3869b; font-weight: bold;">0</span>]) &amp; <span style=
"color: #fe8019;">set</span>(s2.iloc[:, <span style=
"color: #d3869b; font-weight: bold;">0</span>])))
    <span style="color: #fb4934;">return</span>(c)

diff = compare(deet, genes)
<span style="color: #fe8019;">len</span>(deet)
<span style="color: #83a598;">fig</span>, <span style=
"color: #83a598;">axes</span> = plt.subplots(<span style=
"color: #d3869b; font-weight: bold;">2</span>)
v1 = venn2_unweighted(subsets=(<span style=
"color: #fe8019;">len</span>(deet), <span style=
"color: #fe8019;">len</span>(deetcont), <span style=
"color: #fe8019;">len</span>(genes)-<span style=
"color: #fe8019;">len</span>(deetcont)-<span style=
"color: #fe8019;">len</span>(deet)), set_labels=(<span style=
"color: #b8bb26;">"Up Regulated"</span>, <span style=
"color: #b8bb26;">"Down Regulated"</span>), ax=axes[<span style=
"color: #d3869b; font-weight: bold;">0</span>])
v2 = venn2_unweighted(subsets=(<span style=
"color: #fe8019;">len</span>(perm), <span style=
"color: #fe8019;">len</span>(permcont), <span style=
"color: #fe8019;">len</span>(genes)-<span style=
"color: #fe8019;">len</span>(permcont)-<span style=
"color: #fe8019;">len</span>(perm)), set_labels=(<span style=
"color: #b8bb26;">"Up Regulated"</span>, <span style=
"color: #b8bb26;">"Down Regulated"</span>), ax=axes[<span style=
"color: #d3869b; font-weight: bold;">1</span>])
axes[<span style=
"color: #d3869b; font-weight: bold;">0</span>].set_title(<span style="color: #b8bb26;">"Deet"</span>)
axes[<span style=
"color: #d3869b; font-weight: bold;">1</span>].set_title(<span style="color: #b8bb26;">"Perm"</span>)
fig.tight_layout()
fig.savefig(<span style=
"color: #b8bb26;">"VennDiagrams/RevisedDiagrams/DeetnPerm.png"</span>, bbox_inxhes=<span style="color: #b8bb26;">"tight"</span>, dpi=<span style="color: #d3869b; font-weight: bold;">250</span>)
plt.show()
</pre>
            </div>
          </div>
        </div>
        <div id="outline-container-org0a5e26e" class="outline-4">
          <h4 id="org0a5e26e"><span class=
          "section-number-4">7.2.4</span> Venn 3</h4>
          <div class="outline-text-4" id="text-7-2-4">
            <p>This compares the expression of body and legs when
            compared to the total.</p>
            <div class="org-src-container">
              <pre class="src src-jupyter-python"><span style=
              "color: #83a598;">genes</span> = pd.read_csv(<span style=
              "color: #b8bb26;">"GeneReference/listofgenes.csv"</span>, header=<span style="color: #d3869b;">None</span>)
body = pd.read_csv(<span style=
"color: #b8bb26;">"GeneReference/Body.csv"</span>)
leg = pd.read_csv(<span style=
"color: #b8bb26;">"GeneReference/Leg.csv"</span>)

<span style="color: #83a598;">fig</span>, <span style=
"color: #83a598;">axes</span> = plt.subplots()

v = venn2_unweighted(subsets=(<span style=
"color: #fe8019;">len</span>(body), <span style=
"color: #fe8019;">len</span>(leg),<span style=
"color: #fe8019;">len</span>(genes)-<span style=
"color: #fe8019;">len</span>(leg) - <span style=
"color: #fe8019;">len</span>(body)), set_labels=(<span style=
"color: #b8bb26;">"Body"</span>, <span style=
"color: #b8bb26;">"Leg"</span>), ax=axes)
v.get_patch_by_id(<span style=
"color: #b8bb26;">"10"</span>).set_color(<span style=
"color: #b8bb26;">"blue"</span>)
v.get_patch_by_id(<span style=
"color: #b8bb26;">"11"</span>).set_color(<span style=
"color: #b8bb26;">"blue"</span>)
v.get_patch_by_id(<span style=
"color: #b8bb26;">"01"</span>).set_color(<span style=
"color: #b8bb26;">"grey"</span>)
v.get_patch_by_id(<span style=
"color: #b8bb26;">"01"</span>).set_alpha(<span style=
"color: #d3869b; font-weight: bold;">0.5</span>)
axes.set_title(<span style="color: #b8bb26;">"Leg vs Body"</span>)
fig.tight_layout()
fig.savefig(<span style=
"color: #b8bb26;">"figures/figure2/Deseq-BodyvsLeg.png"</span>, bbox_inches=<span style="color: #b8bb26;">"tight"</span>, dpi=<span style="color: #d3869b; font-weight: bold;">250</span>)
fig.savefig(<span style=
"color: #b8bb26;">"figures/figure2/Deseq-BodyvsLeg.pdf"</span>, bbox_inches=<span style="color: #b8bb26;">"tight"</span>, dpi=<span style="color: #d3869b; font-weight: bold;">250</span>)

plt.show()
</pre>
            </div>
            <div id="org4eb5f31" class="figure">
              <p><img src=
              "./.ob-jupyter/2a5270b9b5cd24647389aedd0601cb8de1d7b70d.png"
              alt=
              "2a5270b9b5cd24647389aedd0601cb8de1d7b70d.png" /></p>
            </div>
          </div>
        </div>
        <div id="outline-container-org8068903" class="outline-4">
          <h4 id="org8068903"><span class=
          "section-number-4">7.2.5</span> Venn 4</h4>
          <div class="outline-text-4" id="text-7-2-5">
            <div class="org-src-container">
              <pre class="src src-jupyter-python"><span style=
              "color: #83a598;">genes</span> = pd.read_csv(<span style=
              "color: #b8bb26;">"Genes/listofgenes.csv"</span>, header=<span style="color: #d3869b;">None</span>)
body = pd.read_csv(<span style=
"color: #b8bb26;">"Genes/Body.csv"</span>)
upDeetBody = pd.read_csv(<span style=
"color: #b8bb26;">"Genes/Genelists/upDeetBody.csv"</span>)
downDeetBody = pd.read_csv(<span style=
"color: #b8bb26;">"Genes/Genelists/downDeetBody.csv"</span>)


<span style="color: #83a598;">fig</span>, <span style=
"color: #83a598;">axes</span> = plt.subplots()

v = venn2_unweighted(subsets=(<span style=
"color: #fe8019;">len</span>(upDeetBody), <span style=
"color: #fe8019;">len</span>(downDeetBody), <span style=
"color: #fe8019;">len</span>(body)- <span style=
"color: #fe8019;">len</span>(upDeetBody)-<span style=
"color: #fe8019;">len</span>(downDeetBody)), set_labels=(<span style="color: #b8bb26;">"Up Regulated"</span>, <span style="color: #b8bb26;">"Down Regulated"</span>), ax=axes)
axes.set_title(<span style=
"color: #b8bb26;">"Deet Expression in Body Genes"</span>)
fig.tight_layout()
fig.savefig(<span style=
"color: #b8bb26;">"VennDiagrams/RevisedDiagrams/DeetBody.png"</span>)
plt.show()
</pre>
            </div>
          </div>
        </div>
        <div id="outline-container-org8a20a05" class="outline-4">
          <h4 id="org8a20a05"><span class=
          "section-number-4">7.2.6</span> Venn 5</h4>
          <div class="outline-text-4" id="text-7-2-6">
            <div class="org-src-container">
              <pre class="src src-jupyter-python"><span style=
              "color: #83a598;">leg</span> = pd.read_csv(<span style=
              "color: #b8bb26;">"Genes/Leg.csv"</span>)
<span style=
"color: #83a598;">upDeetLeg</span> = pd.read_csv(<span style=
"color: #b8bb26;">"Genes/Genelists/upDeetLeg.csv"</span>)
<span style=
"color: #83a598;">downDeetLeg</span> = pd.read_csv(<span style=
"color: #b8bb26;">"Genes/Genelists/downDeetLeg.csv"</span>)

<span style="color: #83a598;">fig</span>, <span style=
"color: #83a598;">axes</span> = plt.subplots()
<span style=
"color: #83a598;">v</span> = venn2_unweighted(subsets=(<span style=
"color: #fe8019;">len</span>(upDeetLeg), <span style=
"color: #fe8019;">len</span>(downDeetLeg), <span style=
"color: #fe8019;">len</span>(leg)-<span style=
"color: #fe8019;">len</span>(upDeetLeg)-<span style=
"color: #fe8019;">len</span>(downDeetLeg)), set_labels=(<span style="color: #b8bb26;">"Up Regulated"</span>, <span style="color: #b8bb26;">"Down Regulated"</span>), ax=axes)
axes.set_title(<span style=
"color: #b8bb26;">"Deet Expression in Leg Genes"</span>)
fig.tight_layout()
fig.savefig(<span style=
"color: #b8bb26;">"VennDiagrams/RevisedDiagrams/DeetLeg.png"</span>)
</pre>
            </div>
          </div>
        </div>
        <div id="outline-container-org2d731c5" class="outline-4">
          <h4 id="org2d731c5"><span class=
          "section-number-4">7.2.7</span> Venn 6</h4>
          <div class="outline-text-4" id="text-7-2-7">
            <div class="org-src-container">
              <pre class="src src-jupyter-python"><span style=
              "color: #83a598;">leg</span> = pd.read_csv(<span style=
              "color: #b8bb26;">"Genes/Leg.csv"</span>)
<span style=
"color: #83a598;">upPermLeg</span> = pd.read_csv(<span style=
"color: #b8bb26;">"Genes/Genelists/upPermLeg.csv"</span>)
<span style=
"color: #83a598;">downPermLeg</span> = pd.read_csv(<span style=
"color: #b8bb26;">"Genes/Genelists/downPermLeg.csv"</span>)

<span style="color: #83a598;">fig</span>, <span style=
"color: #83a598;">axes</span> = plt.subplots()
<span style=
"color: #83a598;">v</span> = venn2_unweighted(subsets=(<span style=
"color: #fe8019;">len</span>(upPermLeg), <span style=
"color: #fe8019;">len</span>(downPermLeg), <span style=
"color: #fe8019;">len</span>(leg)-<span style=
"color: #fe8019;">len</span>(downPermLeg)-<span style=
"color: #fe8019;">len</span>(upPermLeg)), set_labels=(<span style=
"color: #b8bb26;">"Up Regulated"</span>, <span style=
"color: #b8bb26;">"Down Regulated"</span>), ax=axes)
axes.set_title(<span style=
"color: #b8bb26;">"Perm Expression in Leg Genes"</span>)
fig.tight_layout()
fig.savefig(<span style=
"color: #b8bb26;">"VennDiagrams/RevisedDiagrams/PermLeg.png"</span>)
</pre>
            </div>
          </div>
        </div>
        <div id="outline-container-orgbbacdd6" class="outline-4">
          <h4 id="orgbbacdd6"><span class=
          "section-number-4">7.2.8</span> Venn 7</h4>
          <div class="outline-text-4" id="text-7-2-8">
            <div class="org-src-container">
              <pre class="src src-jupyter-python"><span style=
              "color: #83a598;">body</span> = pd.read_csv(<span style=
              "color: #b8bb26;">"Genes/Body.csv"</span>)
<span style=
"color: #83a598;">upPermBody</span> = pd.read_csv(<span style=
"color: #b8bb26;">"Genes/Genelists/upPermBody.csv"</span>)
<span style=
"color: #83a598;">downPermBody</span> = pd.read_csv(<span style=
"color: #b8bb26;">"Genes/Genelists/downPermLeg.csv"</span>)

<span style="color: #83a598;">fig</span>, <span style=
"color: #83a598;">axes</span> = plt.subplots()
<span style=
"color: #83a598;">v</span> = venn2_unweighted(subsets=(<span style=
"color: #fe8019;">len</span>(upPermBody), <span style=
"color: #fe8019;">len</span>(downPermBody), <span style=
"color: #fe8019;">len</span>(body)-<span style=
"color: #fe8019;">len</span>(upPermBody)-<span style=
"color: #fe8019;">len</span>(downPermBody)), set_labels=(<span style="color: #b8bb26;">"Up Regulated"</span>, <span style="color: #b8bb26;">"Down Regulated"</span>), ax=axes)
axes.set_title(<span style=
"color: #b8bb26;">"Perm Expression in Body Genes"</span>)
fig.tight_layout()
fig.savefig(<span style=
"color: #b8bb26;">"VennDiagrams/RevisedDiagrams/PermBody.png"</span>)
</pre>
            </div>
          </div>
        </div>
        <div id="outline-container-org14fa8b8" class="outline-4">
          <h4 id="org14fa8b8"><span class=
          "section-number-4">7.2.9</span> Venn 8</h4>
          <div class="outline-text-4" id="text-7-2-9">
            <div class="org-src-container">
              <pre class="src src-jupyter-python"><span style=
              "color: #83a598;">genes</span> = pd.read_csv(<span style=
              "color: #b8bb26;">"Genes/listofgenes.csv"</span>)
<span style=
"color: #83a598;">upPerm</span> = pd.read_csv(<span style=
"color: #b8bb26;">"Genes/Genelists/upPerm.csv"</span>)
<span style=
"color: #83a598;">downPerm</span> = pd.read_csv(<span style=
"color: #b8bb26;">"Genes/Genelists/downPerm.csv"</span>)

<span style="color: #83a598;">fig</span>, <span style=
"color: #83a598;">axes</span> = plt.subplots()
<span style=
"color: #83a598;">v</span> = venn2_unweighted(subsets=(<span style=
"color: #fe8019;">len</span>(upPerm), <span style=
"color: #fe8019;">len</span>(downPerm), <span style=
"color: #fe8019;">len</span>(genes)-<span style=
"color: #fe8019;">len</span>(upPerm)-<span style=
"color: #fe8019;">len</span>(downPerm)), set_labels=(<span style=
"color: #b8bb26;">"Up Regulated"</span>, <span style=
"color: #b8bb26;">"Down Regulated"</span>), ax=axes)
axes.set_title(<span style=
"color: #b8bb26;">"Perm Expression"</span>)
fig.tight_layout()
fig.savefig(<span style=
"color: #b8bb26;">"VennDiagrams/RevisedDiagrams/Perm.png"</span>)
</pre>
            </div>
          </div>
        </div>
        <div id="outline-container-orgdb0e21d" class="outline-4">
          <h4 id="orgdb0e21d"><span class=
          "section-number-4">7.2.10</span> Venn 9</h4>
          <div class="outline-text-4" id="text-7-2-10">
            <p>Body vs leg EdgeR</p>
            <div class="org-src-container">
              <pre class="src src-jupyter-python"><span style=
              "color: #83a598;">genes</span> = pd.read_csv(<span style=
              "color: #b8bb26;">"GeneReference/listofgenes.csv"</span>, header=<span style="color: #d3869b;">None</span>)
body = pd.read_csv(<span style=
"color: #b8bb26;">"edgeR-Genes/bodyExpr.csv"</span>)
leg = pd.read_csv(<span style=
"color: #b8bb26;">"edgeR-Genes/legExpr.csv"</span>)

<span style="color: #83a598;">fig</span>, <span style=
"color: #83a598;">axes</span> = plt.subplots()

v = venn2_unweighted(subsets=(<span style=
"color: #fe8019;">len</span>(body), <span style=
"color: #fe8019;">len</span>(leg),<span style=
"color: #fe8019;">len</span>(genes)-<span style=
"color: #fe8019;">len</span>(leg) - <span style=
"color: #fe8019;">len</span>(body)), set_labels=(<span style=
"color: #b8bb26;">"Body"</span>, <span style=
"color: #b8bb26;">"Leg"</span>), ax=axes)
v.get_patch_by_id(<span style=
"color: #b8bb26;">"10"</span>).set_color(<span style=
"color: #b8bb26;">"blue"</span>)
v.get_patch_by_id(<span style=
"color: #b8bb26;">"11"</span>).set_color(<span style=
"color: #b8bb26;">"blue"</span>)
v.get_patch_by_id(<span style=
"color: #b8bb26;">"01"</span>).set_color(<span style=
"color: #b8bb26;">"grey"</span>)
v.get_patch_by_id(<span style=
"color: #b8bb26;">"01"</span>).set_alpha(<span style=
"color: #d3869b; font-weight: bold;">0.5</span>)
axes.set_title(<span style="color: #b8bb26;">"Leg vs Body"</span>)
fig.tight_layout()
fig.savefig(<span style=
"color: #b8bb26;">"figures/figure2/EdgeR-BodyvsLeg.png"</span>, bbox_inches=<span style="color: #b8bb26;">"tight"</span>, dpi=<span style="color: #d3869b; font-weight: bold;">250</span>)
fig.savefig(<span style=
"color: #b8bb26;">"figures/figure2/EdgeR-BodyvsLeg.pdf"</span>, bbox_inches=<span style="color: #b8bb26;">"tight"</span>, dpi=<span style="color: #d3869b; font-weight: bold;">250</span>)

plt.show()
</pre>
            </div>
            <div id="org162af1f" class="figure">
              <p><img src=
              "./.ob-jupyter/38b67feca428f2217e85968601099bf697f1bfd5.png"
              alt=
              "38b67feca428f2217e85968601099bf697f1bfd5.png" /></p>
            </div>
          </div>
        </div>
      </div>
      <div id="outline-container-org7cc63b1" class="outline-3">
        <h3 id="org7cc63b1"><span class=
        "section-number-3">7.3</span> Pearson Correlation</h3>
        <div class="outline-text-3" id="text-7-3">
          <p>This correlates the results from edgeR and Deseq2.
          Using this, the correlation can make the analysis tool
          interchangeable.</p>
          <p>:header-args: :tangle
          scripts/pearson-correlation.r</p>
        </div>
        <div id="outline-container-org4dc927e" class="outline-4">
          <h4 id="org4dc927e"><span class=
          "section-number-4">7.3.1</span> Libraries</h4>
          <div class="outline-text-4" id="text-7-3-1">
            <div class="org-src-container">
              <pre class="src src-jupyter-R"><span style=
              "color: #d3869b;">library</span>(<span style=
              "color: #b8bb26;">"tidyverse"</span>)
<span style="color: #d3869b;">library</span>(<span style=
"color: #b8bb26;">"ggpubr"</span>)
</pre>
            </div>
            <pre class="example" id="org290aa56">
── Attaching packages ─────────────────────────────────────── tidyverse 1.3.1 ──

✔ ggplot2 3.3.3     ✔ purrr   0.3.4
✔ tibble  3.1.2     ✔ dplyr   1.0.6
✔ tidyr   1.1.3     ✔ stringr 1.4.0
✔ readr   1.4.0     ✔ forcats 0.5.1

── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
</pre>
          </div>
        </div>
        <div id="outline-container-org35653b5" class="outline-4">
          <h4 id="org35653b5"><span class=
          "section-number-4">7.3.2</span> Import the data</h4>
          <div class="outline-text-4" id="text-7-3-2">
            <div class="org-src-container">
              <pre class="src src-jupyter-R">EdgeR <span style=
              "color: #d3869b;">&lt;-</span> read.csv(<span style=
              "color: #b8bb26;">"/home/david/Documents/BenoitLab/RNA-seq/edgeR-Genes/bodyExpr.csv"</span>)
Deseq <span style=
"color: #d3869b;">&lt;-</span> read.csv(<span style=
"color: #b8bb26;">"/home/david/Documents/BenoitLab/RNA-seq/Deseq-Genes/total.csv"</span>)
common <span style=
"color: #d3869b;">&lt;-</span> intersect(row.names(Deseq), row.names(EdgeR))
EdgeRCommon <span style=
"color: #d3869b;">&lt;-</span> EdgeR[common, ]
DeseqCommon <span style=
"color: #d3869b;">&lt;-</span> Deseq[common, ]

ECol <span style=
"color: #d3869b;">&lt;-</span> EdgeRCommon[<span style=
"color: #b8bb26;">"logFC.partbody"</span>]
DCol <span style="color: #d3869b;">&lt;-</span> -<span style=
"color: #d3869b; font-weight: bold;">1</span>*DeseqCommon[<span style="color: #b8bb26;">"log2FoldChange"</span>]
newD <span style=
"color: #d3869b;">&lt;-</span> data.frame(DCol, ECol)
</pre>
            </div>
          </div>
        </div>
        <div id="outline-container-org639cbe6" class="outline-4">
          <h4 id="org639cbe6"><span class=
          "section-number-4">7.3.3</span> Plot the data</h4>
          <div class="outline-text-4" id="text-7-3-3">
            <div class="org-src-container">
              <pre class="src src-jupyter-R"><span style=
              "color: #d3869b;">library</span>(<span style=
              "color: #b8bb26;">"ggpubr"</span>)
pearson <span style=
"color: #d3869b;">&lt;-</span> ggscatter(newD, x = <span style=
"color: #b8bb26;">"log2FoldChange"</span>, y = <span style=
"color: #b8bb26;">"logFC.partbody"</span>,
          add = <span style=
"color: #b8bb26;">"reg.line"</span>, conf.int = <span style=
"color: #fabd2f;">TRUE</span>,
          cor.coef = <span style=
"color: #fabd2f;">TRUE</span>, cor.method = <span style=
"color: #b8bb26;">"pearson"</span>,
          xlab = <span style=
"color: #b8bb26;">"Deseq"</span>, ylab = <span style=
"color: #b8bb26;">"EdgeR"</span>)
pearson
ggsave(<span style=
"color: #b8bb26;">"/home/david/Documents/BenoitLab/RNA-seq/figures/figure1/pearson.png"</span>, pearson, png())
ggsave(<span style=
"color: #b8bb26;">"/home/david/Documents/BenoitLab/RNA-seq/figures/figure1/pearson.pdf"</span>, pearson, pdf())

</pre>
            </div>
            <pre class="example">
`geom_smooth()` using formula 'y ~ x'

Saving 6.67 x 6.67 in image

`geom_smooth()` using formula 'y ~ x'

Saving 7 x 7 in image

`geom_smooth()` using formula 'y ~ x'

</pre>
            <div id="org8949a01" class="figure">
              <p><img src=
              "./.ob-jupyter/f1067a6954fd040fbf8de1c030201143a06483aa.png"
              alt=
              "f1067a6954fd040fbf8de1c030201143a06483aa.png" /></p>
            </div>
          </div>
        </div>
        <div id="outline-container-orgc4cbc9c" class="outline-4">
          <h4 id="orgc4cbc9c"><span class=
          "section-number-4">7.3.4</span> Venn Diagram</h4>
          <div class="outline-text-4" id="text-7-3-4">
            <p>Doesn’t do what I wanted</p>
            <div class="org-src-container">
              <pre class="src src-jupyter-R"><span style=
              "color: #d3869b;">library</span>(VennDiagram)
ECol <span style=
"color: #d3869b;">&lt;-</span> row.names(EdgeR[<span style=
"color: #b8bb26;">"logFC.partbody"</span>])
DCol <span style=
"color: #d3869b;">&lt;-</span> row.names(-<span style=
"color: #d3869b; font-weight: bold;">1</span>*Deseq[<span style=
"color: #b8bb26;">"log2FoldChange"</span>])
venn.diagram(
  x = list(na.omit(ECol), na.omit(DCol)),
  category.names = c(<span style=
"color: #b8bb26;">"EdgeR"</span>, <span style=
"color: #b8bb26;">"Deseq"</span>),
  filename = <span style=
"color: #b8bb26;">"/home/david/Documents/BenoitLab/RNA-seq/figures/figure1/vennBody.png"</span>,
  output=<span style="color: #fabd2f;">TRUE</span>
)
</pre>
            </div>
          </div>
        </div>
      </div>
      <div id="outline-container-org1c5d647" class="outline-3">
        <h3 id="org1c5d647"><span class=
        "section-number-3">7.4</span> Dotplot</h3>
        <div class="outline-text-3" id="text-7-4">
          <p>This figure plots the enrichment of specific genes.
          Ideal for small amounts of genes.</p>
        </div>
        <div id="outline-container-orgca6a5d7" class="outline-4">
          <h4 id="orgca6a5d7"><span class=
          "section-number-4">7.4.1</span> imports</h4>
          <div class="outline-text-4" id="text-7-4-1">
            <div class="org-src-container">
              <pre class="src src-jupyter-python"><span style=
              "color: #fb4934;">import</span> numpy <span style=
              "color: #fb4934;">as</span> np
<span style=
"color: #fb4934;">import</span> matplotlib.pyplot <span style=
"color: #fb4934;">as</span> plt
<span style="color: #fb4934;">import</span> matplotlib
<span style="color: #fb4934;">import</span> pandas <span style=
"color: #fb4934;">as</span> pd
</pre>
            </div>
          </div>
        </div>
        <div id="outline-container-orgf9fc0bb" class="outline-4">
          <h4 id="orgf9fc0bb"><span class=
          "section-number-4">7.4.2</span> Theme settings</h4>
          <div class="outline-text-4" id="text-7-4-2">
            <div class="org-src-container">
              <pre class="src src-jupyter-python">plt.<span style=
              "color: #83a598;">rcParams</span>[<span style=
              "color: #b8bb26;">"font.family"</span>] = <span style=
              "color: #b8bb26;">"Bahnschrift"</span>
<span style=
"color: #83a598;">cmap</span> = matplotlib.colors.LinearSegmentedColormap.from_list(
    <span style="color: #b8bb26;">""</span>, [<span style=
"color: #b8bb26;">"dodgerblue"</span>, <span style=
"color: #b8bb26;">"lightgrey"</span>]
)

</pre>
            </div>
          </div>
        </div>
        <div id="outline-container-orgd2fbb04" class="outline-4">
          <h4 id="orgd2fbb04"><span class=
          "section-number-4">7.4.3</span> Import</h4>
          <div class="outline-text-4" id="text-7-4-3">
            <p>Use the pandas package to import the csv file</p>
            <div class="org-src-container">
              <pre class="src src-jupyter-python"><span style=
              "color: #83a598;">dotplotdata</span> = pd.read_csv(<span style=
              "color: #b8bb26;">"../GeneReference/dotplotdata.csv"</span>)  <span style="color: #928374;"># </span><span style="color: #928374;">infers header existance</span>

<span style=
"color: #83a598;">pesticide</span> = dotplotdata.Category
<span style="color: #83a598;">pathway</span> = dotplotdata.Pathway
<span style=
"color: #83a598;">enrichmentratio</span> = dotplotdata.Enrichment
<span style=
"color: #83a598;">pval</span> = dotplotdata[<span style="color: #b8bb26;">"P-Value"</span>]
</pre>
            </div>
          </div>
        </div>
        <div id="outline-container-orgdc7fda4" class="outline-4">
          <h4 id="orgdc7fda4"><span class=
          "section-number-4">7.4.4</span> Plot the figure</h4>
          <div class="outline-text-4" id="text-7-4-4">
            <div class="org-src-container">
              <pre class="src src-jupyter-python"><span style=
              "color: #83a598;">fig</span>, <span style=
              "color: #83a598;">ax</span> = plt.subplots(<span style=
              "color: #d3869b; font-weight: bold;">1</span>)
<span style="color: #83a598;">p</span> = ax.scatter(
    pathway, pesticide, c=pval, s=enrichmentratio * <span style=
"color: #d3869b; font-weight: bold;">3</span>, cmap=cmap, vmin=<span style="color: #d3869b; font-weight: bold;">0</span>, vmax=<span style="color: #d3869b; font-weight: bold;">0.05</span>
)
plt.legend(loc=<span style=
"color: #b8bb26;">"lower left"</span>, markerscale=<span style=
"color: #d3869b; font-weight: bold;">2.0</span>, scatterpoints=<span style="color: #d3869b; font-weight: bold;">1</span>, fontsize=<span style="color: #d3869b; font-weight: bold;">10</span>)
plt.xticks(rotation=<span style=
"color: #b8bb26;">"vertical"</span>)

<span style="color: #928374;"># </span><span style=
"color: #928374;">Legend:</span>
handles, labels = p.legend_elements(prop=<span style=
"color: #b8bb26;">"sizes"</span>, alpha=<span style=
"color: #d3869b; font-weight: bold;">0.6</span>, num=<span style=
"color: #d3869b; font-weight: bold;">4</span>, func=<span style=
"color: #fb4934;">lambda</span> x : x/<span style=
"color: #d3869b; font-weight: bold;">3</span>)
legend2 = ax.legend(
    handles,
    labels,
    loc=<span style="color: #b8bb26;">"upper right"</span>,
    title=<span style="color: #b8bb26;">"Enrichment Ratio"</span>,
    bbox_to_anchor=(<span style=
"color: #d3869b; font-weight: bold;">1.4</span>, <span style=
"color: #d3869b; font-weight: bold;">1.1</span>),
)

<span style="color: #928374;"># </span><span style=
"color: #928374;">Colorbar:</span>
cbar = fig.colorbar(p, shrink=<span style=
"color: #d3869b; font-weight: bold;">0.5</span>)
cbar.ax.set_ylabel(<span style=
"color: #b8bb26;">"p Value"</span>, rotation=<span style=
"color: #d3869b; font-weight: bold;">270</span>, labelpad=<span style="color: #d3869b; font-weight: bold;">15</span>)

<span style="color: #928374;"># </span><span style=
"color: #928374;">Title, labels, and visual modifications:</span>
plt.title(<span style=
"color: #b8bb26;">"Metabolic Pathway Enrichment"</span>)
plt.xlabel(<span style=
"color: #b8bb26;">"Metabolic Pathway"</span>)
plt.ylabel(<span style="color: #b8bb26;">"Pesticide"</span>)
ax.grid(axis=<span style=
"color: #b8bb26;">"x"</span>, linestyle=<span style=
"color: #b8bb26;">"dotted"</span>)
ax.spines[<span style=
"color: #b8bb26;">"top"</span>].set_visible(<span style=
"color: #d3869b;">False</span>)
ax.spines[<span style=
"color: #b8bb26;">"right"</span>].set_visible(<span style=
"color: #d3869b;">False</span>)

plt.show()
</pre>
            </div>
          </div>
        </div>
        <div id="outline-container-orgdb7651d" class="outline-4">
          <h4 id="orgdb7651d"><span class=
          "section-number-4">7.4.5</span> Save the Figure</h4>
          <div class="outline-text-4" id="text-7-4-5">
            <div class="org-src-container">
              <pre class=
              "src src-jupyter-python">plt.savefig(<span style=
              "color: #b8bb26;">'dotplot.pdf'</span>, dpi=<span style=
              "color: #d3869b; font-weight: bold;">300</span>, bbox_inches=<span style="color: #b8bb26;">'tight'</span>)
plt.savefig(<span style=
"color: #b8bb26;">'dotplot.jpg'</span>, dpi=<span style=
"color: #d3869b; font-weight: bold;">300</span>, bbox_inches=<span style="color: #b8bb26;">'tight'</span>)
</pre>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div id="postamble" class="status">
    <p class="author">Author: David Lewis</p>
    <p class="date">Created: 2021-05-20 Thu 13:36</p>
  </div>
</body>
</html>
