<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
"http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en">
<head>
<!-- 2021-05-27 Thu 10:55 -->
<meta http-equiv="Content-Type" content="text/html;charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1" />
<title>RNA-seq Leg Vs Body workflow</title>
<meta name="author" content="David Lewis" />
<meta name="generator" content="Org Mode" />
<style type="text/css">
 <!--/*--><![CDATA[/*><!--*/
  .title  { text-align: center;
             margin-bottom: .2em; }
  .subtitle { text-align: center;
              font-size: medium;
              font-weight: bold;
              margin-top:0; }
  .todo   { font-family: monospace; color: red; }
  .done   { font-family: monospace; color: green; }
  .priority { font-family: monospace; color: orange; }
  .tag    { background-color: #eee; font-family: monospace;
            padding: 2px; font-size: 80%; font-weight: normal; }
  .timestamp { color: #bebebe; }
  .timestamp-kwd { color: #5f9ea0; }
  .org-right  { margin-left: auto; margin-right: 0px;  text-align: right; }
  .org-left   { margin-left: 0px;  margin-right: auto; text-align: left; }
  .org-center { margin-left: auto; margin-right: auto; text-align: center; }
  .underline { text-decoration: underline; }
  #postamble p, #preamble p { font-size: 90%; margin: .2em; }
  p.verse { margin-left: 3%; }
  pre {
    border: 1px solid #ccc;
    box-shadow: 3px 3px 3px #eee;
    padding: 8pt;
    font-family: monospace;
    overflow: auto;
    margin: 1.2em;
  }
  pre.src {
    position: relative;
    overflow: auto;
    padding-top: 1.2em;
  }
  pre.src:before {
    display: none;
    position: absolute;
    background-color: white;
    top: -10px;
    right: 10px;
    padding: 3px;
    border: 1px solid black;
  }
  pre.src:hover:before { display: inline; margin-top: 14px;}
  /* Languages per Org manual */
  pre.src-asymptote:before { content: 'Asymptote'; }
  pre.src-awk:before { content: 'Awk'; }
  pre.src-C:before { content: 'C'; }
  /* pre.src-C++ doesn't work in CSS */
  pre.src-clojure:before { content: 'Clojure'; }
  pre.src-css:before { content: 'CSS'; }
  pre.src-D:before { content: 'D'; }
  pre.src-ditaa:before { content: 'ditaa'; }
  pre.src-dot:before { content: 'Graphviz'; }
  pre.src-calc:before { content: 'Emacs Calc'; }
  pre.src-emacs-lisp:before { content: 'Emacs Lisp'; }
  pre.src-fortran:before { content: 'Fortran'; }
  pre.src-gnuplot:before { content: 'gnuplot'; }
  pre.src-haskell:before { content: 'Haskell'; }
  pre.src-hledger:before { content: 'hledger'; }
  pre.src-java:before { content: 'Java'; }
  pre.src-js:before { content: 'Javascript'; }
  pre.src-latex:before { content: 'LaTeX'; }
  pre.src-ledger:before { content: 'Ledger'; }
  pre.src-lisp:before { content: 'Lisp'; }
  pre.src-lilypond:before { content: 'Lilypond'; }
  pre.src-lua:before { content: 'Lua'; }
  pre.src-matlab:before { content: 'MATLAB'; }
  pre.src-mscgen:before { content: 'Mscgen'; }
  pre.src-ocaml:before { content: 'Objective Caml'; }
  pre.src-octave:before { content: 'Octave'; }
  pre.src-org:before { content: 'Org mode'; }
  pre.src-oz:before { content: 'OZ'; }
  pre.src-plantuml:before { content: 'Plantuml'; }
  pre.src-processing:before { content: 'Processing.js'; }
  pre.src-python:before { content: 'Python'; }
  pre.src-R:before { content: 'R'; }
  pre.src-ruby:before { content: 'Ruby'; }
  pre.src-sass:before { content: 'Sass'; }
  pre.src-scheme:before { content: 'Scheme'; }
  pre.src-screen:before { content: 'Gnu Screen'; }
  pre.src-sed:before { content: 'Sed'; }
  pre.src-sh:before { content: 'shell'; }
  pre.src-sql:before { content: 'SQL'; }
  pre.src-sqlite:before { content: 'SQLite'; }
  /* additional languages in org.el's org-babel-load-languages alist */
  pre.src-forth:before { content: 'Forth'; }
  pre.src-io:before { content: 'IO'; }
  pre.src-J:before { content: 'J'; }
  pre.src-makefile:before { content: 'Makefile'; }
  pre.src-maxima:before { content: 'Maxima'; }
  pre.src-perl:before { content: 'Perl'; }
  pre.src-picolisp:before { content: 'Pico Lisp'; }
  pre.src-scala:before { content: 'Scala'; }
  pre.src-shell:before { content: 'Shell Script'; }
  pre.src-ebnf2ps:before { content: 'ebfn2ps'; }
  /* additional language identifiers per "defun org-babel-execute"
       in ob-*.el */
  pre.src-cpp:before  { content: 'C++'; }
  pre.src-abc:before  { content: 'ABC'; }
  pre.src-coq:before  { content: 'Coq'; }
  pre.src-groovy:before  { content: 'Groovy'; }
  /* additional language identifiers from org-babel-shell-names in
     ob-shell.el: ob-shell is the only babel language using a lambda to put
     the execution function name together. */
  pre.src-bash:before  { content: 'bash'; }
  pre.src-csh:before  { content: 'csh'; }
  pre.src-ash:before  { content: 'ash'; }
  pre.src-dash:before  { content: 'dash'; }
  pre.src-ksh:before  { content: 'ksh'; }
  pre.src-mksh:before  { content: 'mksh'; }
  pre.src-posh:before  { content: 'posh'; }
  /* Additional Emacs modes also supported by the LaTeX listings package */
  pre.src-ada:before { content: 'Ada'; }
  pre.src-asm:before { content: 'Assembler'; }
  pre.src-caml:before { content: 'Caml'; }
  pre.src-delphi:before { content: 'Delphi'; }
  pre.src-html:before { content: 'HTML'; }
  pre.src-idl:before { content: 'IDL'; }
  pre.src-mercury:before { content: 'Mercury'; }
  pre.src-metapost:before { content: 'MetaPost'; }
  pre.src-modula-2:before { content: 'Modula-2'; }
  pre.src-pascal:before { content: 'Pascal'; }
  pre.src-ps:before { content: 'PostScript'; }
  pre.src-prolog:before { content: 'Prolog'; }
  pre.src-simula:before { content: 'Simula'; }
  pre.src-tcl:before { content: 'tcl'; }
  pre.src-tex:before { content: 'TeX'; }
  pre.src-plain-tex:before { content: 'Plain TeX'; }
  pre.src-verilog:before { content: 'Verilog'; }
  pre.src-vhdl:before { content: 'VHDL'; }
  pre.src-xml:before { content: 'XML'; }
  pre.src-nxml:before { content: 'XML'; }
  /* add a generic configuration mode; LaTeX export needs an additional
     (add-to-list 'org-latex-listings-langs '(conf " ")) in .emacs */
  pre.src-conf:before { content: 'Configuration File'; }

  table { border-collapse:collapse; }
  caption.t-above { caption-side: top; }
  caption.t-bottom { caption-side: bottom; }
  td, th { vertical-align:top;  }
  th.org-right  { text-align: center;  }
  th.org-left   { text-align: center;   }
  th.org-center { text-align: center; }
  td.org-right  { text-align: right;  }
  td.org-left   { text-align: left;   }
  td.org-center { text-align: center; }
  dt { font-weight: bold; }
  .footpara { display: inline; }
  .footdef  { margin-bottom: 1em; }
  .figure { padding: 1em; }
  .figure p { text-align: center; }
  .equation-container {
    display: table;
    text-align: center;
    width: 100%;
  }
  .equation {
    vertical-align: middle;
  }
  .equation-label {
    display: table-cell;
    text-align: right;
    vertical-align: middle;
  }
  .inlinetask {
    padding: 10px;
    border: 2px solid gray;
    margin: 10px;
    background: #ffffcc;
  }
  #org-div-home-and-up
   { text-align: right; font-size: 70%; white-space: nowrap; }
  textarea { overflow-x: auto; }
  .linenr { font-size: smaller }
  .code-highlighted { background-color: #ffff00; }
  .org-info-js_info-navigation { border-style: none; }
  #org-info-js_console-label
    { font-size: 10px; font-weight: bold; white-space: nowrap; }
  .org-info-js_search-highlight
    { background-color: #ffff00; color: #000000; font-weight: bold; }
  .org-svg { width: 90%; }
  /*]]>*/-->
</style>
<link rel="stylesheet" type="text/css" href="https://fniessen.github.io/org-html-themes/src/readtheorg_theme/css/htmlize.css"/>
<link rel="stylesheet" type="text/css" href="https://fniessen.github.io/org-html-themes/src/readtheorg_theme/css/readtheorg.css"/>
<script src="https://ajax.googleapis.com/ajax/libs/jquery/2.1.3/jquery.min.js"></script>
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.4/js/bootstrap.min.js"></script>
<script type="text/javascript" src="https://fniessen.github.io/org-html-themes/src/lib/js/jquery.stickytableheaders.min.js"></script>
<script type="text/javascript" src="https://fniessen.github.io/org-html-themes/src/readtheorg_theme/js/readtheorg.js"></script>
<script type="text/javascript">
// @license magnet:?xt=urn:btih:e95b018ef3580986a04669f1b5879592219e2a7a&dn=public-domain.txt Public Domain
<!--/*--><![CDATA[/*><!--*/
     function CodeHighlightOn(elem, id)
     {
       var target = document.getElementById(id);
       if(null != target) {
         elem.classList.add("code-highlighted");
         target.classList.add("code-highlighted");
       }
     }
     function CodeHighlightOff(elem, id)
     {
       var target = document.getElementById(id);
       if(null != target) {
         elem.classList.remove("code-highlighted");
         target.classList.remove("code-highlighted");
       }
     }
    /*]]>*///-->
// @license-end
</script>
</head>
<body>
<div id="content">
<h1 class="title">RNA-seq Leg Vs Body workflow</h1>
<div id="table-of-contents">
<h2>Table of Contents</h2>
<div id="text-table-of-contents">
<ul>
<li><a href="#org8d6eece">1. Introduction</a></li>
<li><a href="#org6621474">2. Galaxy</a></li>
<li><a href="#org447061d">3. Metadata</a></li>
<li><a href="#orgec10a85">4. R functions</a>
<ul>
<li><a href="#org133f0ea">4.1. txi function</a></li>
<li><a href="#org097c6e4">4.2. Initial Deseq Function</a></li>
<li><a href="#org63fe5ba">4.3. ReDeseq</a></li>
<li><a href="#org78f13ae">4.4. initialEdgeR</a></li>
<li><a href="#org6c52396">4.5. estEdgeR</a></li>
</ul>
</li>
<li><a href="#orgeb3d67b">5. initial R step</a>
<ul>
<li><a href="#org7da62cc">5.1. <span class="todo TODO">TODO</span> Deseq</a>
<ul>
<li><a href="#orgd1eab9f">5.1.1. Package imports</a></li>
<li><a href="#org2deb9df">5.1.2. Initial Setting</a></li>
<li><a href="#org334ae20">5.1.3. Run Initial Deseq Functions</a></li>
<li><a href="#orgaccd7b3">5.1.4. Do initial data manipulation</a></li>
<li><a href="#org1e31def">5.1.5. Initial selection of the data</a></li>
<li><a href="#org4a3d3a6">5.1.6. Save lists of the respective up-regulated genes</a></li>
<li><a href="#orge66b3bd">5.1.7. Create a new analysis design</a></li>
<li><a href="#org9577b7f">5.1.8. Prepare the data for pheatmap plotting</a></li>
<li><a href="#orgf08691a">5.1.9. Integrity plots</a></li>
<li><a href="#orga303746">5.1.10. Another Deseq Design</a></li>
<li><a href="#orge0622af">5.1.11. Altered Selection of genes</a></li>
<li><a href="#org5c0f453">5.1.12. Save new selection to file</a></li>
<li><a href="#org9542019">5.1.13. Extra integrity plot</a></li>
<li><a href="#org770eb89">5.1.14. <span class="todo TODO">TODO</span> Extra EdgeR for venn diagrams</a></li>
</ul>
</li>
<li><a href="#orgec21115">5.2. EdgeR</a>
<ul>
<li><a href="#org25021c7">5.2.1. Initial settings</a></li>
<li><a href="#org934271c">5.2.2. Do the initial processing of the data</a></li>
<li><a href="#orgd35a1e4">5.2.3. Partion the data into body and leg tables</a></li>
<li><a href="#org492add9">5.2.4. Save the data tables to a file</a></li>
<li><a href="#org4857fa0">5.2.5. Find the relative expression values when compared to the control</a></li>
<li><a href="#org86b585d">5.2.6. Selection of the data</a></li>
<li><a href="#org165b956">5.2.7. Create a csv for bar plotting</a></li>
<li><a href="#orgca40d15">5.2.8. Gets the row-names</a></li>
<li><a href="#org7665858">5.2.9. Create a list of dataframes corresponding to the total, leg, and body</a></li>
<li><a href="#org6784337">5.2.10. Save the genelist to files</a></li>
</ul>
</li>
<li><a href="#org03f0886">5.3. WGCNA</a>
<ul>
<li><a href="#org903e254">5.3.1. Build data file for WGCNA from quant files</a></li>
<li><a href="#org73e157a">5.3.2. Build metadata truth table for WGCNA</a></li>
</ul>
</li>
</ul>
</li>
<li><a href="#org334a537">6. Fasta Manipulation</a>
<ul>
<li><a href="#orgc39cee3">6.1. Get Gene names</a></li>
</ul>
</li>
<li><a href="#orgae5df12">7. Blast</a>
<ul>
<li><a href="#org31795de">7.1. Fastblast</a></li>
</ul>
</li>
<li><a href="#org82de278">8. Gene Ontology analysis</a>
<ul>
<li><a href="#orgc7b8854">8.1. Tools</a></li>
<li><a href="#org0139119">8.2. Automation</a></li>
</ul>
</li>
<li><a href="#org815c7b5">9. Plotting</a>
<ul>
<li><a href="#orgd2c26be">9.1. Required Packages</a>
<ul>
<li><a href="#org1f07c6e">9.1.1. R packages</a></li>
<li><a href="#orgfbc0600">9.1.2. Python packages</a></li>
</ul>
</li>
<li><a href="#org2aa0d4b">9.2. Venn Diagrams</a>
<ul>
<li><a href="#org6ff181f">9.2.1. Imports</a></li>
<li><a href="#org5605f54">9.2.2. Venn 1</a></li>
<li><a href="#org91fafe0">9.2.3. Venn 2</a></li>
<li><a href="#orgdec66b7">9.2.4. Venn 3</a></li>
<li><a href="#orgfc39039">9.2.5. Venn 4</a></li>
<li><a href="#orgc131053">9.2.6. Venn 5</a></li>
<li><a href="#org06b0a96">9.2.7. Venn 6</a></li>
<li><a href="#orga1560e1">9.2.8. Venn 7</a></li>
<li><a href="#orgbae9859">9.2.9. Venn 8</a></li>
<li><a href="#org240a2ed">9.2.10. Venn 9</a></li>
</ul>
</li>
<li><a href="#org9896931">9.3. Pearson Correlation</a>
<ul>
<li><a href="#org83e4865">9.3.1. Libraries</a></li>
<li><a href="#orge891897">9.3.2. Import the data</a></li>
<li><a href="#orgf9ff952">9.3.3. Plot the data</a></li>
<li><a href="#orgd779246">9.3.4. Venn Diagram</a></li>
</ul>
</li>
<li><a href="#orgc8ebbcb">9.4. Dotplot</a>
<ul>
<li><a href="#org30cd483">9.4.1. imports</a></li>
<li><a href="#org07063b6">9.4.2. Theme settings</a></li>
<li><a href="#orge13fb43">9.4.3. Import</a></li>
<li><a href="#orgc785c6e">9.4.4. Plot the figure</a></li>
<li><a href="#org2109e46">9.4.5. Save the Figure</a></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<div id="outline-container-org8d6eece" class="outline-2">
<h2 id="org8d6eece"><span class="section-number-2">1</span> Introduction</h2>
<div class="outline-text-2" id="text-1">
<ul class="org-ul">
<li>The Ticks are sprayed with Permethrin and DEET</li>
<li>The Legs are separated from the bodies of the ticks and analyzed separately</li>
<li>The Fasta files + assembly are sent to Galaxy</li>
</ul>
</div>
</div>
<div id="outline-container-org6621474" class="outline-2">
<h2 id="org6621474"><span class="section-number-2">2</span> Galaxy</h2>
<div class="outline-text-2" id="text-2">
<ul class="org-ul">
<li>Import the fasta files + assembly to Galaxy</li>
<li>Run FastQC to determine data integrity and quality on each fasta</li>
<li>Run Trimmomatic to trim superfluous data from the fasta files</li>
<li>Run A fish analysis tool on the data (salmon, sailfish, kallisto quant) on each fasta file from trimmomatic.
This requires the assembly as an input</li>
<li>Export the quant files for local analysis with R</li>
</ul>
</div>
</div>

<div id="outline-container-org447061d" class="outline-2">
<h2 id="org447061d"><span class="section-number-2">3</span> Metadata</h2>
<div class="outline-text-2" id="text-3">
<p>
The metadata for this project describes the treatment, body part, and trial for this project. This metadata is used to interface with the tximport library in R. Tximport uses the metadata to organize the quant data in a way that Deseq2 and EdgeR can understand. An arbitrary number of columns is allowed, but the name column is required. Keep the metadata.csv file in parent directory to the quant files.
</p>

<table border="2" cellspacing="0" cellpadding="6" rules="groups" frame="hsides">


<colgroup>
<col  class="org-left" />

<col  class="org-left" />

<col  class="org-left" />

<col  class="org-right" />
</colgroup>
<thead>
<tr>
<th scope="col" class="org-left">name</th>
<th scope="col" class="org-left">pesticides</th>
<th scope="col" class="org-left">part</th>
<th scope="col" class="org-right">trial</th>
</tr>
</thead>
<tbody>
<tr>
<td class="org-left">Per_Bod_1.tabular</td>
<td class="org-left">perm</td>
<td class="org-left">body</td>
<td class="org-right">1</td>
</tr>

<tr>
<td class="org-left">Per_Bod_3.tabular</td>
<td class="org-left">perm</td>
<td class="org-left">body</td>
<td class="org-right">3</td>
</tr>

<tr>
<td class="org-left">Per_Leg_3.tabular</td>
<td class="org-left">perm</td>
<td class="org-left">leg</td>
<td class="org-right">3</td>
</tr>

<tr>
<td class="org-left">Per_Leg_2.tabular</td>
<td class="org-left">perm</td>
<td class="org-left">leg</td>
<td class="org-right">2</td>
</tr>

<tr>
<td class="org-left">Per_Leg_1.tabular</td>
<td class="org-left">perm</td>
<td class="org-left">leg</td>
<td class="org-right">1</td>
</tr>

<tr>
<td class="org-left">Deet_Bod_3.tabular</td>
<td class="org-left">deet</td>
<td class="org-left">body</td>
<td class="org-right">3</td>
</tr>

<tr>
<td class="org-left">Deet_Bod_2.tabular</td>
<td class="org-left">deet</td>
<td class="org-left">body</td>
<td class="org-right">2</td>
</tr>

<tr>
<td class="org-left">Deet_Bod_1.tabular</td>
<td class="org-left">deet</td>
<td class="org-left">body</td>
<td class="org-right">1</td>
</tr>

<tr>
<td class="org-left">Deet_Leg_3.tabular</td>
<td class="org-left">deet</td>
<td class="org-left">leg</td>
<td class="org-right">3</td>
</tr>

<tr>
<td class="org-left">Deet_Leg_2.tabular</td>
<td class="org-left">deet</td>
<td class="org-left">leg</td>
<td class="org-right">2</td>
</tr>

<tr>
<td class="org-left">Deet_Leg_1.tabular</td>
<td class="org-left">deet</td>
<td class="org-left">leg</td>
<td class="org-right">1</td>
</tr>

<tr>
<td class="org-left">Cont_Bod_3.tabular</td>
<td class="org-left">control</td>
<td class="org-left">body</td>
<td class="org-right">3</td>
</tr>

<tr>
<td class="org-left">Cont_Bod_2.tabular</td>
<td class="org-left">control</td>
<td class="org-left">body</td>
<td class="org-right">2</td>
</tr>

<tr>
<td class="org-left">Cont_Bod_1.tabular</td>
<td class="org-left">control</td>
<td class="org-left">body</td>
<td class="org-right">1</td>
</tr>

<tr>
<td class="org-left">Cont_Leg_3.tabular</td>
<td class="org-left">control</td>
<td class="org-left">leg</td>
<td class="org-right">3</td>
</tr>

<tr>
<td class="org-left">Cont_Leg_2a.tabular</td>
<td class="org-left">control</td>
<td class="org-left">leg</td>
<td class="org-right">2</td>
</tr>

<tr>
<td class="org-left">Cont_Leg_1.tabular</td>
<td class="org-left">control</td>
<td class="org-left">leg</td>
<td class="org-right">1</td>
</tr>
</tbody>
</table>
</div>
</div>

<div id="outline-container-orgec10a85" class="outline-2">
<h2 id="orgec10a85"><span class="section-number-2">4</span> R functions</h2>
<div class="outline-text-2" id="text-4">
</div>
<div id="outline-container-org133f0ea" class="outline-3">
<h3 id="org133f0ea"><span class="section-number-3">4.1</span> txi function</h3>
<div class="outline-text-3" id="text-4-1">
<p>
uses the tximport package to import the quant data. The tximport package is
used in both deseq and edgeR for data import. It is optional, but saves a lot
of time on initial data organization
</p>
<div class="org-src-container">
<pre class="src src-jupyter-R"><span style="color: #b8bb26;">txi</span> <span style="color: #d3869b;">&lt;-</span> <span style="color: #fb4934;">function</span><span style="color: #d3869b;">(</span>directory, quantdatafoldername, metadata, fishtype<span style="color: #d3869b;">){</span>
  <span style="color: #d3869b;">library</span><span style="color: #d3869b;">(</span>tximport<span style="color: #d3869b;">)</span>
  <span style="color: #d3869b;">setwd</span><span style="color: #d3869b;">(</span>directory<span style="color: #d3869b;">)</span>
  metadata <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">read.csv</span><span style="color: #d3869b;">(</span>metadata, header<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">TRUE</span><span style="color: #d3869b;">)</span>
  files <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">file.path</span><span style="color: #d3869b;">(</span>directory, quantdatafoldername, metadata$name<span style="color: #d3869b;">)</span>
  <span style="color: #b8bb26;">names</span><span style="color: #d3869b;">(</span>files<span style="color: #d3869b;">)</span> <span style="color: #d3869b;">&lt;-</span> metadata$name
  genes <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">read.table</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">"genes2.tabular"</span><span style="color: #d3869b;">)</span>
  txi <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">tximport</span><span style="color: #d3869b;">(</span>files, type<span style="color: #d3869b;">=</span>fishtype, tx2gene<span style="color: #d3869b;">=</span>genes<span style="color: #d3869b;">)</span>
  <span style="color: #fb4934;">return</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">list</span><span style="color: #d3869b;">(</span>txi, metadata<span style="color: #d3869b;">))</span>
<span style="color: #d3869b;">}</span>
</pre>
</div>
</div>
</div>

<div id="outline-container-org097c6e4" class="outline-3">
<h3 id="org097c6e4"><span class="section-number-3">4.2</span> Initial Deseq Function</h3>
<div class="outline-text-3" id="text-4-2">
<p>
Sets up the deseq data set object and returns it. Deseq uses a linear model to fit the data. Read more about it here <a href="https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#theory-behind-deseq2">Deseq2</a>.
</p>
<div class="org-src-container">
<pre class="src src-jupyter-R"><span style="color: #b8bb26;">initialDeseq</span> <span style="color: #d3869b;">&lt;-</span> <span style="color: #fb4934;">function</span><span style="color: #d3869b;">(</span>txi, metadata, design<span style="color: #d3869b;">=</span>~part <span style="color: #d3869b;">+</span> pesticides<span style="color: #d3869b;">){</span>
  <span style="color: #d3869b;">library</span><span style="color: #d3869b;">(</span>DESeq2<span style="color: #d3869b;">)</span>
  <span style="color: #928374;"># </span><span style="color: #928374;">Deseq2 dataset from tximport</span>
  data <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">DESeqDataSetFromTximport</span><span style="color: #d3869b;">(</span>txi, colData<span style="color: #d3869b;">=</span>metadata, design <span style="color: #d3869b;">=</span>design<span style="color: #d3869b;">)</span>
  Deseq <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">DESeq</span><span style="color: #d3869b;">(</span>data<span style="color: #d3869b;">)</span>
  results <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">results</span><span style="color: #d3869b;">(</span>Deseq<span style="color: #d3869b;">)</span>
  <span style="color: #b8bb26;">print</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">summary</span><span style="color: #d3869b;">(</span>results<span style="color: #d3869b;">))</span>
  <span style="color: #fb4934;">return</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">list</span><span style="color: #d3869b;">(</span>Deseq, results<span style="color: #d3869b;">))</span>
<span style="color: #d3869b;">}</span>
</pre>
</div>
</div>
</div>

<div id="outline-container-org63fe5ba" class="outline-3">
<h3 id="org63fe5ba"><span class="section-number-3">4.3</span> ReDeseq</h3>
<div class="outline-text-3" id="text-4-3">
<p>
sets up the deseq object with a different design criteria. The design changes how log2foldchange is expressed.
It is more difficult to get multifactorial analysis in a single deseq object than with edgeR. Changing the design is necessary for complicated analysis.
</p>
<div class="org-src-container">
<pre class="src src-jupyter-R"><span style="color: #b8bb26;">reDeseq</span> <span style="color: #d3869b;">&lt;-</span> <span style="color: #fb4934;">function</span><span style="color: #d3869b;">(</span>dds, design<span style="color: #d3869b;">=</span>~part<span style="color: #d3869b;">){</span>
  ddsMF <span style="color: #d3869b;">&lt;-</span> dds
  <span style="color: #b8bb26;">design</span><span style="color: #d3869b;">(</span>ddsMF<span style="color: #d3869b;">)</span> <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">formula</span><span style="color: #d3869b;">(</span>design<span style="color: #d3869b;">)</span>
  ddsMF <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">DESeq</span><span style="color: #d3869b;">(</span>ddsMF<span style="color: #d3869b;">)</span>
  Mres <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">results</span><span style="color: #d3869b;">(</span>ddsMF<span style="color: #d3869b;">)</span>
  <span style="color: #fb4934;">return</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">list</span><span style="color: #d3869b;">(</span>ddsMf, Mres<span style="color: #d3869b;">))</span>
  <span style="color: #d3869b;">}</span>
</pre>
</div>
</div>
</div>

<div id="outline-container-org78f13ae" class="outline-3">
<h3 id="org78f13ae"><span class="section-number-3">4.4</span> initialEdgeR</h3>
<div class="outline-text-3" id="text-4-4">
<p>
Sets up the EdgeR object and returns it. EdgeR uses a different linear model. Read more about it here <a href="https://www.bioconductor.org/packages/release/bioc/vignettes/edgeR/inst/doc/edgeRUsersGuide.pdf">edgeR</a>.
</p>
<div class="org-src-container">
<pre class="src src-jupyter-R"><span style="color: #b8bb26;">initialEdgeR</span> <span style="color: #d3869b;">&lt;-</span> <span style="color: #fb4934;">function</span><span style="color: #d3869b;">(</span>txi, meta<span style="color: #d3869b;">){</span>
  <span style="color: #d3869b;">library</span><span style="color: #d3869b;">(</span>edgeR<span style="color: #d3869b;">)</span>
  <span style="color: #928374;"># </span><span style="color: #928374;">EdgR dataset from tximport</span>
  counts <span style="color: #d3869b;">&lt;-</span> txi$counts
  length <span style="color: #d3869b;">&lt;-</span> txi$length
  normMat <span style="color: #d3869b;">&lt;-</span> length <span style="color: #d3869b;">/</span> <span style="color: #b8bb26;">exp</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">rowMeans</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">log</span><span style="color: #d3869b;">(</span>length<span style="color: #d3869b;">)))</span>
  normCts <span style="color: #d3869b;">&lt;-</span> counts <span style="color: #d3869b;">/</span> normMat
  eff.lib <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">calcNormFactors</span><span style="color: #d3869b;">(</span>normCts<span style="color: #d3869b;">)</span> * <span style="color: #b8bb26;">colSums</span><span style="color: #d3869b;">(</span>normCts<span style="color: #d3869b;">)</span>
  normMat <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">sweep</span><span style="color: #d3869b;">(</span>normMat, <span style="color: #d3869b; font-weight: bold;">2</span>, eff.lib, <span style="color: #b8bb26;">"*"</span><span style="color: #d3869b;">)</span>
  normMat <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">log</span><span style="color: #d3869b;">(</span>normMat<span style="color: #d3869b;">)</span>
  y <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">DGEList</span><span style="color: #d3869b;">(</span>counts<span style="color: #d3869b;">)</span>
  y <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">scaleOffset</span><span style="color: #d3869b;">(</span>y, normMat<span style="color: #d3869b;">)</span>
  
  pest <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">factor</span><span style="color: #d3869b;">(</span>meta$pesticides, levels<span style="color: #d3869b;">=</span><span style="color: #b8bb26;">c</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">"perm"</span>, <span style="color: #b8bb26;">"deet"</span>, <span style="color: #b8bb26;">"control"</span><span style="color: #d3869b;">))</span>
  part <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">factor</span><span style="color: #d3869b;">(</span>meta$part, levels<span style="color: #d3869b;">=</span><span style="color: #b8bb26;">c</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">"leg"</span>, <span style="color: #b8bb26;">"body"</span><span style="color: #d3869b;">))</span>
  design <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">model.matrix</span><span style="color: #d3869b;">(</span>~<span style="color: #d3869b; font-weight: bold;">0</span> <span style="color: #d3869b;">+</span> pest <span style="color: #d3869b;">+</span> part<span style="color: #d3869b;">)</span>
  keep <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">filterByExpr</span><span style="color: #d3869b;">(</span>y, design<span style="color: #d3869b;">)</span>
  EdgeR <span style="color: #d3869b;">&lt;-</span> y<span style="color: #d3869b;">[</span>keep, <span style="color: #d3869b;">]</span>
  <span style="color: #fb4934;">return</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">list</span><span style="color: #d3869b;">(</span>EdgeR, design<span style="color: #d3869b;">))</span>
<span style="color: #d3869b;">}</span>
</pre>
</div>
</div>
</div>
<div id="outline-container-org6c52396" class="outline-3">
<h3 id="org6c52396"><span class="section-number-3">4.5</span> estEdgeR</h3>
<div class="outline-text-3" id="text-4-5">
<p>
finds the fit of the data and returns it for further analysis. Uses the edgeR functions to estimate dispersion of
the genes. It is much faster than deseq2.
</p>
<div class="org-src-container">
<pre class="src src-jupyter-R"><span style="color: #b8bb26;">estEdgeR</span> <span style="color: #d3869b;">&lt;-</span> <span style="color: #fb4934;">function</span><span style="color: #d3869b;">(</span>EdgeR, design<span style="color: #d3869b;">){</span>
  <span style="color: #d3869b;">library</span><span style="color: #d3869b;">(</span>edgeR<span style="color: #d3869b;">)</span>
  y <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">estimateGLMCommonDisp</span><span style="color: #d3869b;">(</span>EdgeR, design<span style="color: #d3869b;">)</span>
  y <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">estimateGLMTrendedDisp</span><span style="color: #d3869b;">(</span>y, design<span style="color: #d3869b;">)</span>
  y <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">estimateGLMTagwiseDisp</span><span style="color: #d3869b;">(</span>y, design<span style="color: #d3869b;">)</span>
  fit <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">glmQLFit</span><span style="color: #d3869b;">(</span>y, design<span style="color: #d3869b;">)</span> 
  
  <span style="color: #fb4934;">return</span><span style="color: #d3869b;">(</span>fit<span style="color: #d3869b;">)</span>
<span style="color: #d3869b;">}</span>
</pre>
</div>
</div>
</div>
</div>
<div id="outline-container-orgeb3d67b" class="outline-2">
<h2 id="orgeb3d67b"><span class="section-number-2">5</span> initial R step</h2>
<div class="outline-text-2" id="text-5">
<p>
EdgeR and Deseq2 are used to find lists of genes that are up-regulated and down-regulated when compared to design criteria. The design is used to curated the value of logfoldchange in Deseq2. They are compared using a pearson correlation in the plotting section.
</p>
</div>
<div id="outline-container-org7da62cc" class="outline-3">
<h3 id="org7da62cc"><span class="section-number-3">5.1</span> <span class="todo TODO">TODO</span> Deseq</h3>
<div class="outline-text-3" id="text-5-1">
<p>
This needs to incorporate the function definitions. Much of it is being restated in an unintuitive way
</p>
</div>
<div id="outline-container-orgd1eab9f" class="outline-4">
<h4 id="orgd1eab9f"><span class="section-number-4">5.1.1</span> Package imports</h4>
<div class="outline-text-4" id="text-5-1-1">
<div class="org-src-container">
<pre class="src src-jupyter-R"><span style="color: #b8bb26;">vignette</span><span style="color: #d3869b;">(</span>package<span style="color: #d3869b;">=</span><span style="color: #b8bb26;">"dplyr"</span><span style="color: #d3869b;">)</span>

rdata <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">list.files</span><span style="color: #d3869b;">(</span>pattern<span style="color: #d3869b;">=</span><span style="color: #b8bb26;">"\\.Rdata"</span><span style="color: #d3869b;">)</span>
args <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">commandArgs</span><span style="color: #d3869b;">(</span>trailingOnly <span style="color: #d3869b;">=</span> <span style="color: #fabd2f;">TRUE</span><span style="color: #d3869b;">)</span>
<span style="color: #fb4934;">if</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">length</span><span style="color: #d3869b;">(</span>args<span style="color: #d3869b;">)</span> <span style="color: #d3869b;">&gt;</span> <span style="color: #d3869b; font-weight: bold;">0</span><span style="color: #d3869b;">){</span>
    <span style="color: #fb4934;">if</span><span style="color: #d3869b;">(</span>args <span style="color: #d3869b;">==</span> <span style="color: #b8bb26;">"rerun"</span><span style="color: #d3869b;">){</span>
        <span style="color: #b8bb26;">print</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">"rerunning data"</span><span style="color: #d3869b;">)</span>
        r
    <span style="color: #d3869b;">}</span>
<span style="color: #d3869b;">}</span>
<span style="color: #928374;"># </span><span style="color: #928374;">general library imports ----</span>
<span style="color: #d3869b;">library</span><span style="color: #d3869b;">(</span>tximport<span style="color: #d3869b;">)</span>
<span style="color: #d3869b;">library</span><span style="color: #d3869b;">(</span>readr<span style="color: #d3869b;">)</span>
<span style="color: #d3869b;">library</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">"pheatmap"</span><span style="color: #d3869b;">)</span>
</pre>
</div>
</div>
</div>

<div id="outline-container-org2deb9df" class="outline-4">
<h4 id="org2deb9df"><span class="section-number-4">5.1.2</span> Initial Setting</h4>
<div class="outline-text-4" id="text-5-1-2">
<p>
These values determine what directory the deseq objects are taken from.
</p>
<div class="org-src-container">
<pre class="src src-jupyter-R"><span style="color: #928374;"># </span><span style="color: #928374;">change these values to your own situations</span>
directory <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">"/home/david/Documents/BenoitLab/R/largeRNASEQ"</span>
quantdatafoldername <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">"CorrectedQuantData"</span>
fishtype <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">"sailfish"</span>
</pre>
</div>
</div>
</div>

<div id="outline-container-org334ae20" class="outline-4">
<h4 id="org334ae20"><span class="section-number-4">5.1.3</span> Run Initial Deseq Functions</h4>
<div class="outline-text-4" id="text-5-1-3">
<div class="org-src-container">
<pre class="src src-jupyter-R"><span style="color: #d3869b;">source</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">"/home/david/Documents/BenoitLab/RNA-seq/scripts/functions.r"</span><span style="color: #d3869b;">)</span>
txi <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">txi</span><span style="color: #d3869b;">(</span>directory, quantdatafoldername, <span style="color: #b8bb26;">"metadata.csv"</span>, fishtype<span style="color: #d3869b;">)</span>
Deseq <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">initialDeseq</span><span style="color: #d3869b;">(</span>txi<span style="color: #d3869b;">[[</span><span style="color: #d3869b; font-weight: bold;">1</span><span style="color: #d3869b;">]]</span>, txi<span style="color: #d3869b;">[[</span><span style="color: #d3869b; font-weight: bold;">2</span><span style="color: #d3869b;">]])</span>
</pre>
</div>

<pre class="example" id="orgb217958">
reading in files with read_tsv

1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17


summarizing abundance

summarizing counts

summarizing length

Warning message in DESeqDataSet(se, design = design, ignoreRank):
“some variables in design formula are characters, converting to factors”
using counts and average transcript lengths from tximport

estimating size factors

using 'avgTxLength' from assays(dds), correcting for library size

estimating dispersions

gene-wise dispersion estimates

mean-dispersion relationship

final dispersion estimates

fitting model and testing


out of 163205 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 1790, 1.1%
LFC &lt; 0 (down)     : 2044, 1.3%
outliers [1]       : 2298, 1.4%
low counts [2]     : 18457, 11%
(mean count &lt; 3)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results

NULL
</pre>
</div>
</div>

<div id="outline-container-orgaccd7b3" class="outline-4">
<h4 id="orgaccd7b3"><span class="section-number-4">5.1.4</span> Do initial data manipulation</h4>
<div class="outline-text-4" id="text-5-1-4">
<p>
The part design is used to compare the leg to the body in total, disregarding pesticide. This is explained in detail in the pearson correlation section.
</p>
<div class="org-src-container">
<pre class="src src-jupyter-R">dds <span style="color: #d3869b;">&lt;-</span> Deseq<span style="color: #d3869b;">[[</span><span style="color: #d3869b; font-weight: bold;">1</span><span style="color: #d3869b;">]]</span>
res <span style="color: #d3869b;">&lt;-</span> Deseq<span style="color: #d3869b;">[[</span><span style="color: #d3869b; font-weight: bold;">2</span><span style="color: #d3869b;">]]</span>
<span style="color: #928374;">#</span><span style="color: #928374;">* vsd ntd rld</span>
vsd <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">vst</span><span style="color: #d3869b;">(</span>dds, blind<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">FALSE</span><span style="color: #d3869b;">)</span>
rld <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">rlog</span><span style="color: #d3869b;">(</span>dds, blind<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">FALSE</span><span style="color: #d3869b;">)</span>
ntd <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">normTransform</span><span style="color: #d3869b;">(</span>dds<span style="color: #d3869b;">)</span>

ddsMF <span style="color: #d3869b;">&lt;-</span> dds
<span style="color: #b8bb26;">design</span><span style="color: #d3869b;">(</span>ddsMF<span style="color: #d3869b;">)</span> <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">formula</span><span style="color: #d3869b;">(</span>~pesticides<span style="color: #d3869b;">)</span>
ddsMF <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">DESeq</span><span style="color: #d3869b;">(</span>ddsMF<span style="color: #d3869b;">)</span>
Mres <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">results</span><span style="color: #d3869b;">(</span>ddsMF<span style="color: #d3869b;">)</span>
<span style="color: #b8bb26;">design</span><span style="color: #d3869b;">(</span>dds<span style="color: #d3869b;">)</span> <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">formula</span><span style="color: #d3869b;">(</span>~part<span style="color: #d3869b;">)</span>
part <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">DESeq</span><span style="color: #d3869b;">(</span>dds<span style="color: #d3869b;">)</span>
res <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">results</span><span style="color: #d3869b;">(</span>part<span style="color: #d3869b;">)</span>
</pre>
</div>

<pre class="example" id="orgc538f1f">
using pre-existing normalization factors

estimating dispersions

found already estimated dispersions, replacing these

gene-wise dispersion estimates

mean-dispersion relationship

final dispersion estimates

fitting model and testing

using pre-existing normalization factors

estimating dispersions

found already estimated dispersions, replacing these

gene-wise dispersion estimates

mean-dispersion relationship

final dispersion estimates

fitting model and testing

-- replacing outliers and refitting for 4107 genes
-- DESeq argument 'minReplicatesForReplace' = 7
-- original counts are preserved in counts(dds)

estimating dispersions

fitting model and testing
</pre>
</div>
</div>
<div id="outline-container-org1e31def" class="outline-4">
<h4 id="org1e31def"><span class="section-number-4">5.1.5</span> Initial selection of the data</h4>
<div class="outline-text-4" id="text-5-1-5">
<ul class="org-ul">
<li>p value of less than 0.01</li>
<li>Leg is compared to body, therefore down-regulated are up-regulated in body</li>
</ul>
<div class="org-src-container">
<pre class="src src-jupyter-R">DR <span style="color: #d3869b;">&lt;-</span> Mres<span style="color: #d3869b;">[</span> <span style="color: #b8bb26;">which</span><span style="color: #d3869b;">(</span>Mres$padj <span style="color: #d3869b;">&lt;</span> <span style="color: #d3869b; font-weight: bold;">0.01</span><span style="color: #d3869b;">)</span>, <span style="color: #d3869b;">]</span>
Body <span style="color: #d3869b;">&lt;-</span> DR<span style="color: #d3869b;">[</span><span style="color: #b8bb26;">which</span><span style="color: #d3869b;">(</span>DR$log2FoldChange <span style="color: #d3869b;">&lt;</span> <span style="color: #d3869b;">-</span><span style="color: #d3869b; font-weight: bold;">2</span><span style="color: #d3869b;">)</span>,<span style="color: #d3869b;">]</span>
Leg <span style="color: #d3869b;">&lt;-</span> DR<span style="color: #d3869b;">[</span><span style="color: #b8bb26;">which</span><span style="color: #d3869b;">(</span>DR$log2FoldChange <span style="color: #d3869b;">&gt;</span> <span style="color: #d3869b; font-weight: bold;">2</span><span style="color: #d3869b;">)</span>, <span style="color: #d3869b;">]</span>
</pre>
</div>

<pre class="example" id="org4918a3a">
log2 fold change (MLE): pesticides perm vs control
Wald test p-value: pesticides perm vs control
DataFrame with 6 rows and 6 columns
            baseMean log2FoldChange     lfcSE       stat    pvalue      padj
           &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt;  &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
dv.1      47203.3539       0.365233  0.488997  0.7469016  0.455123  0.999996
dv.10       724.7318       0.116932  1.232421  0.0948798  0.924410  0.999996
dv.100     1474.4296       0.116376  0.297011  0.3918226  0.695189  0.999996
dv.1000     246.0981       1.380686  0.909599  1.5179057  0.129038  0.999996
dv.10000     56.9568      -0.744815  1.389576 -0.5360020  0.591957  0.999996
dv.100000    19.8319       0.319016  0.972153  0.3281536  0.742796  0.999996
</pre>
</div>
</div>

<div id="outline-container-org4a3d3a6" class="outline-4">
<h4 id="org4a3d3a6"><span class="section-number-4">5.1.6</span> Save lists of the respective up-regulated genes</h4>
<div class="outline-text-4" id="text-5-1-6">
<div class="org-src-container">
<pre class="src src-jupyter-R"><span style="color: #b8bb26;">write.table</span><span style="color: #d3869b;">(</span>Body, file<span style="color: #d3869b;">=</span><span style="color: #b8bb26;">"/home/david/Documents/BenoitLab/RNA-seq/Deseq-Genes/Body.csv"</span>, quote<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">FALSE</span>, col.names<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">TRUE</span>, row.names<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">TRUE</span>, sep<span style="color: #d3869b;">=</span><span style="color: #b8bb26;">","</span><span style="color: #d3869b;">)</span>
<span style="color: #b8bb26;">write.table</span><span style="color: #d3869b;">(</span>Leg, file<span style="color: #d3869b;">=</span><span style="color: #b8bb26;">"/home/david/Documents/BenoitLab/RNA-seq/Deseq-Genes/Leg.csv"</span>, quote<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">FALSE</span>, col.names<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">TRUE</span>, row.names<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">TRUE</span>, sep<span style="color: #d3869b;">=</span><span style="color: #b8bb26;">","</span><span style="color: #d3869b;">)</span>
<span style="color: #b8bb26;">write.table</span><span style="color: #d3869b;">(</span>res, file<span style="color: #d3869b;">=</span><span style="color: #b8bb26;">"/home/david/Documents/BenoitLab/RNA-seq/Deseq-Genes/total.csv"</span>, quote<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">FALSE</span>, col.names<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">TRUE</span>, row.names<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">TRUE</span>, sep<span style="color: #d3869b;">=</span><span style="color: #b8bb26;">","</span><span style="color: #d3869b;">)</span>
</pre>
</div>
</div>
</div>

<div id="outline-container-orge66b3bd" class="outline-4">
<h4 id="orge66b3bd"><span class="section-number-4">5.1.7</span> Create a new analysis design</h4>
<div class="outline-text-4" id="text-5-1-7">
<div class="org-src-container">
<pre class="src src-jupyter-R">newdd <span style="color: #d3869b;">&lt;-</span> dds
<span style="color: #b8bb26;">design</span><span style="color: #d3869b;">(</span>newdd<span style="color: #d3869b;">)</span> <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">formula</span><span style="color: #d3869b;">(</span>~part <span style="color: #d3869b;">+</span> part:pesticides<span style="color: #d3869b;">)</span>
newdd <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">DESeq</span><span style="color: #d3869b;">(</span>newdd<span style="color: #d3869b;">)</span>
newres <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">results</span><span style="color: #d3869b;">(</span>newdd<span style="color: #d3869b;">)</span>
</pre>
</div>
</div>
</div>
<div id="outline-container-org9577b7f" class="outline-4">
<h4 id="org9577b7f"><span class="section-number-4">5.1.8</span> Prepare the data for pheatmap plotting</h4>
<div class="outline-text-4" id="text-5-1-8">
<div class="org-src-container">
<pre class="src src-jupyter-R">select <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">order</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">rowMeans</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">counts</span><span style="color: #d3869b;">(</span>dds, normalized<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">FALSE</span><span style="color: #d3869b;">))</span>, decreasing <span style="color: #d3869b;">=</span> <span style="color: #fabd2f;">TRUE</span><span style="color: #d3869b;">)[</span><span style="color: #d3869b; font-weight: bold;">1</span>:<span style="color: #d3869b; font-weight: bold;">20</span><span style="color: #d3869b;">]</span>
df <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">as.data.frame</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">colData</span><span style="color: #d3869b;">(</span>dds<span style="color: #d3869b;">)[</span>,<span style="color: #b8bb26;">c</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">"pesticides"</span>,<span style="color: #b8bb26;">"part"</span><span style="color: #d3869b;">)])</span>
</pre>
</div>
</div>
</div>
<div id="outline-container-orgf08691a" class="outline-4">
<h4 id="orgf08691a"><span class="section-number-4">5.1.9</span> Integrity plots</h4>
<div class="outline-text-4" id="text-5-1-9">
<div class="org-src-container">
<pre class="src src-jupyter-R"><span style="color: #b8bb26;">jpeg</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">"generalfigures/plotMA_deseq.jpg"</span><span style="color: #d3869b;">)</span>
<span style="color: #b8bb26;">plotMA</span><span style="color: #d3869b;">(</span>results<span style="color: #d3869b;">)</span>
<span style="color: #b8bb26;">dev.off</span><span style="color: #d3869b;">()</span>

<span style="color: #b8bb26;">jpeg</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">"generalfigures/plotcounts_deseq.jpg"</span><span style="color: #d3869b;">)</span>
<span style="color: #b8bb26;">plotCounts</span><span style="color: #d3869b;">(</span>data, gene<span style="color: #d3869b;">=</span><span style="color: #b8bb26;">which.min</span><span style="color: #d3869b;">(</span>results$padj<span style="color: #d3869b;">)</span>, intgroup <span style="color: #d3869b;">=</span> <span style="color: #b8bb26;">"part"</span><span style="color: #d3869b;">)</span>
<span style="color: #b8bb26;">dev.off</span><span style="color: #d3869b;">()</span>

<span style="color: #b8bb26;">jpeg</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">"generalfigures/meanSdvsd_deseq.jpg"</span><span style="color: #d3869b;">)</span>
vsn::<span style="color: #b8bb26;">meanSdPlot</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">assay</span><span style="color: #d3869b;">(</span>vsd<span style="color: #d3869b;">))</span>
<span style="color: #b8bb26;">dev.off</span><span style="color: #d3869b;">()</span>

<span style="color: #b8bb26;">jpeg</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">"generalfigures/meadnSdrld_deseq.jpg"</span><span style="color: #d3869b;">)</span>
vsn::<span style="color: #b8bb26;">meanSdPlot</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">assay</span><span style="color: #d3869b;">(</span>rld<span style="color: #d3869b;">))</span>
<span style="color: #b8bb26;">dev.off</span><span style="color: #d3869b;">()</span>

<span style="color: #b8bb26;">jpeg</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">"generalfigures/meanSdntd_deseq.jpg"</span><span style="color: #d3869b;">)</span>
vsn::<span style="color: #b8bb26;">meanSdPlot</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">assay</span><span style="color: #d3869b;">(</span>ntd<span style="color: #d3869b;">))</span>
<span style="color: #b8bb26;">dev.off</span><span style="color: #d3869b;">()</span>

<span style="color: #b8bb26;">jpeg</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">"generalfigures/pcaplot_deseq.jpg"</span>, width <span style="color: #d3869b;">=</span> <span style="color: #d3869b; font-weight: bold;">800</span>, height<span style="color: #d3869b;">=</span><span style="color: #d3869b; font-weight: bold;">800</span><span style="color: #d3869b;">)</span>
<span style="color: #b8bb26;">plotPCA</span><span style="color: #d3869b;">(</span>vsd, intgroup<span style="color: #d3869b;">=</span><span style="color: #b8bb26;">c</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">"part"</span>, <span style="color: #b8bb26;">"pesticides"</span><span style="color: #d3869b;">))</span>
<span style="color: #b8bb26;">dev.off</span><span style="color: #d3869b;">()</span>

<span style="color: #b8bb26;">jpeg</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">"generalfigures/pheatmapplotntd.jpg"</span>, width<span style="color: #d3869b;">=</span><span style="color: #d3869b; font-weight: bold;">800</span>, height<span style="color: #d3869b;">=</span><span style="color: #d3869b; font-weight: bold;">800</span><span style="color: #d3869b;">)</span>
<span style="color: #b8bb26;">pheatmap</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">assay</span><span style="color: #d3869b;">(</span>ntd<span style="color: #d3869b;">)[</span>select,<span style="color: #d3869b;">]</span>, cluster_rows<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">FALSE</span>, show_rownames<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">TRUE</span>, cluster_cols<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">TRUE</span>, annotation_col<span style="color: #d3869b;">=</span>df<span style="color: #d3869b;">)</span>
<span style="color: #b8bb26;">dev.off</span><span style="color: #d3869b;">()</span>

<span style="color: #b8bb26;">jpeg</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">"generalfigures/pheatmapplotvsd.jpg"</span>, width<span style="color: #d3869b;">=</span><span style="color: #d3869b; font-weight: bold;">800</span>, height<span style="color: #d3869b;">=</span><span style="color: #d3869b; font-weight: bold;">800</span><span style="color: #d3869b;">)</span>
<span style="color: #b8bb26;">pheatmap</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">assay</span><span style="color: #d3869b;">(</span>vsd<span style="color: #d3869b;">)[</span>select,<span style="color: #d3869b;">]</span>, cluster_rows<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">FALSE</span>, show_rownames<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">TRUE</span>, cluster_cols<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">FALSE</span>, annotation_col<span style="color: #d3869b;">=</span>df<span style="color: #d3869b;">)</span>
<span style="color: #b8bb26;">dev.off</span><span style="color: #d3869b;">()</span>

<span style="color: #b8bb26;">jpeg</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">"generalfigures/pheatmapplotrld.jpg"</span>, width<span style="color: #d3869b;">=</span><span style="color: #d3869b; font-weight: bold;">800</span>, height<span style="color: #d3869b;">=</span><span style="color: #d3869b; font-weight: bold;">800</span><span style="color: #d3869b;">)</span>
<span style="color: #b8bb26;">pheatmap</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">assay</span><span style="color: #d3869b;">(</span>rld<span style="color: #d3869b;">)[</span>select,<span style="color: #d3869b;">]</span>, cluster_rows<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">FALSE</span>, show_rownames<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">TRUE</span>, cluster_cols<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">FALSE</span>, annotation_col<span style="color: #d3869b;">=</span>df<span style="color: #d3869b;">)</span>
<span style="color: #b8bb26;">dev.off</span><span style="color: #d3869b;">()</span>
</pre>
</div>
</div>
</div>
<div id="outline-container-orga303746" class="outline-4">
<h4 id="orga303746"><span class="section-number-4">5.1.10</span> Another Deseq Design</h4>
<div class="outline-text-4" id="text-5-1-10">
<div class="org-src-container">
<pre class="src src-jupyter-R">
<span style="color: #d3869b;">library</span><span style="color: #d3869b;">(</span>tidyverse<span style="color: #d3869b;">)</span>
vstarray <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">assay</span><span style="color: #d3869b;">(</span>vsd<span style="color: #d3869b;">)</span>

range <span style="color: #d3869b;">&lt;-</span> vstarray<span style="color: #d3869b;">[]</span>
names <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">rownames</span><span style="color: #d3869b;">(</span>range<span style="color: #d3869b;">)</span>
maxs <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">rowMaxs</span><span style="color: #d3869b;">(</span>range<span style="color: #d3869b;">)</span>
mins <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">rowMins</span><span style="color: #d3869b;">(</span>range<span style="color: #d3869b;">)</span>

maxs <span style="color: #d3869b;">&lt;-</span> maxs<span style="color: #d3869b;">-</span>mins
pt <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">tibble</span><span style="color: #d3869b;">(</span>names, maxs<span style="color: #d3869b;">)</span>
pt
pt <span style="color: #d3869b;">&lt;-</span> pt <span style="color: #d3869b;">%&gt;%</span> <span style="color: #b8bb26;">filter</span><span style="color: #d3869b;">(</span>maxs <span style="color: #d3869b;">&gt;</span> <span style="color: #d3869b; font-weight: bold;">10</span><span style="color: #d3869b;">)</span>
pt
<span style="color: #b8bb26;">ggplot</span><span style="color: #d3869b;">(</span>pt, <span style="color: #b8bb26;">aes</span><span style="color: #d3869b;">(</span>x<span style="color: #d3869b;">=</span>names, y<span style="color: #d3869b;">=</span> maxs<span style="color: #d3869b;">))</span> <span style="color: #d3869b;">+</span> <span style="color: #b8bb26;">geom_col</span><span style="color: #d3869b;">()</span>
descending <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">arrange</span><span style="color: #d3869b;">(</span>pt, <span style="color: #b8bb26;">desc</span><span style="color: #d3869b;">(</span>maxs<span style="color: #d3869b;">))</span>
descending <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">factor</span><span style="color: #d3869b;">(</span>descending<span style="color: #d3869b;">[</span><span style="color: #b8bb26;">"maxs"</span><span style="color: #d3869b;">])</span>
descending <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">tibble</span><span style="color: #d3869b;">(</span>row <span style="color: #d3869b;">=</span> <span style="color: #b8bb26;">seq</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">length</span><span style="color: #d3869b;">(</span>descending<span style="color: #d3869b;">))</span>, descending<span style="color: #d3869b;">)</span>
<span style="color: #b8bb26;">ggplot</span><span style="color: #d3869b;">(</span>descending, <span style="color: #b8bb26;">aes</span><span style="color: #d3869b;">(</span>x<span style="color: #d3869b;">=</span>row, y<span style="color: #d3869b;">=</span> descending<span style="color: #d3869b;">))</span> <span style="color: #d3869b;">+</span> <span style="color: #b8bb26;">geom_col</span><span style="color: #d3869b;">()</span>

<span style="color: #b8bb26;">sum</span><span style="color: #d3869b;">(</span>results$padj <span style="color: #d3869b;">&lt;</span> <span style="color: #d3869b; font-weight: bold;">0.05</span>, na.rm<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">T</span><span style="color: #d3869b;">)</span>
res <span style="color: #d3869b;">&lt;-</span> results
</pre>
</div>
</div>
</div>
<div id="outline-container-orge0622af" class="outline-4">
<h4 id="orge0622af"><span class="section-number-4">5.1.11</span> Altered Selection of genes</h4>
<div class="outline-text-4" id="text-5-1-11">
<div class="org-src-container">
<pre class="src src-jupyter-R">
Upregulated <span style="color: #d3869b;">&lt;-</span> res<span style="color: #d3869b;">[</span> <span style="color: #b8bb26;">which</span><span style="color: #d3869b;">(</span>res$padj <span style="color: #d3869b;">&lt;</span> <span style="color: #d3869b; font-weight: bold;">0.03</span> &amp; res$log2FoldChange <span style="color: #d3869b;">&gt;</span> <span style="color: #d3869b; font-weight: bold;">0</span><span style="color: #d3869b;">)</span>, <span style="color: #d3869b;">]</span>
DownRegulated <span style="color: #d3869b;">&lt;-</span> res<span style="color: #d3869b;">[</span> <span style="color: #b8bb26;">which</span><span style="color: #d3869b;">(</span>res$padj <span style="color: #d3869b;">&lt;</span> <span style="color: #d3869b; font-weight: bold;">0.03</span> &amp; res$log2FoldChange <span style="color: #d3869b;">&lt;</span> <span style="color: #d3869b; font-weight: bold;">0</span><span style="color: #d3869b;">)</span>, <span style="color: #d3869b;">]</span>
DownRegulated<span style="color: #d3869b;">[</span><span style="color: #b8bb26;">order</span><span style="color: #d3869b;">(</span>DownRegulated$padj<span style="color: #d3869b;">)</span>, <span style="color: #d3869b;">]</span>
<span style="color: #b8bb26;">head</span><span style="color: #d3869b;">(</span>DownRegulated$padj<span style="color: #d3869b;">)</span>
D <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">rownames</span><span style="color: #d3869b;">(</span>DownRegulated<span style="color: #d3869b;">)</span>
Up <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">rownames</span><span style="color: #d3869b;">(</span>Upregulated<span style="color: #d3869b;">)</span>
DR <span style="color: #d3869b;">&lt;-</span> res<span style="color: #d3869b;">[</span> <span style="color: #b8bb26;">which</span><span style="color: #d3869b;">(</span>res$padj <span style="color: #d3869b;">&lt;</span> <span style="color: #d3869b; font-weight: bold;">0.03</span><span style="color: #d3869b;">)</span>, <span style="color: #d3869b;">]</span>
DR<span style="color: #d3869b;">[</span><span style="color: #b8bb26;">order</span><span style="color: #d3869b;">(</span>DR$padj<span style="color: #d3869b;">)</span>,<span style="color: #d3869b;">]</span>
</pre>
</div>
</div>
</div>
<div id="outline-container-org5c0f453" class="outline-4">
<h4 id="org5c0f453"><span class="section-number-4">5.1.12</span> Save new selection to file</h4>
<div class="outline-text-4" id="text-5-1-12">
<div class="org-src-container">
<pre class="src src-jupyter-R"><span style="color: #b8bb26;">write.table</span><span style="color: #d3869b;">(</span>D, file<span style="color: #d3869b;">=</span><span style="color: #b8bb26;">"Genes/DownRegulated.txt"</span>, quote<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">FALSE</span>, col.names<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">FALSE</span>, row.names<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">FALSE</span>, sep<span style="color: #d3869b;">=</span><span style="color: #b8bb26;">","</span><span style="color: #d3869b;">)</span>
<span style="color: #b8bb26;">write.table</span><span style="color: #d3869b;">(</span>Up, file<span style="color: #d3869b;">=</span><span style="color: #b8bb26;">"Genes/UpRegulated.txt"</span>, quote<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">FALSE</span>, col.names<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">FALSE</span>, row.names<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">FALSE</span>, sep<span style="color: #d3869b;">=</span><span style="color: #b8bb26;">","</span><span style="color: #d3869b;">)</span>
</pre>
</div>
</div>
</div>
<div id="outline-container-org9542019" class="outline-4">
<h4 id="org9542019"><span class="section-number-4">5.1.13</span> Extra integrity plot</h4>
<div class="outline-text-4" id="text-5-1-13">
<div class="org-src-container">
<pre class="src src-jupyter-R">d <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">mcols</span><span style="color: #d3869b;">(</span>Deseq,use.names<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">TRUE</span><span style="color: #d3869b;">)</span>
a <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">data.frame</span><span style="color: #d3869b;">(</span>rownumb <span style="color: #d3869b;">=</span> <span style="color: #b8bb26;">seq</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">nrow</span><span style="color: #d3869b;">(</span>d<span style="color: #d3869b;">))</span>, d<span style="color: #d3869b;">)</span>
a <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">data.frame</span><span style="color: #d3869b;">(</span>rownumb <span style="color: #d3869b;">=</span> <span style="color: #b8bb26;">seq</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">nrow</span><span style="color: #d3869b;">(</span>d<span style="color: #d3869b;">))</span>, basemean <span style="color: #d3869b;">=</span> d<span style="color: #d3869b;">[</span>,<span style="color: #d3869b; font-weight: bold;">1</span><span style="color: #d3869b;">])</span>
<span style="color: #d3869b;">library</span><span style="color: #d3869b;">(</span>ggplot2<span style="color: #d3869b;">)</span>
<span style="color: #b8bb26;">ggplot</span><span style="color: #d3869b;">(</span>a, <span style="color: #b8bb26;">aes</span><span style="color: #d3869b;">(</span>x<span style="color: #d3869b;">=</span>rownumb, y<span style="color: #d3869b;">=</span>dispGeneEst<span style="color: #d3869b;">))</span> <span style="color: #d3869b;">+</span> <span style="color: #b8bb26;">geom_point</span><span style="color: #d3869b;">()</span>
</pre>
</div>
</div>
</div>
<div id="outline-container-org770eb89" class="outline-4">
<h4 id="org770eb89"><span class="section-number-4">5.1.14</span> <span class="todo TODO">TODO</span> Extra EdgeR for venn diagrams</h4>
<div class="outline-text-4" id="text-5-1-14">
<p>
This is mostly contained in the EdgeR section.
</p>
<div class="org-src-container">
<pre class="src src-jupyter-R">
<span style="color: #928374;">#</span><span style="color: #928374;">! EdgeR section ----</span>
<span style="color: #d3869b;">library</span><span style="color: #d3869b;">(</span>edgeR<span style="color: #d3869b;">)</span>
<span style="color: #928374;"># </span><span style="color: #928374;">EdgR dataset from tximport</span>
counts <span style="color: #d3869b;">&lt;-</span> txi$counts
length <span style="color: #d3869b;">&lt;-</span> txi$length
normMat <span style="color: #d3869b;">&lt;-</span> length <span style="color: #d3869b;">/</span> <span style="color: #b8bb26;">exp</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">rowMeans</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">log</span><span style="color: #d3869b;">(</span>length<span style="color: #d3869b;">)))</span>
normCts <span style="color: #d3869b;">&lt;-</span> counts <span style="color: #d3869b;">/</span> normMat
eff.lib <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">calcNormFactors</span><span style="color: #d3869b;">(</span>normCts<span style="color: #d3869b;">)</span> * <span style="color: #b8bb26;">colSums</span><span style="color: #d3869b;">(</span>normCts<span style="color: #d3869b;">)</span>
normMat <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">sweep</span><span style="color: #d3869b;">(</span>normMat, <span style="color: #d3869b; font-weight: bold;">2</span>, eff.lib, <span style="color: #b8bb26;">"*"</span><span style="color: #d3869b;">)</span>
normMat <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">log</span><span style="color: #d3869b;">(</span>normMat<span style="color: #d3869b;">)</span>
y <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">DGEList</span><span style="color: #d3869b;">(</span>counts<span style="color: #d3869b;">)</span>
y <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">scaleOffset</span><span style="color: #d3869b;">(</span>y, normMat<span style="color: #d3869b;">)</span>



<span style="color: #928374;"># </span><span style="color: #928374;">EdgR heatmap</span>
logcpm <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">cpm</span><span style="color: #d3869b;">(</span>EdgeR<span style="color: #d3869b;">)</span>
meta <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">read.csv</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">"metadata.csv"</span>, sep<span style="color: #d3869b;">=</span> <span style="color: #b8bb26;">","</span><span style="color: #d3869b;">)</span>
pest <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">factor</span><span style="color: #d3869b;">(</span>meta$pesticides, levels<span style="color: #d3869b;">=</span><span style="color: #b8bb26;">c</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">"perm"</span>, <span style="color: #b8bb26;">"deet"</span>, <span style="color: #b8bb26;">"control"</span><span style="color: #d3869b;">))</span>
part <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">factor</span><span style="color: #d3869b;">(</span>meta$part, levels<span style="color: #d3869b;">=</span><span style="color: #b8bb26;">c</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">"leg"</span>, <span style="color: #b8bb26;">"body"</span><span style="color: #d3869b;">))</span>
design <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">model.matrix</span><span style="color: #d3869b;">(</span>~<span style="color: #d3869b; font-weight: bold;">0</span> <span style="color: #d3869b;">+</span> pest <span style="color: #d3869b;">+</span> part<span style="color: #d3869b;">)</span>
design
keep <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">filterByExpr</span><span style="color: #d3869b;">(</span>y, design<span style="color: #d3869b;">)</span>
EdgeR <span style="color: #d3869b;">&lt;-</span> y<span style="color: #d3869b;">[</span>keep, <span style="color: #d3869b;">]</span>
<span style="color: #b8bb26;">estimateDisp</span><span style="color: #d3869b;">(</span>EdgeR<span style="color: #d3869b;">)</span>
y <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">estimateGLMCommonDisp</span><span style="color: #d3869b;">(</span>EdgeR, design<span style="color: #d3869b;">)</span>
y <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">estimateGLMTrendedDisp</span><span style="color: #d3869b;">(</span>y, design<span style="color: #d3869b;">)</span>
y <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">estimateGLMTagwiseDisp</span><span style="color: #d3869b;">(</span>y, design<span style="color: #d3869b;">)</span>
est <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">estimateDisp</span><span style="color: #d3869b;">(</span>EdgeR, design, robust<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">TRUE</span><span style="color: #d3869b;">)</span>
design
fit <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">glmQLFit</span><span style="color: #d3869b;">(</span>y, design<span style="color: #d3869b;">)</span>
lrt <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">glmLRT</span><span style="color: #d3869b;">(</span>fit, coef<span style="color: #d3869b;">=</span><span style="color: #d3869b; font-weight: bold;">1</span>:<span style="color: #d3869b; font-weight: bold;">4</span><span style="color: #d3869b;">)</span>
test <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">glmQLFTest</span><span style="color: #d3869b;">(</span>fit, coef<span style="color: #d3869b;">=</span><span style="color: #d3869b; font-weight: bold;">1</span>:<span style="color: #d3869b; font-weight: bold;">4</span><span style="color: #d3869b;">)</span>
lrc$PValue
lrt

test$logFC.pestperm <span style="color: #d3869b;">-</span> test$logFC.pestdeet

<span style="color: #b8bb26;">ggplot</span><span style="color: #d3869b;">(</span>test$table, <span style="color: #b8bb26;">aes</span><span style="color: #d3869b;">(</span>y<span style="color: #d3869b;">=</span>logFC.partleg, x<span style="color: #d3869b;">=</span><span style="color: #b8bb26;">seq</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">nrow</span><span style="color: #d3869b;">(</span>test<span style="color: #d3869b;">))))</span> <span style="color: #d3869b;">+</span> <span style="color: #b8bb26;">geom_hex</span><span style="color: #d3869b;">()</span>
<span style="color: #928374;">#</span><span style="color: #928374;">additional analysis----</span>
table <span style="color: #d3869b;">&lt;-</span> test$table
body <span style="color: #d3869b;">&lt;-</span> table<span style="color: #d3869b;">[</span>table$logFC.partbody <span style="color: #d3869b;">&gt;</span> <span style="color: #d3869b; font-weight: bold;">2</span>, <span style="color: #d3869b;">]</span>
leg <span style="color: #d3869b;">&lt;-</span> table<span style="color: #d3869b;">[</span>table$logFC.partbody <span style="color: #d3869b;">&lt;</span> <span style="color: #d3869b;">-</span><span style="color: #d3869b; font-weight: bold;">2</span>, <span style="color: #d3869b;">]</span>
table <span style="color: #d3869b;">&lt;-</span> body
diff <span style="color: #d3869b;">&lt;-</span> table$logFC.pestcontrol <span style="color: #d3869b;">-</span> table$logFC.pestdeet
<span style="color: #d3869b;">library</span><span style="color: #d3869b;">(</span>tidyverse<span style="color: #d3869b;">)</span>
data <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">tibble</span><span style="color: #d3869b;">(</span>data <span style="color: #d3869b;">=</span> diff, rows <span style="color: #d3869b;">=</span> <span style="color: #b8bb26;">seq</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">length</span><span style="color: #d3869b;">(</span>diff<span style="color: #d3869b;">)))</span>
<span style="color: #b8bb26;">ggplot</span><span style="color: #d3869b;">(</span>data, <span style="color: #b8bb26;">aes</span><span style="color: #d3869b;">(</span>x<span style="color: #d3869b;">=</span>rows, y<span style="color: #d3869b;">=</span>data<span style="color: #d3869b;">))</span> <span style="color: #d3869b;">+</span> <span style="color: #b8bb26;">geom_point</span><span style="color: #d3869b;">()</span>
table$diff <span style="color: #d3869b;">&lt;-</span> diff
cont <span style="color: #d3869b;">&lt;-</span> table<span style="color: #d3869b;">[</span>table$diff <span style="color: #d3869b;">&gt;</span> <span style="color: #d3869b; font-weight: bold;">2</span>, <span style="color: #d3869b;">]</span>
deet <span style="color: #d3869b;">&lt;-</span> table<span style="color: #d3869b;">[</span>table$diff <span style="color: #d3869b;">&lt;</span> <span style="color: #d3869b;">-</span><span style="color: #d3869b; font-weight: bold;">2</span>, <span style="color: #d3869b;">]</span>
diff2 <span style="color: #d3869b;">&lt;-</span> table$logFC.pestcontrol <span style="color: #d3869b;">-</span> table$logFC.pestperm
table$diff2 <span style="color: #d3869b;">&lt;-</span> diff2
cont2 <span style="color: #d3869b;">&lt;-</span> table<span style="color: #d3869b;">[</span>table$diff2 <span style="color: #d3869b;">&gt;</span> <span style="color: #d3869b; font-weight: bold;">2</span>, <span style="color: #d3869b;">]</span>
perm <span style="color: #d3869b;">&lt;-</span> table<span style="color: #d3869b;">[</span>table$diff2 <span style="color: #d3869b;">&lt;</span> <span style="color: #d3869b;">-</span><span style="color: #d3869b; font-weight: bold;">2</span>, <span style="color: #d3869b;">]</span>



<span style="color: #d3869b;">library</span><span style="color: #d3869b;">(</span>VennDiagram<span style="color: #d3869b;">)</span>
futile.logger::<span style="color: #b8bb26;">flog.threshold</span><span style="color: #d3869b;">(</span>futile.logger::ERROR, name <span style="color: #d3869b;">=</span> <span style="color: #b8bb26;">"VennDiagramLogger"</span><span style="color: #d3869b;">)</span>
Legcont1 <span style="color: #d3869b;">&lt;-</span> cont
Legcont2 <span style="color: #d3869b;">&lt;-</span> cont2
deet1 <span style="color: #d3869b;">&lt;-</span> deet
perm1 <span style="color: #d3869b;">&lt;-</span> perm
d <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">data.frame</span><span style="color: #d3869b;">(</span>Pest <span style="color: #d3869b;">=</span> <span style="color: #b8bb26;">c</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">rep</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">c</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">"DEET"</span>, <span style="color: #b8bb26;">"DEET"</span>, <span style="color: #b8bb26;">"PERM"</span>, <span style="color: #b8bb26;">"PERM"</span><span style="color: #d3869b;">)))</span>,
                Part <span style="color: #d3869b;">=</span> <span style="color: #b8bb26;">c</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">rep</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">"Body"</span>, <span style="color: #d3869b; font-weight: bold;">4</span><span style="color: #d3869b;">)</span>, <span style="color: #b8bb26;">rep</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">"Leg"</span>, <span style="color: #d3869b; font-weight: bold;">4</span><span style="color: #d3869b;">))</span>,
                Names <span style="color: #d3869b;">=</span> <span style="color: #b8bb26;">c</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">rep</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">c</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">"Up Regulated"</span>, <span style="color: #b8bb26;">"Down Regulated"</span><span style="color: #d3869b;">)</span>, <span style="color: #d3869b; font-weight: bold;">4</span><span style="color: #d3869b;">))</span>,
                GeneCount <span style="color: #d3869b;">=</span> <span style="color: #b8bb26;">c</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">nrow</span><span style="color: #d3869b;">(</span>deet<span style="color: #d3869b;">)</span>, <span style="color: #b8bb26;">nrow</span><span style="color: #d3869b;">(</span>cont<span style="color: #d3869b;">)</span>, <span style="color: #b8bb26;">nrow</span><span style="color: #d3869b;">(</span>perm<span style="color: #d3869b;">)</span>, <span style="color: #b8bb26;">nrow</span><span style="color: #d3869b;">(</span>cont2<span style="color: #d3869b;">)</span>, <span style="color: #b8bb26;">nrow</span><span style="color: #d3869b;">(</span>deet1<span style="color: #d3869b;">)</span>, <span style="color: #b8bb26;">nrow</span><span style="color: #d3869b;">(</span>Legcont1<span style="color: #d3869b;">)</span>, <span style="color: #b8bb26;">nrow</span><span style="color: #d3869b;">(</span>perm1<span style="color: #d3869b;">)</span>, <span style="color: #b8bb26;">nrow</span><span style="color: #d3869b;">(</span>Legcont2<span style="color: #d3869b;">)))</span>

<span style="color: #b8bb26;">write.table</span><span style="color: #d3869b;">(</span>d, file<span style="color: #d3869b;">=</span><span style="color: #b8bb26;">"Genes/regulation.csv"</span>, quote<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">FALSE</span>, col.names<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">TRUE</span>, row.names<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">FALSE</span>, sep<span style="color: #d3869b;">=</span><span style="color: #b8bb26;">","</span><span style="color: #d3869b;">)</span>
<span style="color: #b8bb26;">write.table</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">rownames</span><span style="color: #d3869b;">(</span>perm<span style="color: #d3869b;">)</span>, file<span style="color: #d3869b;">=</span><span style="color: #b8bb26;">"Genes/perm.csv"</span>, quote<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">FALSE</span>, col.names<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">FALSE</span>, row.names<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">FALSE</span>, sep<span style="color: #d3869b;">=</span><span style="color: #b8bb26;">","</span><span style="color: #d3869b;">)</span>

pl <span style="color: #b8bb26;">ggplot</span><span style="color: #d3869b;">(</span>d, <span style="color: #b8bb26;">aes</span><span style="color: #d3869b;">(</span>x<span style="color: #d3869b;">=</span>Names, y <span style="color: #d3869b;">=</span> GeneCount, fill<span style="color: #d3869b;">=</span>Names<span style="color: #d3869b;">))</span> <span style="color: #d3869b;">+</span> <span style="color: #b8bb26;">geom_col</span><span style="color: #d3869b;">()</span>
<span style="color: #b8bb26;">venn.diagram</span><span style="color: #d3869b;">(</span>
  x <span style="color: #d3869b;">=</span> <span style="color: #b8bb26;">list</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">rownames</span><span style="color: #d3869b;">(</span>deet<span style="color: #d3869b;">)</span>, <span style="color: #b8bb26;">rownames</span><span style="color: #d3869b;">(</span>cont<span style="color: #d3869b;">))</span>,
  category.names <span style="color: #d3869b;">=</span> <span style="color: #b8bb26;">c</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">"DEET"</span> , <span style="color: #b8bb26;">"Control"</span><span style="color: #d3869b;">)</span>,
  filename <span style="color: #d3869b;">=</span> <span style="color: #b8bb26;">"Deetvcontorlvleg.png"</span>,
  compression <span style="color: #d3869b;">=</span> <span style="color: #b8bb26;">"lzw"</span>,


  lwd <span style="color: #d3869b;">=</span> <span style="color: #d3869b; font-weight: bold;">2</span>,
  lty <span style="color: #d3869b;">=</span> <span style="color: #b8bb26;">'blank'</span>,
  fill <span style="color: #d3869b;">=</span> <span style="color: #b8bb26;">c</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">"red"</span>, <span style="color: #b8bb26;">"blue"</span><span style="color: #d3869b;">)</span>,
<span style="color: #d3869b;">)</span>
<span style="color: #b8bb26;">grid.draw</span><span style="color: #d3869b;">(</span>tmp<span style="color: #d3869b;">)</span>

<span style="color: #b8bb26;">write.table</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">rownames</span><span style="color: #d3869b;">(</span>perm<span style="color: #d3869b;">)</span>, file<span style="color: #d3869b;">=</span><span style="color: #b8bb26;">"Genes/perm.csv"</span>, quote<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">FALSE</span>, col.names<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">FALSE</span>, row.names<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">FALSE</span>, sep<span style="color: #d3869b;">=</span><span style="color: #b8bb26;">","</span><span style="color: #d3869b;">)</span>
<span style="color: #b8bb26;">write.table</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">rownames</span><span style="color: #d3869b;">(</span>deet<span style="color: #d3869b;">)</span>, file<span style="color: #d3869b;">=</span><span style="color: #b8bb26;">"Genes/deet.csv"</span>, quote<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">FALSE</span>, col.names<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">FALSE</span>, row.names<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">FALSE</span>, sep<span style="color: #d3869b;">=</span><span style="color: #b8bb26;">","</span><span style="color: #d3869b;">)</span>
<span style="color: #b8bb26;">write.table</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">rownames</span><span style="color: #d3869b;">(</span>cont<span style="color: #d3869b;">)</span>, file<span style="color: #d3869b;">=</span><span style="color: #b8bb26;">"Genes/deetcont.csv"</span>, quote<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">FALSE</span>, col.names<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">FALSE</span>, row.names<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">FALSE</span>, sep<span style="color: #d3869b;">=</span><span style="color: #b8bb26;">","</span><span style="color: #d3869b;">)</span>
<span style="color: #b8bb26;">write.table</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">rownames</span><span style="color: #d3869b;">(</span>cont2<span style="color: #d3869b;">)</span>, file<span style="color: #d3869b;">=</span><span style="color: #b8bb26;">"Genes/permcont.csv"</span>, quote<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">FALSE</span>, col.names<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">FALSE</span>, row.names<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">FALSE</span>, sep<span style="color: #d3869b;">=</span><span style="color: #b8bb26;">","</span><span style="color: #d3869b;">)</span>

<span style="color: #b8bb26;">ggplot</span><span style="color: #d3869b;">(</span>contdata, <span style="color: #b8bb26;">aes</span><span style="color: #d3869b;">(</span>x<span style="color: #d3869b;">=</span>rows, y<span style="color: #d3869b;">=</span>data<span style="color: #d3869b;">))</span> <span style="color: #d3869b;">+</span> <span style="color: #b8bb26;">geom_point</span><span style="color: #d3869b;">()</span>

<span style="color: #b8bb26;">rownames</span><span style="color: #d3869b;">(</span>cont<span style="color: #d3869b;">)</span>
<span style="color: #928374;"># </span><span style="color: #928374;">EdgeR plots ----</span>
<span style="color: #b8bb26;">jpeg</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">"generalfigures/BCVplot_edgeR.jpg"</span><span style="color: #d3869b;">)</span>
<span style="color: #b8bb26;">plotBCV</span><span style="color: #d3869b;">(</span>est<span style="color: #d3869b;">)</span>
<span style="color: #b8bb26;">dev.off</span><span style="color: #d3869b;">()</span>

<span style="color: #b8bb26;">jpeg</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">"generalfigures/QLDISP_edgeR.jpg"</span><span style="color: #d3869b;">)</span>
<span style="color: #b8bb26;">plotQLDisp</span><span style="color: #d3869b;">(</span>fit<span style="color: #d3869b;">)</span>
<span style="color: #b8bb26;">dev.off</span><span style="color: #d3869b;">()</span>

<span style="color: #b8bb26;">jpeg</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">"generalfigures/MDS_edgeR.jpg"</span><span style="color: #d3869b;">)</span>
<span style="color: #b8bb26;">plotMDS</span><span style="color: #d3869b;">(</span>logcpm<span style="color: #d3869b;">)</span>
<span style="color: #b8bb26;">dev.off</span><span style="color: #d3869b;">()</span>


<span style="color: #928374;"># </span><span style="color: #928374;">jpeg("generalfigur&gt;es/SpliceDGE_edgeR.jpg")</span>
<span style="color: #928374;"># </span><span style="color: #928374;">plotSpliceDGE(logcpm)</span>
<span style="color: #928374;"># </span><span style="color: #928374;">dev.off()</span>

<span style="color: #928374;"># </span><span style="color: #928374;">end ----</span>
<span style="color: #d3869b;">library</span><span style="color: #d3869b;">(</span>matrixStats<span style="color: #d3869b;">)</span>
stats <span style="color: #d3869b;">&lt;-</span> txi$abundance
tpm <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">transform</span><span style="color: #d3869b;">(</span>stats, sd<span style="color: #d3869b;">=</span><span style="color: #b8bb26;">rowSds</span><span style="color: #d3869b;">(</span>stats<span style="color: #d3869b;">)</span>, avg<span style="color: #d3869b;">=</span><span style="color: #b8bb26;">rowMeans</span><span style="color: #d3869b;">(</span>stats<span style="color: #d3869b;">)</span>, median <span style="color: #d3869b;">=</span> <span style="color: #b8bb26;">rowMedians</span><span style="color: #d3869b;">(</span>stats<span style="color: #d3869b;">))</span>

dge <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">DGEList</span><span style="color: #d3869b;">(</span>txi$counts<span style="color: #d3869b;">)</span>
cpm <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">cpm</span><span style="color: #d3869b;">(</span>dge, log<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">TRUE</span><span style="color: #d3869b;">)</span>
<span style="color: #b8bb26;">head</span><span style="color: #d3869b;">(</span>cpm<span style="color: #d3869b;">)</span>
<span style="color: #b8bb26;">data.frame</span><span style="color: #d3869b;">(</span>cpm<span style="color: #d3869b;">)</span>
cpmtibble <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">as_tibble</span><span style="color: #d3869b;">(</span>cpm, rownames<span style="color: #d3869b;">=</span><span style="color: #b8bb26;">"geneID"</span><span style="color: #d3869b;">)</span>
cpmpivot <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">pivot_longer</span><span style="color: #d3869b;">(</span>cpmtibble, cols<span style="color: #d3869b;">=</span><span style="color: #b8bb26;">seq</span><span style="color: #d3869b;">(</span><span style="color: #d3869b; font-weight: bold;">2</span>, <span style="color: #b8bb26;">ncol</span><span style="color: #d3869b;">(</span>cpmtibble<span style="color: #d3869b;">))</span>, names_to <span style="color: #d3869b;">=</span> <span style="color: #b8bb26;">"samples"</span>, values_to <span style="color: #d3869b;">=</span> <span style="color: #b8bb26;">"expression"</span><span style="color: #d3869b;">)</span>



<span style="color: #b8bb26;">ggplot</span><span style="color: #d3869b;">(</span>tpm<span style="color: #d3869b;">)</span> <span style="color: #d3869b;">+</span>
  <span style="color: #b8bb26;">aes</span><span style="color: #d3869b;">(</span>x <span style="color: #d3869b;">=</span> sd, y <span style="color: #d3869b;">=</span> median<span style="color: #d3869b;">)</span> <span style="color: #d3869b;">+</span>
  <span style="color: #b8bb26;">geom_hex</span><span style="color: #d3869b;">()</span>

<span style="color: #b8bb26;">ggplot</span><span style="color: #d3869b;">(</span>tpm<span style="color: #d3869b;">)</span> <span style="color: #d3869b;">+</span>
  <span style="color: #b8bb26;">aes</span><span style="color: #d3869b;">(</span>x <span style="color: #d3869b;">=</span> sd, y <span style="color: #d3869b;">=</span> avg<span style="color: #d3869b;">)</span> <span style="color: #d3869b;">+</span>
  <span style="color: #b8bb26;">geom_point</span><span style="color: #d3869b;">(</span>shape<span style="color: #d3869b;">=</span><span style="color: #d3869b; font-weight: bold;">16</span>, size<span style="color: #d3869b;">=</span><span style="color: #d3869b; font-weight: bold;">2</span><span style="color: #d3869b;">)</span> <span style="color: #d3869b;">+</span>
  <span style="color: #b8bb26;">geom_smooth</span><span style="color: #d3869b;">(</span>method<span style="color: #d3869b;">=</span>lm<span style="color: #d3869b;">)</span> <span style="color: #d3869b;">+</span>
  <span style="color: #b8bb26;">geom_hex</span><span style="color: #d3869b;">()</span> <span style="color: #d3869b;">+</span>
  <span style="color: #b8bb26;">labs</span><span style="color: #d3869b;">(</span>y<span style="color: #d3869b;">=</span><span style="color: #b8bb26;">"Median"</span>, x <span style="color: #d3869b;">=</span> <span style="color: #b8bb26;">"Standard deviation"</span>, title<span style="color: #d3869b;">=</span><span style="color: #b8bb26;">"Transcripts per million (TPM)"</span>, subtitle<span style="color: #d3869b;">=</span><span style="color: #b8bb26;">"unfiltered, non-normalized data"</span><span style="color: #d3869b;">)</span> <span style="color: #d3869b;">+</span>
  <span style="color: #b8bb26;">theme_classic</span><span style="color: #d3869b;">()</span> <span style="color: #d3869b;">+</span>
  <span style="color: #b8bb26;">theme_dark</span><span style="color: #d3869b;">()</span> <span style="color: #d3869b;">+</span>
  <span style="color: #b8bb26;">theme_bw</span><span style="color: #d3869b;">()</span>

<span style="color: #b8bb26;">ggplot</span><span style="color: #d3869b;">(</span>cpmpivot<span style="color: #d3869b;">)</span> <span style="color: #d3869b;">+</span>
  <span style="color: #b8bb26;">aes</span><span style="color: #d3869b;">(</span>x<span style="color: #d3869b;">=</span>samples, y<span style="color: #d3869b;">=</span>expression, fill<span style="color: #d3869b;">=</span>samples<span style="color: #d3869b;">)</span> <span style="color: #d3869b;">+</span>
  <span style="color: #b8bb26;">geom_violin</span><span style="color: #d3869b;">(</span>trim <span style="color: #d3869b;">=</span> <span style="color: #fabd2f;">FALSE</span>, show.legend <span style="color: #d3869b;">=</span> <span style="color: #fabd2f;">FALSE</span><span style="color: #d3869b;">)</span> <span style="color: #d3869b;">+</span>
  <span style="color: #b8bb26;">stat_summary</span><span style="color: #d3869b;">(</span>fun <span style="color: #d3869b;">=</span> <span style="color: #b8bb26;">"median"</span>,geom <span style="color: #d3869b;">=</span> <span style="color: #b8bb26;">"point"</span>,shape <span style="color: #d3869b;">=</span> <span style="color: #d3869b; font-weight: bold;">95</span>,size <span style="color: #d3869b;">=</span> <span style="color: #d3869b; font-weight: bold;">10</span>,color <span style="color: #d3869b;">=</span> <span style="color: #b8bb26;">"black"</span>,show.legend <span style="color: #d3869b;">=</span> <span style="color: #fabd2f;">FALSE</span><span style="color: #d3869b;">)</span> <span style="color: #d3869b;">+</span>
  <span style="color: #b8bb26;">labs</span><span style="color: #d3869b;">(</span>y<span style="color: #d3869b;">=</span><span style="color: #b8bb26;">"log2 expression"</span>, x <span style="color: #d3869b;">=</span> <span style="color: #b8bb26;">"sample"</span>,title<span style="color: #d3869b;">=</span><span style="color: #b8bb26;">"Log2 Counts per Million (CPM)"</span>,subtitle<span style="color: #d3869b;">=</span><span style="color: #b8bb26;">"unfiltered, non-normalized"</span>,caption<span style="color: #d3869b;">=</span><span style="color: #b8bb26;">paste0</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">"produced on "</span>, <span style="color: #b8bb26;">Sys.time</span><span style="color: #d3869b;">()))</span> <span style="color: #d3869b;">+</span>
  <span style="color: #b8bb26;">theme_bw</span><span style="color: #d3869b;">()</span>

keepers <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">rowSums</span><span style="color: #d3869b;">(</span>cpm <span style="color: #d3869b;">&gt;</span> <span style="color: #d3869b; font-weight: bold;">1</span><span style="color: #d3869b;">)</span><span style="color: #d3869b;">&gt;=</span><span style="color: #d3869b; font-weight: bold;">3</span>
<span style="color: #b8bb26;">table</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">rowSums</span><span style="color: #d3869b;">(</span>dge$counts<span style="color: #d3869b;">==</span><span style="color: #d3869b; font-weight: bold;">0</span><span style="color: #d3869b;">)</span><span style="color: #d3869b;">==</span><span style="color: #d3869b; font-weight: bold;">10</span><span style="color: #d3869b;">)</span>


<span style="color: #928374;"># </span><span style="color: #928374;">venndiagrams ----</span>

<span style="color: #d3869b;">library</span><span style="color: #d3869b;">(</span>VennDiagram<span style="color: #d3869b;">)</span>


bodygenes <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">names</span><span style="color: #d3869b;">(</span>Deseq<span style="color: #d3869b;">)</span>
bodygenes <span style="color: #d3869b;">&lt;-</span> bodygenes<span style="color: #d3869b;">[</span>!bodygenes <span style="color: #d3869b;">%in%</span> <span style="color: #b8bb26;">rownames</span><span style="color: #d3869b;">(</span>Leg<span style="color: #d3869b;">)]</span>
leggenes <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">names</span><span style="color: #d3869b;">(</span>Deseq<span style="color: #d3869b;">)</span>
leggenes <span style="color: #d3869b;">&lt;-</span> leggenes<span style="color: #d3869b;">[</span>!leggenes <span style="color: #d3869b;">%in%</span> <span style="color: #b8bb26;">rownames</span><span style="color: #d3869b;">(</span>Body<span style="color: #d3869b;">)]</span>
<span style="color: #b8bb26;">names</span><span style="color: #d3869b;">(</span>Deseq<span style="color: #d3869b;">)</span> <span style="color: #d3869b;">-</span> <span style="color: #b8bb26;">rownames</span><span style="color: #d3869b;">(</span>Body<span style="color: #d3869b;">)</span>

<span style="color: #928374;">#</span>
bodyup <span style="color: #d3869b;">&lt;-</span> table<span style="color: #d3869b;">[</span>table<span style="color: #d3869b;">[</span><span style="color: #b8bb26;">"partleg"</span><span style="color: #d3869b;">]</span> <span style="color: #d3869b;">&gt;</span> <span style="color: #d3869b; font-weight: bold;">0</span><span style="color: #d3869b;">]</span>


<span style="color: #928374;"># </span><span style="color: #928374;">Venn Diagram 1----</span>
futile.logger::<span style="color: #b8bb26;">flog.threshold</span><span style="color: #d3869b;">(</span>futile.logger::ERROR, name <span style="color: #d3869b;">=</span> <span style="color: #b8bb26;">"VennDiagramLogger"</span><span style="color: #d3869b;">)</span>
<span style="color: #b8bb26;">venn.diagram</span><span style="color: #d3869b;">(</span>
  x <span style="color: #d3869b;">=</span> <span style="color: #b8bb26;">list</span><span style="color: #d3869b;">(</span>bodygenes, leggenes<span style="color: #d3869b;">)</span>,
  category.names <span style="color: #d3869b;">=</span> <span style="color: #b8bb26;">c</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">"Body"</span> , <span style="color: #b8bb26;">"Leg"</span><span style="color: #d3869b;">)</span>,

  filename <span style="color: #d3869b;">=</span> <span style="color: #b8bb26;">'14_venn_diagramm.png'</span>,
  output<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">FALSE</span>,
  imagetype<span style="color: #d3869b;">=</span><span style="color: #b8bb26;">"png"</span>,
  compression<span style="color: #d3869b;">=</span><span style="color: #b8bb26;">"lzw"</span>,


  lwd <span style="color: #d3869b;">=</span> <span style="color: #d3869b; font-weight: bold;">2</span>,
  lty <span style="color: #d3869b;">=</span> <span style="color: #b8bb26;">'blank'</span>,
  fill <span style="color: #d3869b;">=</span> <span style="color: #b8bb26;">c</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">"red"</span>, <span style="color: #b8bb26;">"blue"</span><span style="color: #d3869b;">)</span>,
<span style="color: #d3869b;">)</span>


</pre>
</div>
</div>
</div>
</div>
<div id="outline-container-orgec21115" class="outline-3">
<h3 id="orgec21115"><span class="section-number-3">5.2</span> EdgeR</h3>
<div class="outline-text-3" id="text-5-2">
</div>
<div id="outline-container-org25021c7" class="outline-4">
<h4 id="org25021c7"><span class="section-number-4">5.2.1</span> Initial settings</h4>
<div class="outline-text-4" id="text-5-2-1">
<p>
Directory and function setup, configure this when running the script. This is an attempt at simplifying the analysis process in R to a point that is hopefully nonspecific to deseq2 or edgeR. Enter the directory where the RNA-seq data is located, the folder containing the quant data, and the type of quant tool used (fishtype). The script should take care of the rest.
</p>
<div class="org-src-container">
<pre class="src src-jupyter-R"><span style="color: #d3869b;">setwd</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">"/home/david/Documents/BenoitLab/RNA-seq/"</span><span style="color: #d3869b;">)</span>
<span style="color: #d3869b;">source</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">"scripts/functions.r"</span><span style="color: #d3869b;">)</span>
directory <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">"/home/david/Documents/BenoitLab/R/largeRNASEQ/"</span>
quantdatafoldername <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">"CorrectedQuantData"</span>
fishtype <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">"sailfish"</span>
</pre>
</div>
</div>
</div>

<div id="outline-container-org934271c" class="outline-4">
<h4 id="org934271c"><span class="section-number-4">5.2.2</span> Do the initial processing of the data</h4>
<div class="outline-text-4" id="text-5-2-2">
<p>
Imports the data and runs it through initial edgeR analysis using the functions created above. The functions do the initial manipulation of the data as detailed in the edgeR paper. This outputs the objects we will trim based on expression.
</p>
<div class="org-src-container">
<pre class="src src-jupyter-R">a <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">txi</span><span style="color: #d3869b;">(</span>directory, quantdatafoldername, <span style="color: #b8bb26;">"metadata.csv"</span>, fishtype<span style="color: #d3869b;">)</span>
<span style="color: #928374;">#</span><span style="color: #928374;">De &lt;- initialDeseq(a[[1]], a[[2]])</span>
Ed <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">initialEdgeR</span><span style="color: #d3869b;">(</span>a<span style="color: #d3869b;">[[</span><span style="color: #d3869b; font-weight: bold;">1</span><span style="color: #d3869b;">]]</span>, a<span style="color: #d3869b;">[[</span><span style="color: #d3869b; font-weight: bold;">2</span><span style="color: #d3869b;">]])</span>
Eest <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">estEdgeR</span><span style="color: #d3869b;">(</span>Ed<span style="color: #d3869b;">[[</span><span style="color: #d3869b; font-weight: bold;">1</span><span style="color: #d3869b;">]]</span>, Ed<span style="color: #d3869b;">[[</span><span style="color: #d3869b; font-weight: bold;">2</span><span style="color: #d3869b;">]])</span>
lrt <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">glmLRT</span><span style="color: #d3869b;">(</span>Eest, coef<span style="color: #d3869b;">=</span><span style="color: #d3869b; font-weight: bold;">1</span>:<span style="color: #d3869b; font-weight: bold;">4</span><span style="color: #d3869b;">)</span>
test <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">glmQLFTest</span><span style="color: #d3869b;">(</span>Eest, coef<span style="color: #d3869b;">=</span><span style="color: #d3869b; font-weight: bold;">1</span>:<span style="color: #d3869b; font-weight: bold;">4</span><span style="color: #d3869b;">)</span>
</pre>
</div>
<pre class="example" id="org536bed2">
reading in files with read_tsv

1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17


summarizing abundance

summarizing counts

summarizing length

Loading required package: limma
</pre>
</div>
</div>
<div id="outline-container-orgd35a1e4" class="outline-4">
<h4 id="orgd35a1e4"><span class="section-number-4">5.2.3</span> Partion the data into body and leg tables</h4>
<div class="outline-text-4" id="text-5-2-3">
<p>
The Table object from the test object is separated into a body and a leg section, where the genes in the leg section are down regulated in the body and vice versa.
</p>
<div class="org-src-container">
<pre class="src src-jupyter-R">table <span style="color: #d3869b;">&lt;-</span> test$table
pSelectedTable <span style="color: #d3869b;">&lt;-</span> table<span style="color: #d3869b;">[</span>table$PValue <span style="color: #d3869b;">&lt;</span> <span style="color: #d3869b; font-weight: bold;">0.01</span>, <span style="color: #d3869b;">]</span>
body <span style="color: #d3869b;">&lt;-</span> pSelectedTable<span style="color: #d3869b;">[</span>pSelectedTable$logFC.partbody <span style="color: #d3869b;">&gt;</span> <span style="color: #d3869b; font-weight: bold;">2</span>, <span style="color: #d3869b;">]</span>
leg <span style="color: #d3869b;">&lt;-</span> pSelectedTable<span style="color: #d3869b;">[</span>pSelectedTable$logFC.partbody <span style="color: #d3869b;">&lt;</span> <span style="color: #d3869b;">-</span><span style="color: #d3869b; font-weight: bold;">2</span>, <span style="color: #d3869b;">]</span>
<span style="color: #b8bb26;">head</span><span style="color: #d3869b;">(</span>table<span style="color: #d3869b;">)</span>
</pre>
</div>
</div>
</div>
<div id="outline-container-org492add9" class="outline-4">
<h4 id="org492add9"><span class="section-number-4">5.2.4</span> Save the data tables to a file</h4>
<div class="outline-text-4" id="text-5-2-4">
<p>
Configure this to fit your computer. This section saves csv files with the expression data from the tables in the previous section.
</p>

<div class="org-src-container">
<pre class="src src-jupyter-R"><span style="color: #d3869b;">setwd</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">"/home/david/Documents/BenoitLab/RNA-seq/"</span><span style="color: #d3869b;">)</span>
<span style="color: #b8bb26;">write.table</span><span style="color: #d3869b;">(</span>table, file<span style="color: #d3869b;">=</span><span style="color: #b8bb26;">"edgeR-Genes/TotalExpr.csv"</span>, quote<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">FALSE</span>, col.names<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">TRUE</span>, row.names<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">TRUE</span>, sep<span style="color: #d3869b;">=</span><span style="color: #b8bb26;">","</span><span style="color: #d3869b;">)</span>
<span style="color: #b8bb26;">write.table</span><span style="color: #d3869b;">(</span>leg, file<span style="color: #d3869b;">=</span><span style="color: #b8bb26;">"edgeR-Genes/legExpr.csv"</span>, quote<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">FALSE</span>, col.names<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">TRUE</span>, row.names<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">TRUE</span>, sep<span style="color: #d3869b;">=</span><span style="color: #b8bb26;">","</span><span style="color: #d3869b;">)</span>
<span style="color: #b8bb26;">write.table</span><span style="color: #d3869b;">(</span>body, file<span style="color: #d3869b;">=</span><span style="color: #b8bb26;">"edgeR-Genes/bodyExpr.csv"</span>, quote<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">FALSE</span>, col.names<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">TRUE</span>, row.names<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">TRUE</span>, sep<span style="color: #d3869b;">=</span><span style="color: #b8bb26;">","</span><span style="color: #d3869b;">)</span>
</pre>
</div>
</div>
</div>

<div id="outline-container-org4857fa0" class="outline-4">
<h4 id="org4857fa0"><span class="section-number-4">5.2.5</span> Find the relative expression values when compared to the control</h4>
<div class="outline-text-4" id="text-5-2-5">
<p>
This is the main comparison with the pesticides done with the edgeR table. The control is subtracted from the pesticide expression to give a relative expression value. This is not necessary in Deseq and is instead done by manipulating the design.
</p>
<div class="org-src-container">
<pre class="src src-jupyter-R">deetvcontrol <span style="color: #d3869b;">&lt;-</span> table$logFC.pestdeet <span style="color: #d3869b;">-</span> table$logFC.pestcontrol
permvcontrol <span style="color: #d3869b;">&lt;-</span> table$logFC.pestperm <span style="color: #d3869b;">-</span> table$logFC.pestcontrol
deetvcontrolvbody <span style="color: #d3869b;">&lt;-</span> body$logFC.pestdeet <span style="color: #d3869b;">-</span> body$logFC.pestcontrol
permvcontrolvbody <span style="color: #d3869b;">&lt;-</span> body$logFC.pestperm <span style="color: #d3869b;">-</span> body$logFC.pestcontrol
deetvcontrolvleg <span style="color: #d3869b;">&lt;-</span> leg$logFC.pestdeet <span style="color: #d3869b;">-</span> leg$logFC.pestcontrol
permvcontrolvleg <span style="color: #d3869b;">&lt;-</span> leg$logFC.pestperm <span style="color: #d3869b;">-</span> leg$logFC.pestcontrol

table$deetvcontrol <span style="color: #d3869b;">&lt;-</span> deetvcontrol
table$permvcontrol <span style="color: #d3869b;">&lt;-</span> permvcontrol
body$deetvcontrol <span style="color: #d3869b;">&lt;-</span> deetvcontrolvbody
body$permvcontrol <span style="color: #d3869b;">&lt;-</span> permvcontrolvbody
leg$deetvcontrol <span style="color: #d3869b;">&lt;-</span> deetvcontrolvleg
leg$permvcontrol <span style="color: #d3869b;">&lt;-</span> permvcontrolvleg
</pre>
</div>
</div>
</div>
<div id="outline-container-org86b585d" class="outline-4">
<h4 id="org86b585d"><span class="section-number-4">5.2.6</span> Selection of the data</h4>
<div class="outline-text-4" id="text-5-2-6">
<p>
Selects only the columns that have an expression value greater than 2(up-regulated) or less than -2(down-regulated). This is the same selection done earlier in the script with the body and leg.
</p>
<div class="org-src-container">
<pre class="src src-jupyter-R">updeet <span style="color: #d3869b;">&lt;-</span> table<span style="color: #d3869b;">[</span>table$deetvcontrol <span style="color: #d3869b;">&gt;</span> <span style="color: #d3869b; font-weight: bold;">2</span>,<span style="color: #d3869b;">]</span>
downdeet <span style="color: #d3869b;">&lt;-</span> table<span style="color: #d3869b;">[</span>table$deetvcontrol <span style="color: #d3869b;">&lt;</span> <span style="color: #d3869b;">-</span><span style="color: #d3869b; font-weight: bold;">2</span>,<span style="color: #d3869b;">]</span>



updeetbody <span style="color: #d3869b;">&lt;-</span> body<span style="color: #d3869b;">[</span>body$deetvcontrol <span style="color: #d3869b;">&gt;</span> <span style="color: #d3869b; font-weight: bold;">2</span>,<span style="color: #d3869b;">]</span>
downdeetbody <span style="color: #d3869b;">&lt;-</span> body<span style="color: #d3869b;">[</span>body$deetvcontrol <span style="color: #d3869b;">&lt;</span> <span style="color: #d3869b;">-</span><span style="color: #d3869b; font-weight: bold;">2</span>,<span style="color: #d3869b;">]</span>
updeetleg <span style="color: #d3869b;">&lt;-</span> leg<span style="color: #d3869b;">[</span>leg$deetvcontrol <span style="color: #d3869b;">&gt;</span> <span style="color: #d3869b; font-weight: bold;">2</span>,<span style="color: #d3869b;">]</span>
downdeetleg <span style="color: #d3869b;">&lt;-</span> leg<span style="color: #d3869b;">[</span>leg$deetvcontrol <span style="color: #d3869b;">&lt;</span> <span style="color: #d3869b;">-</span><span style="color: #d3869b; font-weight: bold;">2</span>,<span style="color: #d3869b;">]</span>


upperm <span style="color: #d3869b;">&lt;-</span> table<span style="color: #d3869b;">[</span>table$permvcontrol <span style="color: #d3869b;">&gt;</span> <span style="color: #d3869b; font-weight: bold;">2</span>,<span style="color: #d3869b;">]</span>
downperm <span style="color: #d3869b;">&lt;-</span> table<span style="color: #d3869b;">[</span>table$permvcontrol <span style="color: #d3869b;">&lt;</span> <span style="color: #d3869b;">-</span><span style="color: #d3869b; font-weight: bold;">2</span>,<span style="color: #d3869b;">]</span>
uppermbody <span style="color: #d3869b;">&lt;-</span> body<span style="color: #d3869b;">[</span>body$permvcontrol <span style="color: #d3869b;">&gt;</span> <span style="color: #d3869b; font-weight: bold;">2</span>,<span style="color: #d3869b;">]</span>
downpermbody <span style="color: #d3869b;">&lt;-</span> body<span style="color: #d3869b;">[</span>body$permvcontrol <span style="color: #d3869b;">&lt;</span> <span style="color: #d3869b;">-</span><span style="color: #d3869b; font-weight: bold;">2</span>,<span style="color: #d3869b;">]</span>
uppermleg <span style="color: #d3869b;">&lt;-</span> leg<span style="color: #d3869b;">[</span>leg$permvcontrol <span style="color: #d3869b;">&gt;</span> <span style="color: #d3869b; font-weight: bold;">2</span>,<span style="color: #d3869b;">]</span>
downpermleg <span style="color: #d3869b;">&lt;-</span> leg<span style="color: #d3869b;">[</span>leg$permvcontrol <span style="color: #d3869b;">&lt;</span> <span style="color: #d3869b;">-</span><span style="color: #d3869b; font-weight: bold;">2</span>,<span style="color: #d3869b;">]</span>
</pre>
</div>
</div>
</div>
<div id="outline-container-org165b956" class="outline-4">
<h4 id="org165b956"><span class="section-number-4">5.2.7</span> Create a csv for bar plotting</h4>
<div class="outline-text-4" id="text-5-2-7">
<p>
This is just a simple organization of the number of genes deferentially expressed for each comparison. The csv is saved using the write.table line, so change the path to fit your directory structure.
</p>
<div class="org-src-container">
<pre class="src src-jupyter-R">d <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">data.frame</span><span style="color: #d3869b;">(</span>Pest <span style="color: #d3869b;">=</span> <span style="color: #b8bb26;">c</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">rep</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">c</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">"DEET"</span>, <span style="color: #b8bb26;">"DEET"</span>, <span style="color: #b8bb26;">"PERM"</span>, <span style="color: #b8bb26;">"PERM"</span><span style="color: #d3869b;">)))</span>,
                Part <span style="color: #d3869b;">=</span> <span style="color: #b8bb26;">c</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">rep</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">"Body"</span>, <span style="color: #d3869b; font-weight: bold;">4</span><span style="color: #d3869b;">)</span>, <span style="color: #b8bb26;">rep</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">"Leg"</span>, <span style="color: #d3869b; font-weight: bold;">4</span><span style="color: #d3869b;">))</span>,
                Names <span style="color: #d3869b;">=</span> <span style="color: #b8bb26;">c</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">rep</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">c</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">"Up Regulated"</span>, <span style="color: #b8bb26;">"Down Regulated"</span><span style="color: #d3869b;">)</span>, <span style="color: #d3869b; font-weight: bold;">4</span><span style="color: #d3869b;">))</span>,
                GeneCount <span style="color: #d3869b;">=</span> <span style="color: #b8bb26;">c</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">nrow</span><span style="color: #d3869b;">(</span>downdeetbody<span style="color: #d3869b;">)</span>, <span style="color: #b8bb26;">nrow</span><span style="color: #d3869b;">(</span>updeetbody<span style="color: #d3869b;">)</span>, <span style="color: #b8bb26;">nrow</span><span style="color: #d3869b;">(</span>downpermbody<span style="color: #d3869b;">)</span>, <span style="color: #b8bb26;">nrow</span><span style="color: #d3869b;">(</span>uppermbody<span style="color: #d3869b;">)</span>, <span style="color: #b8bb26;">nrow</span><span style="color: #d3869b;">(</span>downdeetleg<span style="color: #d3869b;">)</span>, <span style="color: #b8bb26;">nrow</span><span style="color: #d3869b;">(</span>updeetleg<span style="color: #d3869b;">)</span>, <span style="color: #b8bb26;">nrow</span><span style="color: #d3869b;">(</span>downpermleg<span style="color: #d3869b;">)</span>, <span style="color: #b8bb26;">nrow</span><span style="color: #d3869b;">(</span>uppermleg<span style="color: #d3869b;">)))</span>
<span style="color: #b8bb26;">write.table</span><span style="color: #d3869b;">(</span>d, file<span style="color: #d3869b;">=</span><span style="color: #b8bb26;">"edgeR-Genes/venn-regulation.csv"</span>, quote<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">FALSE</span>, col.names<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">TRUE</span>, row.names<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">FALSE</span>, sep<span style="color: #d3869b;">=</span><span style="color: #b8bb26;">","</span><span style="color: #d3869b;">)</span>
</pre>
</div>
</div>
</div>
<div id="outline-container-orgca40d15" class="outline-4">
<h4 id="orgca40d15"><span class="section-number-4">5.2.8</span> Gets the row-names</h4>
<div class="outline-text-4" id="text-5-2-8">
<p>
This separates the data into just the gene names for each comparison. This is helpful for venn diagrams and other comparison.
</p>
<div class="org-src-container">
<pre class="src src-jupyter-R">upperm <span style="color: #d3869b;">&lt;-</span> table<span style="color: #d3869b;">[</span>table$permvcontrol <span style="color: #d3869b;">&gt;</span> <span style="color: #d3869b; font-weight: bold;">2</span>,<span style="color: #d3869b;">]</span>
downperm <span style="color: #d3869b;">&lt;-</span> table<span style="color: #d3869b;">[</span>table$permvcontrol <span style="color: #d3869b;">&lt;</span> <span style="color: #d3869b;">-</span><span style="color: #d3869b; font-weight: bold;">2</span>,<span style="color: #d3869b;">]</span>
uppermbody <span style="color: #d3869b;">&lt;-</span> body<span style="color: #d3869b;">[</span>body$permvcontrol <span style="color: #d3869b;">&gt;</span> <span style="color: #d3869b; font-weight: bold;">2</span>,<span style="color: #d3869b;">]</span>
downpermbody <span style="color: #d3869b;">&lt;-</span> body<span style="color: #d3869b;">[</span>body$permvcontrol <span style="color: #d3869b;">&lt;</span> <span style="color: #d3869b;">-</span><span style="color: #d3869b; font-weight: bold;">2</span>,<span style="color: #d3869b;">]</span>
uppermleg <span style="color: #d3869b;">&lt;-</span> leg<span style="color: #d3869b;">[</span>leg$permvcontrol <span style="color: #d3869b;">&gt;</span> <span style="color: #d3869b; font-weight: bold;">2</span>,<span style="color: #d3869b;">]</span>
downpermleg <span style="color: #d3869b;">&lt;-</span> leg<span style="color: #d3869b;">[</span>leg$permvcontrol <span style="color: #d3869b;">&lt;</span> <span style="color: #d3869b;">-</span><span style="color: #d3869b; font-weight: bold;">2</span>,<span style="color: #d3869b;">]</span>

uppermgenes <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">rownames</span><span style="color: #d3869b;">(</span>upperm<span style="color: #d3869b;">)</span>
downpermgenes <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">rownames</span><span style="color: #d3869b;">(</span>downperm<span style="color: #d3869b;">)</span>
uppermleggenes <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">rownames</span><span style="color: #d3869b;">(</span>uppermleg<span style="color: #d3869b;">)</span>
downpermleggenes <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">rownames</span><span style="color: #d3869b;">(</span>downpermleg<span style="color: #d3869b;">)</span>
uppermbodygenes <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">rownames</span><span style="color: #d3869b;">(</span>uppermbody<span style="color: #d3869b;">)</span>
downpermbodygenes <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">rownames</span><span style="color: #d3869b;">(</span>downpermbody<span style="color: #d3869b;">)</span>

updeetgenes <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">rownames</span><span style="color: #d3869b;">(</span>updeet<span style="color: #d3869b;">)</span>
downdeetgenes <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">rownames</span><span style="color: #d3869b;">(</span>downdeet<span style="color: #d3869b;">)</span>
updeetleggenes <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">rownames</span><span style="color: #d3869b;">(</span>updeetleg<span style="color: #d3869b;">)</span>
downdeetleggenes <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">rownames</span><span style="color: #d3869b;">(</span>downdeetleg<span style="color: #d3869b;">)</span>
updeetbodygenes <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">rownames</span><span style="color: #d3869b;">(</span>updeetbody<span style="color: #d3869b;">)</span>
downdeetbodygenes <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">rownames</span><span style="color: #d3869b;">(</span>downdeetbody<span style="color: #d3869b;">)</span>
</pre>
</div>
</div>
</div>

<div id="outline-container-org7665858" class="outline-4">
<h4 id="org7665858"><span class="section-number-4">5.2.9</span> Create a list of dataframes corresponding to the total, leg, and body</h4>
<div class="outline-text-4" id="text-5-2-9">
<p>
Consolidates the genes into a list of dataframes for easy saving
</p>
<div class="org-src-container">
<pre class="src src-jupyter-R">genelist <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">list</span><span style="color: #d3869b;">(</span>upPerm <span style="color: #d3869b;">=</span> uppermgenes, downPerm <span style="color: #d3869b;">=</span> downpermgenes,
                upPermBody<span style="color: #d3869b;">=</span>uppermbodygenes, downPermBody<span style="color: #d3869b;">=</span>downpermbodygenes,
                upPermLeg <span style="color: #d3869b;">=</span> uppermleggenes, downPermLeg <span style="color: #d3869b;">=</span> downpermleggenes,
                upDeet <span style="color: #d3869b;">=</span> updeetgenes, downDeet <span style="color: #d3869b;">=</span> downdeetgenes,
                upDeetLeg <span style="color: #d3869b;">=</span> updeetleggenes, downDeetLeg <span style="color: #d3869b;">=</span> downdeetleggenes,
                upDeetBody <span style="color: #d3869b;">=</span> updeetbodygenes, downDeetBody <span style="color: #d3869b;">=</span> downdeetbodygenes, table<span style="color: #d3869b;">=</span><span style="color: #b8bb26;">rownames</span><span style="color: #d3869b;">(</span>table<span style="color: #d3869b;">)</span>, leg<span style="color: #d3869b;">=</span><span style="color: #b8bb26;">rownames</span><span style="color: #d3869b;">(</span>leg<span style="color: #d3869b;">)</span>, body<span style="color: #d3869b;">=</span><span style="color: #b8bb26;">rownames</span><span style="color: #d3869b;">(</span>body<span style="color: #d3869b;">))</span>
</pre>
</div>
</div>
</div>
<div id="outline-container-org6784337" class="outline-4">
<h4 id="org6784337"><span class="section-number-4">5.2.10</span> Save the genelist to files</h4>
<div class="outline-text-4" id="text-5-2-10">
<p>
Saves all of the comparisons to file.
</p>
<div class="org-src-container">
<pre class="src src-jupyter-R"><span style="color: #fb4934;">for</span> <span style="color: #d3869b;">(</span>i <span style="color: #fb4934;">in</span> <span style="color: #b8bb26;">seq</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">length</span><span style="color: #d3869b;">(</span>genelist<span style="color: #d3869b;">))){</span>
    namelist <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">names</span><span style="color: #d3869b;">(</span>genelist<span style="color: #d3869b;">)</span>
    <span style="color: #b8bb26;">write.table</span><span style="color: #d3869b;">(</span>genelist<span style="color: #d3869b;">[</span>i<span style="color: #d3869b;">]</span>, file<span style="color: #d3869b;">=</span><span style="color: #b8bb26;">paste</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">"edgeR-Genes/Genelists"</span>,<span style="color: #b8bb26;">"/"</span>, namelist<span style="color: #d3869b;">[</span>i<span style="color: #d3869b;">]</span>, <span style="color: #b8bb26;">".csv"</span>, sep<span style="color: #d3869b;">=</span><span style="color: #b8bb26;">""</span><span style="color: #d3869b;">)</span>, quote<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">FALSE</span>, col.names<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">FALSE</span>, row.names<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">FALSE</span>, sep<span style="color: #d3869b;">=</span><span style="color: #b8bb26;">","</span><span style="color: #d3869b;">)</span>
<span style="color: #d3869b;">}</span>
</pre>
</div>
</div>
</div>
</div>
<div id="outline-container-org03f0886" class="outline-3">
<h3 id="org03f0886"><span class="section-number-3">5.3</span> WGCNA</h3>
<div class="outline-text-3" id="text-5-3">
</div>
<div id="outline-container-org903e254" class="outline-4">
<h4 id="org903e254"><span class="section-number-4">5.3.1</span> Build data file for WGCNA from quant files</h4>
<div class="outline-text-4" id="text-5-3-1">
<div class="org-src-container">
<pre class="src src-jupyter-R"><span style="color: #d3869b;">setwd</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">"/home/david/Documents/BenoitLab/R/largeRNASEQ/CorrectedQuantData/"</span><span style="color: #d3869b;">)</span>
quantfiles <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">list.files</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">"."</span>, pattern <span style="color: #d3869b;">=</span> <span style="color: #b8bb26;">".tabular"</span><span style="color: #d3869b;">)</span>
df <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">data.frame</span><span style="color: #d3869b;">(</span>row.names <span style="color: #d3869b;">=</span> <span style="color: #b8bb26;">read.delim</span><span style="color: #d3869b;">(</span>quantfiles<span style="color: #d3869b;">[</span><span style="color: #d3869b; font-weight: bold;">1</span><span style="color: #d3869b;">])</span>$Name<span style="color: #d3869b;">)</span>
<span style="color: #fb4934;">for</span> <span style="color: #d3869b;">(</span>fileName <span style="color: #fb4934;">in</span> quantfiles<span style="color: #d3869b;">)</span> <span style="color: #d3869b;">{</span>
  file <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">read.delim</span><span style="color: #d3869b;">(</span>fileName<span style="color: #d3869b;">)</span>
  df<span style="color: #d3869b;">[</span>tools::<span style="color: #b8bb26;">file_path_sans_ext</span><span style="color: #d3869b;">(</span>fileName<span style="color: #d3869b;">)]</span> <span style="color: #d3869b;">&lt;-</span> file$TPM
<span style="color: #d3869b;">}</span>
<span style="color: #b8bb26;">write.csv</span><span style="color: #d3869b;">(</span>df, <span style="color: #b8bb26;">"../WGCNACorrectedQuant.csv"</span>, quote <span style="color: #d3869b;">=</span> <span style="color: #fabd2f;">FALSE</span><span style="color: #d3869b;">)</span>
<span style="color: #b8bb26;">head</span><span style="color: #d3869b;">(</span>df<span style="color: #d3869b;">)</span>
</pre>
</div>

<div class="org-src-container">
<pre class="src src-rst">
A data.frame: 6 &#215; 18

| &lt;!--/--&gt; | Cont_Bod_1 &amp;lt;dbl&amp;gt; | Cont_Bod_2 &amp;lt;dbl&amp;gt; | Cont_Bod_3 &amp;lt;dbl&amp;gt; | Cont_Leg_1 &amp;lt;dbl&amp;gt; | Cont_Leg_2 &amp;lt;dbl&amp;gt; | Cont_Leg_2a &amp;lt;dbl&amp;gt; | Cont_Leg_3 &amp;lt;dbl&amp;gt; | Deet_Bod_1 &amp;lt;dbl&amp;gt; | Deet_Bod_2 &amp;lt;dbl&amp;gt; | Deet_Bod_3 &amp;lt;dbl&amp;gt; | Deet_Leg_1 &amp;lt;dbl&amp;gt; | Deet_Leg_2 &amp;lt;dbl&amp;gt; | Deet_Leg_3 &amp;lt;dbl&amp;gt; | Per_Bod_1 &amp;lt;dbl&amp;gt; | Per_Bod_3 &amp;lt;dbl&amp;gt; | Per_Leg_1 &amp;lt;dbl&amp;gt; | Per_Leg_2 &amp;lt;dbl&amp;gt; | Per_Leg_3 &amp;lt;dbl&amp;gt; |
<span style="color: #83a598;">|---|</span>---<span style="color: #83a598;">|---|</span>---<span style="color: #83a598;">|---|</span>---<span style="color: #83a598;">|---|</span>---<span style="color: #83a598;">|---|</span>---<span style="color: #83a598;">|---|</span>---<span style="color: #83a598;">|---|</span>---<span style="color: #83a598;">|---|</span>---<span style="color: #83a598;">|---|</span>---<span style="color: #83a598;">|---|</span>
| dv.1 | 22.2763 | 37.8193 | 40.3742 | 61.417900 | 0.0681662 | 83.278200 | 69.070000 | 48.1591 |  43.6397 | 50.0788 | 86.710000 | 94.912900 | 85.979000 | 52.7464 | 56.6623 | 9.02202e+01 | 96.067400 | 76.999000 |
| dv.2 | 15.9128 | 44.7599 | 54.8798 |  0.786553 | 0.0756211 |  0.722977 |  0.493229 | 93.6505 | 105.8530 | 72.7003 |  2.753670 |  0.870014 |  3.473720 | 73.2376 | 80.6752 | 1.26751e+00 |  0.855653 |  0.255794 |
| dv.3 | 10.6041 | 11.7978 | 14.3419 |  9.187200 | 0.0756683 | 14.169700 | 12.297800 | 16.7729 |  14.5622 | 18.3525 | 14.070000 | 17.049100 | 15.897900 | 16.5017 | 18.0218 | 1.62399e+01 | 18.330600 | 14.676500 |
| dv.4 | 16.5992 | 19.8922 | 23.9323 | 24.027200 | 0.0794267 | 30.721800 | 27.661300 | 24.9121 |  21.8748 | 27.2285 | 29.715900 | 35.491900 | 34.275800 | 23.6021 | 25.1121 | 2.79280e+01 | 31.287200 | 25.525300 |
| dv.5 |  0.0000 |  0.0000 |  0.0000 |  0.294700 | 0.0834384 |  0.875706 |  0.715919 |  0.0000 |   0.0000 |  0.0000 |  0.320064 |  0.391045 |  0.193081 |  0.0000 |  0.0000 | 2.79643e-05 |  0.671768 |  0.217907 |
| dv.6 |  0.0000 |  0.0000 |  0.0000 |  0.484270 | 0.0837346 |  0.431047 |  0.759755 |  0.0000 |   0.0000 |  0.0000 |  0.329439 |  0.940694 |  1.231160 |  0.0000 |  0.0000 | 1.37827e+00 |  0.962562 |  0.141731 |
</pre>
</div>
</div>
</div>

<div id="outline-container-org73e157a" class="outline-4">
<h4 id="org73e157a"><span class="section-number-4">5.3.2</span> Build metadata truth table for WGCNA</h4>
<div class="outline-text-4" id="text-5-3-2">
<p>
This is the format for the WGCNA metadata. This can be generated from the metadata used in the edgeR and Deseq2 sections. It can also be written by hand.
</p>
<table border="2" cellspacing="0" cellpadding="6" rules="groups" frame="hsides">


<colgroup>
<col  class="org-left" />

<col  class="org-right" />

<col  class="org-right" />

<col  class="org-right" />

<col  class="org-right" />

<col  class="org-right" />

<col  class="org-right" />

<col  class="org-right" />

<col  class="org-right" />
</colgroup>
<tbody>
<tr>
<td class="org-left">sample</td>
<td class="org-right">pesticides_perm</td>
<td class="org-right">pesticides_deet</td>
<td class="org-right">pesticides_control</td>
<td class="org-right">part_body</td>
<td class="org-right">part_leg</td>
<td class="org-right">trial_1</td>
<td class="org-right">trial_3</td>
<td class="org-right">trial_2</td>
</tr>

<tr>
<td class="org-left">Per_Bod_1.tabular</td>
<td class="org-right">1</td>
<td class="org-right">0</td>
<td class="org-right">0</td>
<td class="org-right">1</td>
<td class="org-right">0</td>
<td class="org-right">1</td>
<td class="org-right">0</td>
<td class="org-right">0</td>
</tr>

<tr>
<td class="org-left">Per_Bod_3.tabular</td>
<td class="org-right">1</td>
<td class="org-right">0</td>
<td class="org-right">0</td>
<td class="org-right">1</td>
<td class="org-right">0</td>
<td class="org-right">0</td>
<td class="org-right">1</td>
<td class="org-right">0</td>
</tr>

<tr>
<td class="org-left">Per_Leg_3.tabular</td>
<td class="org-right">1</td>
<td class="org-right">0</td>
<td class="org-right">0</td>
<td class="org-right">0</td>
<td class="org-right">1</td>
<td class="org-right">0</td>
<td class="org-right">1</td>
<td class="org-right">0</td>
</tr>

<tr>
<td class="org-left">Per_Leg_2.tabular</td>
<td class="org-right">1</td>
<td class="org-right">0</td>
<td class="org-right">0</td>
<td class="org-right">0</td>
<td class="org-right">1</td>
<td class="org-right">0</td>
<td class="org-right">0</td>
<td class="org-right">1</td>
</tr>

<tr>
<td class="org-left">Per_Leg_1.tabular</td>
<td class="org-right">1</td>
<td class="org-right">0</td>
<td class="org-right">0</td>
<td class="org-right">0</td>
<td class="org-right">1</td>
<td class="org-right">1</td>
<td class="org-right">0</td>
<td class="org-right">0</td>
</tr>

<tr>
<td class="org-left">Deet_Bod_3.tabular</td>
<td class="org-right">0</td>
<td class="org-right">1</td>
<td class="org-right">0</td>
<td class="org-right">1</td>
<td class="org-right">0</td>
<td class="org-right">0</td>
<td class="org-right">1</td>
<td class="org-right">0</td>
</tr>

<tr>
<td class="org-left">Deet_Bod_2.tabular</td>
<td class="org-right">0</td>
<td class="org-right">1</td>
<td class="org-right">0</td>
<td class="org-right">1</td>
<td class="org-right">0</td>
<td class="org-right">0</td>
<td class="org-right">0</td>
<td class="org-right">1</td>
</tr>

<tr>
<td class="org-left">Deet_Bod_1.tabular</td>
<td class="org-right">0</td>
<td class="org-right">1</td>
<td class="org-right">0</td>
<td class="org-right">1</td>
<td class="org-right">0</td>
<td class="org-right">1</td>
<td class="org-right">0</td>
<td class="org-right">0</td>
</tr>

<tr>
<td class="org-left">Deet_Leg_3.tabular</td>
<td class="org-right">0</td>
<td class="org-right">1</td>
<td class="org-right">0</td>
<td class="org-right">0</td>
<td class="org-right">1</td>
<td class="org-right">0</td>
<td class="org-right">1</td>
<td class="org-right">0</td>
</tr>

<tr>
<td class="org-left">Deet_Leg_2.tabular</td>
<td class="org-right">0</td>
<td class="org-right">1</td>
<td class="org-right">0</td>
<td class="org-right">0</td>
<td class="org-right">1</td>
<td class="org-right">0</td>
<td class="org-right">0</td>
<td class="org-right">1</td>
</tr>

<tr>
<td class="org-left">Deet_Leg_1.tabular</td>
<td class="org-right">0</td>
<td class="org-right">1</td>
<td class="org-right">0</td>
<td class="org-right">0</td>
<td class="org-right">1</td>
<td class="org-right">1</td>
<td class="org-right">0</td>
<td class="org-right">0</td>
</tr>

<tr>
<td class="org-left">Cont_Bod_3.tabular</td>
<td class="org-right">0</td>
<td class="org-right">0</td>
<td class="org-right">1</td>
<td class="org-right">1</td>
<td class="org-right">0</td>
<td class="org-right">0</td>
<td class="org-right">1</td>
<td class="org-right">0</td>
</tr>

<tr>
<td class="org-left">Cont_Bod_2.tabular</td>
<td class="org-right">0</td>
<td class="org-right">0</td>
<td class="org-right">1</td>
<td class="org-right">1</td>
<td class="org-right">0</td>
<td class="org-right">0</td>
<td class="org-right">0</td>
<td class="org-right">1</td>
</tr>

<tr>
<td class="org-left">Cont_Bod_1.tabular</td>
<td class="org-right">0</td>
<td class="org-right">0</td>
<td class="org-right">1</td>
<td class="org-right">1</td>
<td class="org-right">0</td>
<td class="org-right">1</td>
<td class="org-right">0</td>
<td class="org-right">0</td>
</tr>

<tr>
<td class="org-left">Cont_Leg_3.tabular</td>
<td class="org-right">0</td>
<td class="org-right">0</td>
<td class="org-right">1</td>
<td class="org-right">0</td>
<td class="org-right">1</td>
<td class="org-right">0</td>
<td class="org-right">1</td>
<td class="org-right">0</td>
</tr>

<tr>
<td class="org-left">Cont_Leg_2a.tabular</td>
<td class="org-right">0</td>
<td class="org-right">0</td>
<td class="org-right">1</td>
<td class="org-right">0</td>
<td class="org-right">1</td>
<td class="org-right">0</td>
<td class="org-right">0</td>
<td class="org-right">1</td>
</tr>

<tr>
<td class="org-left">Cont_Leg_1.tabular</td>
<td class="org-right">0</td>
<td class="org-right">0</td>
<td class="org-right">1</td>
<td class="org-right">0</td>
<td class="org-right">1</td>
<td class="org-right">1</td>
<td class="org-right">0</td>
<td class="org-right">0</td>
</tr>
</tbody>
</table>
</div>
<ol class="org-ol">
<li><a id="org09b87bc"></a>From existing metadata file<br />
<div class="outline-text-5" id="text-5-3-2-1">
<p>
This can be automated from existing metadata file in the deseq section.
</p>
<div class="org-src-container">
<pre class="src src-jupyter-R"><span style="color: #d3869b;">setwd</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">"/home/david/Documents/BenoitLab/R/largeRNASEQ"</span><span style="color: #d3869b;">)</span>
metadata <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">read.csv</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">"metadata.csv"</span>, row.names <span style="color: #d3869b;">=</span> <span style="color: #d3869b; font-weight: bold;">1</span><span style="color: #d3869b;">)</span>
truthTable <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">data.frame</span><span style="color: #d3869b;">(</span>row.names <span style="color: #d3869b;">=</span> <span style="color: #b8bb26;">row.names</span><span style="color: #d3869b;">(</span>metadata<span style="color: #d3869b;">))</span>
<span style="color: #fb4934;">for</span> <span style="color: #d3869b;">(</span>column <span style="color: #fb4934;">in</span> <span style="color: #b8bb26;">colnames</span><span style="color: #d3869b;">(</span>metadata<span style="color: #d3869b;">))</span> <span style="color: #d3869b;">{</span>
  <span style="color: #fb4934;">for</span> <span style="color: #d3869b;">(</span>row <span style="color: #fb4934;">in</span> <span style="color: #b8bb26;">unique</span><span style="color: #d3869b;">(</span>metadata<span style="color: #d3869b;">[</span>column<span style="color: #d3869b;">]))</span> <span style="color: #d3869b;">{</span>
    <span style="color: #fb4934;">for</span> <span style="color: #d3869b;">(</span>item <span style="color: #fb4934;">in</span> row<span style="color: #d3869b;">)</span> <span style="color: #d3869b;">{</span>
      truthTable<span style="color: #d3869b;">[</span>, <span style="color: #b8bb26;">paste</span><span style="color: #d3869b;">(</span>column, item, sep <span style="color: #d3869b;">=</span> <span style="color: #b8bb26;">"_"</span><span style="color: #d3869b;">)]</span> <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">as.integer</span><span style="color: #d3869b;">(</span>item <span style="color: #d3869b;">==</span> metadata<span style="color: #d3869b;">[[</span>column<span style="color: #d3869b;">]])</span>
    <span style="color: #d3869b;">}</span>
  <span style="color: #d3869b;">}</span>
<span style="color: #d3869b;">}</span>
truthTable <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">data.frame</span><span style="color: #d3869b;">(</span>sample <span style="color: #d3869b;">=</span> <span style="color: #b8bb26;">row.names</span><span style="color: #d3869b;">(</span>truthTable<span style="color: #d3869b;">)</span>, truthTable<span style="color: #d3869b;">)</span>
<span style="color: #b8bb26;">write.csv</span><span style="color: #d3869b;">(</span>truthTable, <span style="color: #b8bb26;">"WGCNATruthTable.csv"</span>, row.names<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">FALSE</span><span style="color: #d3869b;">)</span>
truthTable
</pre>
</div>
</div>
</li>
<li><a id="org0741d33"></a>From scratch<br />
<div class="outline-text-5" id="text-5-3-2-2">
<p>
It can also be built from scratch
</p>
<div class="org-src-container">
<pre class="src src-jupyter-R">truthTable <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">data.frame</span><span style="color: #d3869b;">(</span>sample <span style="color: #d3869b;">=</span> <span style="color: #b8bb26;">colnames</span><span style="color: #d3869b;">(</span>df<span style="color: #d3869b;">))</span>
truthTable$Control <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">c</span><span style="color: #d3869b;">(</span>
  <span style="color: #d3869b; font-weight: bold;">1</span>, <span style="color: #d3869b; font-weight: bold;">1</span>, <span style="color: #d3869b; font-weight: bold;">1</span>, <span style="color: #d3869b; font-weight: bold;">1</span>, <span style="color: #d3869b; font-weight: bold;">1</span>, <span style="color: #d3869b; font-weight: bold;">1</span>,
  <span style="color: #d3869b; font-weight: bold;">1</span>, <span style="color: #d3869b; font-weight: bold;">0</span>, <span style="color: #d3869b; font-weight: bold;">0</span>, <span style="color: #d3869b; font-weight: bold;">0</span>, <span style="color: #d3869b; font-weight: bold;">0</span>, <span style="color: #d3869b; font-weight: bold;">0</span>,
  <span style="color: #d3869b; font-weight: bold;">0</span>, <span style="color: #d3869b; font-weight: bold;">0</span>, <span style="color: #d3869b; font-weight: bold;">0</span>, <span style="color: #d3869b; font-weight: bold;">0</span>, <span style="color: #d3869b; font-weight: bold;">0</span>, <span style="color: #d3869b; font-weight: bold;">0</span>
<span style="color: #d3869b;">)</span>
truthTable$DEET <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">c</span><span style="color: #d3869b;">(</span>
  <span style="color: #d3869b; font-weight: bold;">0</span>, <span style="color: #d3869b; font-weight: bold;">0</span>, <span style="color: #d3869b; font-weight: bold;">0</span>, <span style="color: #d3869b; font-weight: bold;">0</span>, <span style="color: #d3869b; font-weight: bold;">0</span>, <span style="color: #d3869b; font-weight: bold;">0</span>,
  <span style="color: #d3869b; font-weight: bold;">0</span>, <span style="color: #d3869b; font-weight: bold;">1</span>, <span style="color: #d3869b; font-weight: bold;">1</span>, <span style="color: #d3869b; font-weight: bold;">1</span>, <span style="color: #d3869b; font-weight: bold;">1</span>, <span style="color: #d3869b; font-weight: bold;">1</span>,
  <span style="color: #d3869b; font-weight: bold;">1</span>, <span style="color: #d3869b; font-weight: bold;">0</span>, <span style="color: #d3869b; font-weight: bold;">0</span>, <span style="color: #d3869b; font-weight: bold;">0</span>, <span style="color: #d3869b; font-weight: bold;">0</span>, <span style="color: #d3869b; font-weight: bold;">0</span>
<span style="color: #d3869b;">)</span>
truthTable$Permethrin <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">c</span><span style="color: #d3869b;">(</span>
  <span style="color: #d3869b; font-weight: bold;">0</span>, <span style="color: #d3869b; font-weight: bold;">0</span>, <span style="color: #d3869b; font-weight: bold;">0</span>, <span style="color: #d3869b; font-weight: bold;">0</span>, <span style="color: #d3869b; font-weight: bold;">0</span>, <span style="color: #d3869b; font-weight: bold;">0</span>,
  <span style="color: #d3869b; font-weight: bold;">0</span>, <span style="color: #d3869b; font-weight: bold;">0</span>, <span style="color: #d3869b; font-weight: bold;">0</span>, <span style="color: #d3869b; font-weight: bold;">0</span>, <span style="color: #d3869b; font-weight: bold;">0</span>, <span style="color: #d3869b; font-weight: bold;">0</span>,
  <span style="color: #d3869b; font-weight: bold;">0</span>, <span style="color: #d3869b; font-weight: bold;">1</span>, <span style="color: #d3869b; font-weight: bold;">1</span>, <span style="color: #d3869b; font-weight: bold;">1</span>, <span style="color: #d3869b; font-weight: bold;">1</span>, <span style="color: #d3869b; font-weight: bold;">1</span>
<span style="color: #d3869b;">)</span>
truthTable$Body <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">c</span><span style="color: #d3869b;">(</span>
  <span style="color: #d3869b; font-weight: bold;">1</span>, <span style="color: #d3869b; font-weight: bold;">1</span>, <span style="color: #d3869b; font-weight: bold;">1</span>, <span style="color: #d3869b; font-weight: bold;">0</span>, <span style="color: #d3869b; font-weight: bold;">0</span>, <span style="color: #d3869b; font-weight: bold;">0</span>,
  <span style="color: #d3869b; font-weight: bold;">0</span>, <span style="color: #d3869b; font-weight: bold;">1</span>, <span style="color: #d3869b; font-weight: bold;">1</span>, <span style="color: #d3869b; font-weight: bold;">1</span>, <span style="color: #d3869b; font-weight: bold;">0</span>, <span style="color: #d3869b; font-weight: bold;">0</span>,
  <span style="color: #d3869b; font-weight: bold;">0</span>, <span style="color: #d3869b; font-weight: bold;">1</span>, <span style="color: #d3869b; font-weight: bold;">1</span>, <span style="color: #d3869b; font-weight: bold;">0</span>, <span style="color: #d3869b; font-weight: bold;">0</span>, <span style="color: #d3869b; font-weight: bold;">0</span>
<span style="color: #d3869b;">)</span>
truthTable$Leg <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">c</span><span style="color: #d3869b;">(</span>
  <span style="color: #d3869b; font-weight: bold;">0</span>, <span style="color: #d3869b; font-weight: bold;">0</span>, <span style="color: #d3869b; font-weight: bold;">0</span>, <span style="color: #d3869b; font-weight: bold;">1</span>, <span style="color: #d3869b; font-weight: bold;">1</span>, <span style="color: #d3869b; font-weight: bold;">1</span>,
  <span style="color: #d3869b; font-weight: bold;">1</span>, <span style="color: #d3869b; font-weight: bold;">0</span>, <span style="color: #d3869b; font-weight: bold;">0</span>, <span style="color: #d3869b; font-weight: bold;">0</span>, <span style="color: #d3869b; font-weight: bold;">1</span>, <span style="color: #d3869b; font-weight: bold;">1</span>,
  <span style="color: #d3869b; font-weight: bold;">1</span>, <span style="color: #d3869b; font-weight: bold;">0</span>, <span style="color: #d3869b; font-weight: bold;">0</span>, <span style="color: #d3869b; font-weight: bold;">1</span>, <span style="color: #d3869b; font-weight: bold;">1</span>, <span style="color: #d3869b; font-weight: bold;">1</span>

<span style="color: #b8bb26;">write.csv</span><span style="color: #d3869b;">(</span>truthTable, <span style="color: #b8bb26;">"../WGCNATruthTable.csv"</span>, row.names <span style="color: #d3869b;">=</span> <span style="color: #fabd2f;">FALSE</span><span style="color: #d3869b;">)</span>
</pre>
</div>
</div>
</li>
</ol>
</div>
</div>
</div>
<div id="outline-container-org334a537" class="outline-2">
<h2 id="org334a537"><span class="section-number-2">6</span> Fasta Manipulation</h2>
<div class="outline-text-2" id="text-6">
</div>
<div id="outline-container-orgc39cee3" class="outline-3">
<h3 id="orgc39cee3"><span class="section-number-3">6.1</span> Get Gene names</h3>
<div class="outline-text-3" id="text-6-1">
<p>
This program just pulls the names of the genes out of the fasta file. There are probably less messy ways to do this. Online programs exist for this as well, but this is a fairly simple c++ program.
</p>
<div class="org-src-container">
<pre class="src src-cpp"><span style="color: #ebdbb2; font-weight: bold;">#include</span> <span style="color: #fe8019;">&lt;</span><span style="color: #b8bb26;">iostream</span><span style="color: #fe8019;">&gt;</span>
<span style="color: #ebdbb2; font-weight: bold;">#include</span> <span style="color: #fe8019;">&lt;</span><span style="color: #b8bb26;">fstream</span><span style="color: #fe8019;">&gt;</span>
<span style="color: #ebdbb2; font-weight: bold;">#include</span> <span style="color: #fe8019;">&lt;</span><span style="color: #b8bb26;">string</span><span style="color: #fe8019;">&gt;</span>
<span style="color: #ebdbb2; font-weight: bold;">#include</span> <span style="color: #fe8019;">&lt;</span><span style="color: #b8bb26;">vector</span><span style="color: #fe8019;">&gt;</span>

<span style="color: #fabd2f;">int</span> <span style="color: #b8bb26;">main</span><span style="color: #fe8019;">(){</span>
    <span style="color: #d3869b;">std</span>::<span style="color: #fabd2f;">ifstream</span> <span style="color: #83a598;">in</span>;
    in.open<span style="color: #fb4934;">(</span><span style="color: #b8bb26;">"GeneReference/genelist.fasta"</span><span style="color: #fb4934;">)</span>;
    <span style="color: #d3869b;">std</span>::<span style="color: #fabd2f;">string</span> <span style="color: #83a598;">line</span>;
    <span style="color: #d3869b;">std</span>::<span style="color: #fabd2f;">ofstream</span> <span style="color: #83a598;">out</span>;
    out.open<span style="color: #fb4934;">(</span><span style="color: #b8bb26;">"GeneReference/listofgenes.csv"</span><span style="color: #fb4934;">)</span>;
    <span style="color: #fb4934;">if</span> <span style="color: #fb4934;">(</span>in.is_open<span style="color: #cc241d;">()</span><span style="color: #fb4934;">){</span>
        <span style="color: #fb4934;">while</span><span style="color: #cc241d;">(</span>getline<span style="color: #83a598;">(</span>in, line<span style="color: #83a598;">)</span><span style="color: #cc241d;">){</span>
            <span style="color: #fb4934;">if</span> <span style="color: #83a598;">(</span>line<span style="color: #fe8019;">[</span><span style="color: #d3869b; font-weight: bold;">0</span><span style="color: #fe8019;">]</span> == <span style="color: #b8bb26;">'&gt;'</span><span style="color: #83a598;">){</span>
                line.erase<span style="color: #fe8019;">(</span><span style="color: #d3869b; font-weight: bold;">0</span>, <span style="color: #d3869b; font-weight: bold;">1</span><span style="color: #fe8019;">)</span>;
                out &lt;&lt; line &lt;&lt; <span style="color: #d3869b;">std</span>::endl;
            <span style="color: #83a598;">}</span>
        <span style="color: #cc241d;">}</span>
    <span style="color: #fb4934;">}</span>
<span style="color: #fe8019;">}</span>
</pre>
</div>
</div>
</div>
</div>

<div id="outline-container-orgae5df12" class="outline-2">
<h2 id="orgae5df12"><span class="section-number-2">7</span> Blast</h2>
<div class="outline-text-2" id="text-7">
<p>
You have 2 choices for this step.
</p>
<ul class="org-ul">
<li>If you are familiar with <a href="https://digitalinsights.qiagen.com/products-overview/discovery-insights-portfolio/analysis-and-visualization/qiagen-clc-main-workbench/">CLC main workbench</a> then you can use that to interface with <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi">NCBI blast</a> to compare the genes to another organism with annotations (Ixodes Scapularius in my case).</li>
<li>The other option is to use <a href="https://github.com/IllustratedMan-code/FastBlast">This Repository</a> that I wrote called FastBlast.</li>
</ul>
</div>
<div id="outline-container-org31795de" class="outline-3">
<h3 id="org31795de"><span class="section-number-3">7.1</span> Fastblast</h3>
<div class="outline-text-3" id="text-7-1">
<p>
The repository has a more detailed README but I will relate it to the rest of the code here
</p>
<ul class="org-ul">
<li>The files that have been generated from Deseq and EdgeR need to be compared to another genome that has annotations, so to do that I am going to compare my organism (Dermacentor Variablis) to Ixodes Scapularis using blast.</li>
<li>FastBlast automates the blast analysis with the blast command line tool</li>
<li>A c++ file parses the csv files that were generated previously and creates .fasta files using the DV assembly.</li>
<li>Blast compares the new fasta files with a blast database created with the Ixodes to give gene-ontology terms in a .blastout file.</li>
<li>A c++ file parses the blast output and extracts only the genes that were most similar in the blast output and exports the new list as a .csv</li>
</ul>
</div>
</div>
</div>

<div id="outline-container-org82de278" class="outline-2">
<h2 id="org82de278"><span class="section-number-2">8</span> Gene Ontology analysis</h2>
<div class="outline-text-2" id="text-8">
</div>
<div id="outline-container-orgc7b8854" class="outline-3">
<h3 id="orgc7b8854"><span class="section-number-3">8.1</span> Tools</h3>
<div class="outline-text-3" id="text-8-1">
<ul class="org-ul">
<li><a href="http://revigo.irb.hr/">Revigo</a></li>
<li><a href="http://geneontology.org">geneontology.org</a></li>
<li><a href="https://biit.cs.ut.ee/gprofiler/gost">G:profiler</a></li>
</ul>
</div>
</div>
<div id="outline-container-org0139119" class="outline-3">
<h3 id="org0139119"><span class="section-number-3">8.2</span> Automation</h3>
<div class="outline-text-3" id="text-8-2">
<p>
I wanted to automate these tools even though they are only available as websites (except for G:profiler) because I have a lot of different csv files to manage in this project, and doing it all by hand seemed like a recipe for disaster.
</p>
<ul class="org-ul">
<li>Refer to the FastBlast repository for instructions on how to run these web-scraper scripts</li>
<li>The files outputted from the blast step are in the correct format to use with the GO tools</li>
<li>Using the web scraper scripts, the data is transformed into a format that is plottable as treemaps and dotplots (plotting section)</li>
</ul>
</div>
</div>
</div>

<div id="outline-container-org815c7b5" class="outline-2">
<h2 id="org815c7b5"><span class="section-number-2">9</span> Plotting</h2>
<div class="outline-text-2" id="text-9">
<p>
This uses R and python to plot data.
</p>
</div>
<div id="outline-container-orgd2c26be" class="outline-3">
<h3 id="orgd2c26be"><span class="section-number-3">9.1</span> Required Packages</h3>
<div class="outline-text-3" id="text-9-1">
<p>
These packages are only required for plotting the data
</p>
</div>
<div id="outline-container-org1f07c6e" class="outline-4">
<h4 id="org1f07c6e"><span class="section-number-4">9.1.1</span> R packages</h4>
<div class="outline-text-4" id="text-9-1-1">
<ul class="org-ul">
<li>tidyverse (ggplot)</li>
<li>ggpubr (correlation methods)</li>
</ul>
</div>
</div>
<div id="outline-container-orgfbc0600" class="outline-4">
<h4 id="orgfbc0600"><span class="section-number-4">9.1.2</span> Python packages</h4>
<div class="outline-text-4" id="text-9-1-2">
<ul class="org-ul">
<li>Matplotlib</li>
<li>pandas</li>
<li>numpy</li>
<li>matplotlib_venn</li>
</ul>
</div>
</div>
</div>
<div id="outline-container-org2aa0d4b" class="outline-3">
<h3 id="org2aa0d4b"><span class="section-number-3">9.2</span> Venn Diagrams</h3>
<div class="outline-text-3" id="text-9-2">
</div>
<div id="outline-container-org6ff181f" class="outline-4">
<h4 id="org6ff181f"><span class="section-number-4">9.2.1</span> Imports</h4>
<div class="outline-text-4" id="text-9-2-1">
<p>
Change directory based on project
</p>
<div class="org-src-container">
<pre class="src src-jupyter-python"><span style="color: #fb4934;">import</span> matplotlib.pyplot <span style="color: #fb4934;">as</span> plt
<span style="color: #fb4934;">from</span> matplotlib_venn <span style="color: #fb4934;">import</span> venn2, venn2_circles, venn2_unweighted
<span style="color: #fb4934;">import</span> pandas <span style="color: #fb4934;">as</span> pd
<span style="color: #fb4934;">import</span> numpy <span style="color: #fb4934;">as</span> np
<span style="color: #fb4934;">import</span> os
plt.style.use(<span style="color: #b8bb26;">"ggplot"</span>)
os.chdir(<span style="color: #b8bb26;">"/home/david/Documents/BenoitLab/RNA-seq"</span>)
</pre>
</div>
</div>
</div>


<div id="outline-container-org5605f54" class="outline-4">
<h4 id="org5605f54"><span class="section-number-4">9.2.2</span> Venn 1</h4>
<div class="outline-text-4" id="text-9-2-2">
<div class="org-src-container">
<pre class="src src-jupyter-python"><span style="color: #83a598;">data</span> = pd.read_csv(<span style="color: #b8bb26;">"edgeR-Genes/venn-regulation.csv"</span>)
plt.style.use(<span style="color: #b8bb26;">"ggplot"</span>)
<span style="color: #83a598;">fig</span>, <span style="color: #83a598;">axes</span> = plt.subplots(<span style="color: #d3869b; font-weight: bold;">1</span>, <span style="color: #d3869b; font-weight: bold;">4</span>)


<span style="color: #83a598;">i</span> = <span style="color: #d3869b; font-weight: bold;">0</span>

<span style="color: #83a598;">maximum</span> = data[<span style="color: #b8bb26;">"GeneCount"</span>].<span style="color: #fe8019;">max</span>()

<span style="color: #fb4934;">for</span> pest <span style="color: #fb4934;">in</span> data[<span style="color: #b8bb26;">"Part"</span>].unique():
    <span style="color: #83a598;">pestdata</span> = data[data[<span style="color: #b8bb26;">"Part"</span>] == pest]
    <span style="color: #fb4934;">for</span> part <span style="color: #fb4934;">in</span> pestdata[<span style="color: #b8bb26;">"Pest"</span>].unique():
        partdata = pestdata[pestdata[<span style="color: #b8bb26;">"Pest"</span>] == part]
        x = partdata[<span style="color: #b8bb26;">"Names"</span>]
        y = partdata[<span style="color: #b8bb26;">"GeneCount"</span>]
        axes[i].bar(x, y, color=[<span style="color: #b8bb26;">"r"</span>, <span style="color: #b8bb26;">"b"</span>])
        axes[i].set_xticks(partdata[<span style="color: #b8bb26;">"Names"</span>])
        axes[i].set_ylim(<span style="color: #d3869b; font-weight: bold;">0</span>, maximum)
        <span style="color: #fb4934;">if</span> i != <span style="color: #d3869b; font-weight: bold;">0</span>:
            axes[i].get_yaxis().set_visible(<span style="color: #d3869b;">False</span>)
        axes[i].set_xlabel(part + <span style="color: #b8bb26;">"\n"</span> + pest)
        axes[<span style="color: #d3869b; font-weight: bold;">0</span>].set_ylabel(<span style="color: #b8bb26;">"Gene Count"</span>)

        i += <span style="color: #d3869b; font-weight: bold;">1</span>
fig.suptitle(<span style="color: #b8bb26;">"Gene Regulation"</span>)
fig.savefig(<span style="color: #b8bb26;">"fig.png"</span>, bbox_inches=<span style="color: #b8bb26;">"tight"</span>, dpi=<span style="color: #d3869b; font-weight: bold;">250</span>)
</pre>
</div>
</div>
</div>
<div id="outline-container-org91fafe0" class="outline-4">
<h4 id="org91fafe0"><span class="section-number-4">9.2.3</span> Venn 2</h4>
<div class="outline-text-4" id="text-9-2-3">
<div class="org-src-container">
<pre class="src src-jupyter-python">plt.style.use(<span style="color: #b8bb26;">"ggplot"</span>)

<span style="color: #fb4934;">from</span> matplotlib_venn <span style="color: #fb4934;">import</span> venn2, venn2_circles, venn2_unweighted
<span style="color: #fb4934;">from</span> matplotlib_venn <span style="color: #fb4934;">import</span> venn3, venn3_circles


<span style="color: #83a598;">deet</span> = pd.read_csv(<span style="color: #b8bb26;">"Genes/deet.csv"</span>, header=<span style="color: #d3869b;">None</span>)
deetcont = pd.read_csv(<span style="color: #b8bb26;">"Genes/deetcont.csv"</span>, header=<span style="color: #d3869b;">None</span>)
perm = pd.read_csv(<span style="color: #b8bb26;">"Genes/perm.csv"</span>, header=<span style="color: #d3869b;">None</span>)
permcont = pd.read_csv(<span style="color: #b8bb26;">"Genes/permcont.csv"</span>, header=<span style="color: #d3869b;">None</span>)
genes = pd.read_csv(<span style="color: #b8bb26;">"Genes/listofgenes.csv"</span>, header=<span style="color: #d3869b;">None</span>)
<span style="color: #fb4934;">def</span> <span style="color: #b8bb26;">compare</span>(s1, s2):
    c = <span style="color: #fe8019;">len</span>(<span style="color: #fe8019;">list</span>(<span style="color: #fe8019;">set</span>(s1.iloc[:, <span style="color: #d3869b; font-weight: bold;">0</span>]) &amp; <span style="color: #fe8019;">set</span>(s2.iloc[:, <span style="color: #d3869b; font-weight: bold;">0</span>])))
    <span style="color: #fb4934;">return</span>(c)

diff = compare(deet, genes)
<span style="color: #fe8019;">len</span>(deet)
<span style="color: #83a598;">fig</span>, <span style="color: #83a598;">axes</span> = plt.subplots(<span style="color: #d3869b; font-weight: bold;">2</span>)
v1 = venn2_unweighted(subsets=(<span style="color: #fe8019;">len</span>(deet), <span style="color: #fe8019;">len</span>(deetcont), <span style="color: #fe8019;">len</span>(genes)-<span style="color: #fe8019;">len</span>(deetcont)-<span style="color: #fe8019;">len</span>(deet)), set_labels=(<span style="color: #b8bb26;">"Up Regulated"</span>, <span style="color: #b8bb26;">"Down Regulated"</span>), ax=axes[<span style="color: #d3869b; font-weight: bold;">0</span>])
v2 = venn2_unweighted(subsets=(<span style="color: #fe8019;">len</span>(perm), <span style="color: #fe8019;">len</span>(permcont), <span style="color: #fe8019;">len</span>(genes)-<span style="color: #fe8019;">len</span>(permcont)-<span style="color: #fe8019;">len</span>(perm)), set_labels=(<span style="color: #b8bb26;">"Up Regulated"</span>, <span style="color: #b8bb26;">"Down Regulated"</span>), ax=axes[<span style="color: #d3869b; font-weight: bold;">1</span>])
axes[<span style="color: #d3869b; font-weight: bold;">0</span>].set_title(<span style="color: #b8bb26;">"Deet"</span>)
axes[<span style="color: #d3869b; font-weight: bold;">1</span>].set_title(<span style="color: #b8bb26;">"Perm"</span>)
fig.tight_layout()
fig.savefig(<span style="color: #b8bb26;">"VennDiagrams/RevisedDiagrams/DeetnPerm.png"</span>, bbox_inxhes=<span style="color: #b8bb26;">"tight"</span>, dpi=<span style="color: #d3869b; font-weight: bold;">250</span>)
plt.show()
</pre>
</div>
</div>
</div>
<div id="outline-container-orgdec66b7" class="outline-4">
<h4 id="orgdec66b7"><span class="section-number-4">9.2.4</span> Venn 3</h4>
<div class="outline-text-4" id="text-9-2-4">
<p>
This compares the expression of body and legs when compared to the total.
</p>
<div class="org-src-container">
<pre class="src src-jupyter-python"><span style="color: #83a598;">genes</span> = pd.read_csv(<span style="color: #b8bb26;">"GeneReference/listofgenes.csv"</span>, header=<span style="color: #d3869b;">None</span>)
body = pd.read_csv(<span style="color: #b8bb26;">"GeneReference/Body.csv"</span>)
leg = pd.read_csv(<span style="color: #b8bb26;">"GeneReference/Leg.csv"</span>)

<span style="color: #83a598;">fig</span>, <span style="color: #83a598;">axes</span> = plt.subplots()

v = venn2_unweighted(subsets=(<span style="color: #fe8019;">len</span>(body), <span style="color: #fe8019;">len</span>(leg),<span style="color: #fe8019;">len</span>(genes)-<span style="color: #fe8019;">len</span>(leg) - <span style="color: #fe8019;">len</span>(body)), set_labels=(<span style="color: #b8bb26;">"Body"</span>, <span style="color: #b8bb26;">"Leg"</span>), ax=axes)
v.get_patch_by_id(<span style="color: #b8bb26;">"10"</span>).set_color(<span style="color: #b8bb26;">"blue"</span>)
v.get_patch_by_id(<span style="color: #b8bb26;">"11"</span>).set_color(<span style="color: #b8bb26;">"blue"</span>)
v.get_patch_by_id(<span style="color: #b8bb26;">"01"</span>).set_color(<span style="color: #b8bb26;">"grey"</span>)
v.get_patch_by_id(<span style="color: #b8bb26;">"01"</span>).set_alpha(<span style="color: #d3869b; font-weight: bold;">0.5</span>)
axes.set_title(<span style="color: #b8bb26;">"Leg vs Body"</span>)
fig.tight_layout()
fig.savefig(<span style="color: #b8bb26;">"figures/figure2/Deseq-BodyvsLeg.png"</span>, bbox_inches=<span style="color: #b8bb26;">"tight"</span>, dpi=<span style="color: #d3869b; font-weight: bold;">250</span>)
fig.savefig(<span style="color: #b8bb26;">"figures/figure2/Deseq-BodyvsLeg.pdf"</span>, bbox_inches=<span style="color: #b8bb26;">"tight"</span>, dpi=<span style="color: #d3869b; font-weight: bold;">250</span>)

plt.show()
</pre>
</div>


<div id="orgecdf245" class="figure">
<p><img src="./.ob-jupyter/2a5270b9b5cd24647389aedd0601cb8de1d7b70d.png" alt="2a5270b9b5cd24647389aedd0601cb8de1d7b70d.png" />
</p>
</div>
</div>
</div>

<div id="outline-container-orgfc39039" class="outline-4">
<h4 id="orgfc39039"><span class="section-number-4">9.2.5</span> Venn 4</h4>
<div class="outline-text-4" id="text-9-2-5">
<div class="org-src-container">
<pre class="src src-jupyter-python"><span style="color: #83a598;">genes</span> = pd.read_csv(<span style="color: #b8bb26;">"Genes/listofgenes.csv"</span>, header=<span style="color: #d3869b;">None</span>)
body = pd.read_csv(<span style="color: #b8bb26;">"Genes/Body.csv"</span>)
upDeetBody = pd.read_csv(<span style="color: #b8bb26;">"Genes/Genelists/upDeetBody.csv"</span>)
downDeetBody = pd.read_csv(<span style="color: #b8bb26;">"Genes/Genelists/downDeetBody.csv"</span>)


<span style="color: #83a598;">fig</span>, <span style="color: #83a598;">axes</span> = plt.subplots()

v = venn2_unweighted(subsets=(<span style="color: #fe8019;">len</span>(upDeetBody), <span style="color: #fe8019;">len</span>(downDeetBody), <span style="color: #fe8019;">len</span>(body)- <span style="color: #fe8019;">len</span>(upDeetBody)-<span style="color: #fe8019;">len</span>(downDeetBody)), set_labels=(<span style="color: #b8bb26;">"Up Regulated"</span>, <span style="color: #b8bb26;">"Down Regulated"</span>), ax=axes)
axes.set_title(<span style="color: #b8bb26;">"Deet Expression in Body Genes"</span>)
fig.tight_layout()
fig.savefig(<span style="color: #b8bb26;">"VennDiagrams/RevisedDiagrams/DeetBody.png"</span>)
plt.show()
</pre>
</div>
</div>
</div>
<div id="outline-container-orgc131053" class="outline-4">
<h4 id="orgc131053"><span class="section-number-4">9.2.6</span> Venn 5</h4>
<div class="outline-text-4" id="text-9-2-6">
<div class="org-src-container">
<pre class="src src-jupyter-python"><span style="color: #83a598;">leg</span> = pd.read_csv(<span style="color: #b8bb26;">"Genes/Leg.csv"</span>)
<span style="color: #83a598;">upDeetLeg</span> = pd.read_csv(<span style="color: #b8bb26;">"Genes/Genelists/upDeetLeg.csv"</span>)
<span style="color: #83a598;">downDeetLeg</span> = pd.read_csv(<span style="color: #b8bb26;">"Genes/Genelists/downDeetLeg.csv"</span>)

<span style="color: #83a598;">fig</span>, <span style="color: #83a598;">axes</span> = plt.subplots()
<span style="color: #83a598;">v</span> = venn2_unweighted(subsets=(<span style="color: #fe8019;">len</span>(upDeetLeg), <span style="color: #fe8019;">len</span>(downDeetLeg), <span style="color: #fe8019;">len</span>(leg)-<span style="color: #fe8019;">len</span>(upDeetLeg)-<span style="color: #fe8019;">len</span>(downDeetLeg)), set_labels=(<span style="color: #b8bb26;">"Up Regulated"</span>, <span style="color: #b8bb26;">"Down Regulated"</span>), ax=axes)
axes.set_title(<span style="color: #b8bb26;">"Deet Expression in Leg Genes"</span>)
fig.tight_layout()
fig.savefig(<span style="color: #b8bb26;">"VennDiagrams/RevisedDiagrams/DeetLeg.png"</span>)
</pre>
</div>
</div>
</div>
<div id="outline-container-org06b0a96" class="outline-4">
<h4 id="org06b0a96"><span class="section-number-4">9.2.7</span> Venn 6</h4>
<div class="outline-text-4" id="text-9-2-7">
<div class="org-src-container">
<pre class="src src-jupyter-python"><span style="color: #83a598;">leg</span> = pd.read_csv(<span style="color: #b8bb26;">"Genes/Leg.csv"</span>)
<span style="color: #83a598;">upPermLeg</span> = pd.read_csv(<span style="color: #b8bb26;">"Genes/Genelists/upPermLeg.csv"</span>)
<span style="color: #83a598;">downPermLeg</span> = pd.read_csv(<span style="color: #b8bb26;">"Genes/Genelists/downPermLeg.csv"</span>)

<span style="color: #83a598;">fig</span>, <span style="color: #83a598;">axes</span> = plt.subplots()
<span style="color: #83a598;">v</span> = venn2_unweighted(subsets=(<span style="color: #fe8019;">len</span>(upPermLeg), <span style="color: #fe8019;">len</span>(downPermLeg), <span style="color: #fe8019;">len</span>(leg)-<span style="color: #fe8019;">len</span>(downPermLeg)-<span style="color: #fe8019;">len</span>(upPermLeg)), set_labels=(<span style="color: #b8bb26;">"Up Regulated"</span>, <span style="color: #b8bb26;">"Down Regulated"</span>), ax=axes)
axes.set_title(<span style="color: #b8bb26;">"Perm Expression in Leg Genes"</span>)
fig.tight_layout()
fig.savefig(<span style="color: #b8bb26;">"VennDiagrams/RevisedDiagrams/PermLeg.png"</span>)
</pre>
</div>
</div>
</div>
<div id="outline-container-orga1560e1" class="outline-4">
<h4 id="orga1560e1"><span class="section-number-4">9.2.8</span> Venn 7</h4>
<div class="outline-text-4" id="text-9-2-8">
<div class="org-src-container">
<pre class="src src-jupyter-python"><span style="color: #83a598;">body</span> = pd.read_csv(<span style="color: #b8bb26;">"Genes/Body.csv"</span>)
<span style="color: #83a598;">upPermBody</span> = pd.read_csv(<span style="color: #b8bb26;">"Genes/Genelists/upPermBody.csv"</span>)
<span style="color: #83a598;">downPermBody</span> = pd.read_csv(<span style="color: #b8bb26;">"Genes/Genelists/downPermLeg.csv"</span>)

<span style="color: #83a598;">fig</span>, <span style="color: #83a598;">axes</span> = plt.subplots()
<span style="color: #83a598;">v</span> = venn2_unweighted(subsets=(<span style="color: #fe8019;">len</span>(upPermBody), <span style="color: #fe8019;">len</span>(downPermBody), <span style="color: #fe8019;">len</span>(body)-<span style="color: #fe8019;">len</span>(upPermBody)-<span style="color: #fe8019;">len</span>(downPermBody)), set_labels=(<span style="color: #b8bb26;">"Up Regulated"</span>, <span style="color: #b8bb26;">"Down Regulated"</span>), ax=axes)
axes.set_title(<span style="color: #b8bb26;">"Perm Expression in Body Genes"</span>)
fig.tight_layout()
fig.savefig(<span style="color: #b8bb26;">"VennDiagrams/RevisedDiagrams/PermBody.png"</span>)
</pre>
</div>
</div>
</div>
<div id="outline-container-orgbae9859" class="outline-4">
<h4 id="orgbae9859"><span class="section-number-4">9.2.9</span> Venn 8</h4>
<div class="outline-text-4" id="text-9-2-9">
<div class="org-src-container">
<pre class="src src-jupyter-python"><span style="color: #83a598;">genes</span> = pd.read_csv(<span style="color: #b8bb26;">"Genes/listofgenes.csv"</span>)
<span style="color: #83a598;">upPerm</span> = pd.read_csv(<span style="color: #b8bb26;">"Genes/Genelists/upPerm.csv"</span>)
<span style="color: #83a598;">downPerm</span> = pd.read_csv(<span style="color: #b8bb26;">"Genes/Genelists/downPerm.csv"</span>)

<span style="color: #83a598;">fig</span>, <span style="color: #83a598;">axes</span> = plt.subplots()
<span style="color: #83a598;">v</span> = venn2_unweighted(subsets=(<span style="color: #fe8019;">len</span>(upPerm), <span style="color: #fe8019;">len</span>(downPerm), <span style="color: #fe8019;">len</span>(genes)-<span style="color: #fe8019;">len</span>(upPerm)-<span style="color: #fe8019;">len</span>(downPerm)), set_labels=(<span style="color: #b8bb26;">"Up Regulated"</span>, <span style="color: #b8bb26;">"Down Regulated"</span>), ax=axes)
axes.set_title(<span style="color: #b8bb26;">"Perm Expression"</span>)
fig.tight_layout()
fig.savefig(<span style="color: #b8bb26;">"VennDiagrams/RevisedDiagrams/Perm.png"</span>)
</pre>
</div>
</div>
</div>
<div id="outline-container-org240a2ed" class="outline-4">
<h4 id="org240a2ed"><span class="section-number-4">9.2.10</span> Venn 9</h4>
<div class="outline-text-4" id="text-9-2-10">
<p>
Body vs leg EdgeR
</p>
<div class="org-src-container">
<pre class="src src-jupyter-python"><span style="color: #83a598;">genes</span> = pd.read_csv(<span style="color: #b8bb26;">"GeneReference/listofgenes.csv"</span>, header=<span style="color: #d3869b;">None</span>)
body = pd.read_csv(<span style="color: #b8bb26;">"edgeR-Genes/bodyExpr.csv"</span>)
leg = pd.read_csv(<span style="color: #b8bb26;">"edgeR-Genes/legExpr.csv"</span>)

<span style="color: #83a598;">fig</span>, <span style="color: #83a598;">axes</span> = plt.subplots()

v = venn2_unweighted(subsets=(<span style="color: #fe8019;">len</span>(body), <span style="color: #fe8019;">len</span>(leg),<span style="color: #fe8019;">len</span>(genes)-<span style="color: #fe8019;">len</span>(leg) - <span style="color: #fe8019;">len</span>(body)), set_labels=(<span style="color: #b8bb26;">"Body"</span>, <span style="color: #b8bb26;">"Leg"</span>), ax=axes)
v.get_patch_by_id(<span style="color: #b8bb26;">"10"</span>).set_color(<span style="color: #b8bb26;">"blue"</span>)
v.get_patch_by_id(<span style="color: #b8bb26;">"11"</span>).set_color(<span style="color: #b8bb26;">"blue"</span>)
v.get_patch_by_id(<span style="color: #b8bb26;">"01"</span>).set_color(<span style="color: #b8bb26;">"grey"</span>)
v.get_patch_by_id(<span style="color: #b8bb26;">"01"</span>).set_alpha(<span style="color: #d3869b; font-weight: bold;">0.5</span>)
axes.set_title(<span style="color: #b8bb26;">"Leg vs Body"</span>)
fig.tight_layout()
fig.savefig(<span style="color: #b8bb26;">"figures/figure2/EdgeR-BodyvsLeg.png"</span>, bbox_inches=<span style="color: #b8bb26;">"tight"</span>, dpi=<span style="color: #d3869b; font-weight: bold;">250</span>)
fig.savefig(<span style="color: #b8bb26;">"figures/figure2/EdgeR-BodyvsLeg.pdf"</span>, bbox_inches=<span style="color: #b8bb26;">"tight"</span>, dpi=<span style="color: #d3869b; font-weight: bold;">250</span>)

plt.show()
</pre>
</div>


<div id="orge881a97" class="figure">
<p><img src="./.ob-jupyter/38b67feca428f2217e85968601099bf697f1bfd5.png" alt="38b67feca428f2217e85968601099bf697f1bfd5.png" />
</p>
</div>
</div>
</div>
</div>

<div id="outline-container-org9896931" class="outline-3">
<h3 id="org9896931"><span class="section-number-3">9.3</span> Pearson Correlation</h3>
<div class="outline-text-3" id="text-9-3">
<p>
This correlates the results from edgeR and Deseq2. Using this, the correlation can make the analysis tool interchangeable.
</p>
</div>
<div id="outline-container-org83e4865" class="outline-4">
<h4 id="org83e4865"><span class="section-number-4">9.3.1</span> Libraries</h4>
<div class="outline-text-4" id="text-9-3-1">
<div class="org-src-container">
<pre class="src src-jupyter-R"><span style="color: #d3869b;">library</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">"tidyverse"</span><span style="color: #d3869b;">)</span>
<span style="color: #d3869b;">library</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">"ggpubr"</span><span style="color: #d3869b;">)</span>
</pre>
</div>

<pre class="example" id="org47469c0">
── [1mAttaching packages[22m ─────────────────────────────────────── tidyverse 1.3.1 ──

[32m✔[39m [34mggplot2[39m 3.3.3     [32m✔[39m [34mpurrr  [39m 0.3.4
[32m✔[39m [34mtibble [39m 3.1.2     [32m✔[39m [34mdplyr  [39m 1.0.6
[32m✔[39m [34mtidyr  [39m 1.1.3     [32m✔[39m [34mstringr[39m 1.4.0
[32m✔[39m [34mreadr  [39m 1.4.0     [32m✔[39m [34mforcats[39m 0.5.1

── [1mConflicts[22m ────────────────────────────────────────── tidyverse_conflicts() ──
[31m✖[39m [34mdplyr[39m::[32mfilter()[39m masks [34mstats[39m::filter()
[31m✖[39m [34mdplyr[39m::[32mlag()[39m    masks [34mstats[39m::lag()
</pre>
</div>
</div>

<div id="outline-container-orge891897" class="outline-4">
<h4 id="orge891897"><span class="section-number-4">9.3.2</span> Import the data</h4>
<div class="outline-text-4" id="text-9-3-2">
<div class="org-src-container">
<pre class="src src-jupyter-R">EdgeR <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">read.csv</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">"/home/david/Documents/BenoitLab/RNA-seq/edgeR-Genes/bodyExpr.csv"</span><span style="color: #d3869b;">)</span>
Deseq <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">read.csv</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">"/home/david/Documents/BenoitLab/RNA-seq/Deseq-Genes/total.csv"</span><span style="color: #d3869b;">)</span>
common <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">intersect</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">row.names</span><span style="color: #d3869b;">(</span>Deseq<span style="color: #d3869b;">)</span>, <span style="color: #b8bb26;">row.names</span><span style="color: #d3869b;">(</span>EdgeR<span style="color: #d3869b;">))</span>
EdgeRCommon <span style="color: #d3869b;">&lt;-</span> EdgeR<span style="color: #d3869b;">[</span>common, <span style="color: #d3869b;">]</span>
DeseqCommon <span style="color: #d3869b;">&lt;-</span> Deseq<span style="color: #d3869b;">[</span>common, <span style="color: #d3869b;">]</span>

ECol <span style="color: #d3869b;">&lt;-</span> EdgeRCommon<span style="color: #d3869b;">[</span><span style="color: #b8bb26;">"logFC.partbody"</span><span style="color: #d3869b;">]</span>
DCol <span style="color: #d3869b;">&lt;-</span> <span style="color: #d3869b;">-</span><span style="color: #d3869b; font-weight: bold;">1</span>*DeseqCommon<span style="color: #d3869b;">[</span><span style="color: #b8bb26;">"log2FoldChange"</span><span style="color: #d3869b;">]</span>
newD <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">data.frame</span><span style="color: #d3869b;">(</span>DCol, ECol<span style="color: #d3869b;">)</span>
</pre>
</div>
</div>
</div>

<div id="outline-container-orgf9ff952" class="outline-4">
<h4 id="orgf9ff952"><span class="section-number-4">9.3.3</span> Plot the data</h4>
<div class="outline-text-4" id="text-9-3-3">
<div class="org-src-container">
<pre class="src src-jupyter-R"><span style="color: #d3869b;">library</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">"ggpubr"</span><span style="color: #d3869b;">)</span>
pearson <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">ggscatter</span><span style="color: #d3869b;">(</span>newD, x <span style="color: #d3869b;">=</span> <span style="color: #b8bb26;">"log2FoldChange"</span>, y <span style="color: #d3869b;">=</span> <span style="color: #b8bb26;">"logFC.partbody"</span>,
          add <span style="color: #d3869b;">=</span> <span style="color: #b8bb26;">"reg.line"</span>, conf.int <span style="color: #d3869b;">=</span> <span style="color: #fabd2f;">TRUE</span>,
          cor.coef <span style="color: #d3869b;">=</span> <span style="color: #fabd2f;">TRUE</span>, cor.method <span style="color: #d3869b;">=</span> <span style="color: #b8bb26;">"pearson"</span>,
          xlab <span style="color: #d3869b;">=</span> <span style="color: #b8bb26;">"Deseq"</span>, ylab <span style="color: #d3869b;">=</span> <span style="color: #b8bb26;">"EdgeR"</span><span style="color: #d3869b;">)</span>
pearson
<span style="color: #b8bb26;">ggsave</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">"/home/david/Documents/BenoitLab/RNA-seq/figures/figure1/pearson.png"</span>, pearson, <span style="color: #b8bb26;">png</span><span style="color: #d3869b;">())</span>
<span style="color: #b8bb26;">ggsave</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">"/home/david/Documents/BenoitLab/RNA-seq/figures/figure1/pearson.pdf"</span>, pearson, <span style="color: #b8bb26;">pdf</span><span style="color: #d3869b;">())</span>

</pre>
</div>

<pre class="example">
`geom_smooth()` using formula 'y ~ x'

Saving 6.67 x 6.67 in image

`geom_smooth()` using formula 'y ~ x'

Saving 7 x 7 in image

`geom_smooth()` using formula 'y ~ x'

</pre>


<div id="orgd98d93a" class="figure">
<p><img src="./.ob-jupyter/f1067a6954fd040fbf8de1c030201143a06483aa.png" alt="f1067a6954fd040fbf8de1c030201143a06483aa.png" />
</p>
</div>
</div>
</div>
<div id="outline-container-orgd779246" class="outline-4">
<h4 id="orgd779246"><span class="section-number-4">9.3.4</span> Venn Diagram</h4>
<div class="outline-text-4" id="text-9-3-4">
<p>
Doesn&rsquo;t do what I wanted
</p>
<div class="org-src-container">
<pre class="src src-jupyter-R"><span style="color: #d3869b;">library</span><span style="color: #d3869b;">(</span>VennDiagram<span style="color: #d3869b;">)</span>
ECol <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">row.names</span><span style="color: #d3869b;">(</span>EdgeR<span style="color: #d3869b;">[</span><span style="color: #b8bb26;">"logFC.partbody"</span><span style="color: #d3869b;">])</span>
DCol <span style="color: #d3869b;">&lt;-</span> <span style="color: #b8bb26;">row.names</span><span style="color: #d3869b;">(</span><span style="color: #d3869b;">-</span><span style="color: #d3869b; font-weight: bold;">1</span>*Deseq<span style="color: #d3869b;">[</span><span style="color: #b8bb26;">"log2FoldChange"</span><span style="color: #d3869b;">])</span>
<span style="color: #b8bb26;">venn.diagram</span><span style="color: #d3869b;">(</span>
  x <span style="color: #d3869b;">=</span> <span style="color: #b8bb26;">list</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">na.omit</span><span style="color: #d3869b;">(</span>ECol<span style="color: #d3869b;">)</span>, <span style="color: #b8bb26;">na.omit</span><span style="color: #d3869b;">(</span>DCol<span style="color: #d3869b;">))</span>,
  category.names <span style="color: #d3869b;">=</span> <span style="color: #b8bb26;">c</span><span style="color: #d3869b;">(</span><span style="color: #b8bb26;">"EdgeR"</span>, <span style="color: #b8bb26;">"Deseq"</span><span style="color: #d3869b;">)</span>,
  filename <span style="color: #d3869b;">=</span> <span style="color: #b8bb26;">"/home/david/Documents/BenoitLab/RNA-seq/figures/figure1/vennBody.png"</span>,
  output<span style="color: #d3869b;">=</span><span style="color: #fabd2f;">TRUE</span>
<span style="color: #d3869b;">)</span>
</pre>
</div>
</div>
</div>
</div>
<div id="outline-container-orgc8ebbcb" class="outline-3">
<h3 id="orgc8ebbcb"><span class="section-number-3">9.4</span> Dotplot</h3>
<div class="outline-text-3" id="text-9-4">
<p>
This figure plots the enrichment of specific genes. Ideal for small amounts of genes.
</p>
</div>
<div id="outline-container-org30cd483" class="outline-4">
<h4 id="org30cd483"><span class="section-number-4">9.4.1</span> imports</h4>
<div class="outline-text-4" id="text-9-4-1">
<div class="org-src-container">
<pre class="src src-jupyter-python"><span style="color: #fb4934;">import</span> numpy <span style="color: #fb4934;">as</span> np
<span style="color: #fb4934;">import</span> matplotlib.pyplot <span style="color: #fb4934;">as</span> plt
<span style="color: #fb4934;">import</span> matplotlib
<span style="color: #fb4934;">import</span> pandas <span style="color: #fb4934;">as</span> pd
</pre>
</div>
</div>
</div>
<div id="outline-container-org07063b6" class="outline-4">
<h4 id="org07063b6"><span class="section-number-4">9.4.2</span> Theme settings</h4>
<div class="outline-text-4" id="text-9-4-2">
<div class="org-src-container">
<pre class="src src-jupyter-python">plt.<span style="color: #83a598;">rcParams</span>[<span style="color: #b8bb26;">"font.family"</span>] = <span style="color: #b8bb26;">"Bahnschrift"</span>
<span style="color: #83a598;">cmap</span> = matplotlib.colors.LinearSegmentedColormap.from_list(
    <span style="color: #b8bb26;">""</span>, [<span style="color: #b8bb26;">"dodgerblue"</span>, <span style="color: #b8bb26;">"lightgrey"</span>]
)

</pre>
</div>
</div>
</div>
<div id="outline-container-orge13fb43" class="outline-4">
<h4 id="orge13fb43"><span class="section-number-4">9.4.3</span> Import</h4>
<div class="outline-text-4" id="text-9-4-3">
<p>
Use the pandas package to import the csv file
</p>
<div class="org-src-container">
<pre class="src src-jupyter-python"><span style="color: #83a598;">dotplotdata</span> = pd.read_csv(<span style="color: #b8bb26;">"../GeneReference/dotplotdata.csv"</span>)  <span style="color: #928374;"># </span><span style="color: #928374;">infers header existance</span>

<span style="color: #83a598;">pesticide</span> = dotplotdata.Category
<span style="color: #83a598;">pathway</span> = dotplotdata.Pathway
<span style="color: #83a598;">enrichmentratio</span> = dotplotdata.Enrichment
<span style="color: #83a598;">pval</span> = dotplotdata[<span style="color: #b8bb26;">"P-Value"</span>]
</pre>
</div>
</div>
</div>
<div id="outline-container-orgc785c6e" class="outline-4">
<h4 id="orgc785c6e"><span class="section-number-4">9.4.4</span> Plot the figure</h4>
<div class="outline-text-4" id="text-9-4-4">
<div class="org-src-container">
<pre class="src src-jupyter-python"><span style="color: #83a598;">fig</span>, <span style="color: #83a598;">ax</span> = plt.subplots(<span style="color: #d3869b; font-weight: bold;">1</span>)
<span style="color: #83a598;">p</span> = ax.scatter(
    pathway, pesticide, c=pval, s=enrichmentratio * <span style="color: #d3869b; font-weight: bold;">3</span>, cmap=cmap, vmin=<span style="color: #d3869b; font-weight: bold;">0</span>, vmax=<span style="color: #d3869b; font-weight: bold;">0.05</span>
)
plt.legend(loc=<span style="color: #b8bb26;">"lower left"</span>, markerscale=<span style="color: #d3869b; font-weight: bold;">2.0</span>, scatterpoints=<span style="color: #d3869b; font-weight: bold;">1</span>, fontsize=<span style="color: #d3869b; font-weight: bold;">10</span>)
plt.xticks(rotation=<span style="color: #b8bb26;">"vertical"</span>)

<span style="color: #928374;"># </span><span style="color: #928374;">Legend:</span>
handles, labels = p.legend_elements(prop=<span style="color: #b8bb26;">"sizes"</span>, alpha=<span style="color: #d3869b; font-weight: bold;">0.6</span>, num=<span style="color: #d3869b; font-weight: bold;">4</span>, func=<span style="color: #fb4934;">lambda</span> x : x/<span style="color: #d3869b; font-weight: bold;">3</span>)
legend2 = ax.legend(
    handles,
    labels,
    loc=<span style="color: #b8bb26;">"upper right"</span>,
    title=<span style="color: #b8bb26;">"Enrichment Ratio"</span>,
    bbox_to_anchor=(<span style="color: #d3869b; font-weight: bold;">1.4</span>, <span style="color: #d3869b; font-weight: bold;">1.1</span>),
)

<span style="color: #928374;"># </span><span style="color: #928374;">Colorbar:</span>
cbar = fig.colorbar(p, shrink=<span style="color: #d3869b; font-weight: bold;">0.5</span>)
cbar.ax.set_ylabel(<span style="color: #b8bb26;">"p Value"</span>, rotation=<span style="color: #d3869b; font-weight: bold;">270</span>, labelpad=<span style="color: #d3869b; font-weight: bold;">15</span>)

<span style="color: #928374;"># </span><span style="color: #928374;">Title, labels, and visual modifications:</span>
plt.title(<span style="color: #b8bb26;">"Metabolic Pathway Enrichment"</span>)
plt.xlabel(<span style="color: #b8bb26;">"Metabolic Pathway"</span>)
plt.ylabel(<span style="color: #b8bb26;">"Pesticide"</span>)
ax.grid(axis=<span style="color: #b8bb26;">"x"</span>, linestyle=<span style="color: #b8bb26;">"dotted"</span>)
ax.spines[<span style="color: #b8bb26;">"top"</span>].set_visible(<span style="color: #d3869b;">False</span>)
ax.spines[<span style="color: #b8bb26;">"right"</span>].set_visible(<span style="color: #d3869b;">False</span>)

plt.show()
</pre>
</div>
</div>
</div>
<div id="outline-container-org2109e46" class="outline-4">
<h4 id="org2109e46"><span class="section-number-4">9.4.5</span> Save the Figure</h4>
<div class="outline-text-4" id="text-9-4-5">
<div class="org-src-container">
<pre class="src src-jupyter-python">plt.savefig(<span style="color: #b8bb26;">'dotplot.pdf'</span>, dpi=<span style="color: #d3869b; font-weight: bold;">300</span>, bbox_inches=<span style="color: #b8bb26;">'tight'</span>)
plt.savefig(<span style="color: #b8bb26;">'dotplot.jpg'</span>, dpi=<span style="color: #d3869b; font-weight: bold;">300</span>, bbox_inches=<span style="color: #b8bb26;">'tight'</span>)
</pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div id="postamble" class="status">
<p class="author">Author: David Lewis</p>
<p class="date">Created: 2021-05-27 Thu 10:55</p>
</div>
</body>
</html>
